
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    <meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

    <title>Genomics &mdash; Bioinformatics at COMAV  documentation</title>
    <link rel="stylesheet" href="_static/style.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <link rel="stylesheet" href="_static/style.css" type="text/css" />
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    './',
        VERSION:     '',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="_static/documentation_options.js"></script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <link rel="shortcut icon" href="_static/favicon.ico"/>
    <link rel="search" title="Search" href="search.html" />
    <link rel="top" title="Bioinformatics at COMAV  documentation" href="index.html" />
    <link rel="next" title="Services" href="services.html" />
    <link rel="prev" title="Bioinformatics & genomics" href="index.html" /> 
  </head>
  <body>

<div class="body">
    <header>
        <h1><a href="index.html">Bioinformatics at COMAV</a></h1>
    </header>


    <nav class="horizontal_nav page_nav">
        <ul>
            <li><a href="/index.html">Bioinformatics & genomics</a>
                <ul>
                    <li><a href="/services.html">Services</a></li>
                    <li><a href="/people.html">People</a></li>
                    <li><a href="/publications.html">Publications</a></li>
                    <li><a href="/getting_here.html">Getting here</a></li>
<!--                    <li><a href="http://bioinf.comav.upv.es/notes/">Wiki</a></li>-->
               </ul>
            </li>
            <li><a href="/software.html">Software</a>
                <ul>

                    <li><a href="https://github.com/JoseBlanca/">Git</a></li><!--
                    <li><a href="/seq_crumbs/index.html">seq_crumbs</a></li>
                    <li><a href="/ngs_backbone/index.html">ngs_backbone</a></li>
                    <li><a href="/clean_reads/index.html">clean_reads</a></li>
                    <li><a href="/sff_extract/index.html">sff_extract</a></li>
                    <li><a href="/psubprocess/index.html">psubprocess</a></li>-->
               </ul>
            </li>
            <li><a href="/databases.html">Databases</a>
                <ul>
                    <li><a href="https://melogene.upv.es">Melogene</a></li>
                    <li><a href="https://cucurbigene.upv.es">Cucurbigene</a></li>
               </ul>
            </li>
            <li><a ref=''>Downloads</a>
                <ul>
                    <li><a href="/downloads/zucchini">Zucchini</a></li>
               </ul>
            </li>
            <li><a href="/courses.html">Courses</a>
                <ul>
                    <li><a href="/courses/intro_bioinf/index.html">Bionformática básica</a></li>
                    <li><a href="/courses/sequence_analysis/index.html">NGS sequence analysis</a></li>
                    <li><a href="/courses/linux/index.html">Linux y Python</a></li>
                    <li><a href="/courses/unix/index.html">Linux para Biologos</a></li>
               </ul>
        </ul>
    </nav>

    <main>
                
    
  <section id="genomics">
<h1>Genomics<a class="headerlink" href="#genomics" title="Permalink to this headline">¶</a></h1>
<section id="ngs-sequencing">
<h2>NGS sequencing<a class="headerlink" href="#ngs-sequencing" title="Permalink to this headline">¶</a></h2>
<p>Our group have experience and capacity to sequence and analyse NGS transcriptomes. We have developed normalized cDNA libraries from tomato, melon and pumpkin, that have been sequenced in external services by 454, Illumina and Solid. We are able to analyze the sequences and perform transcriptome assembly, annotation and SNP identification.</p>
</section>
<section id="transcriptomic-analysis">
<h2>Transcriptomic analysis<a class="headerlink" href="#transcriptomic-analysis" title="Permalink to this headline">¶</a></h2>
<p>There are different technologies to analyze gene expression, that have evolved from Northern blot to microarrays or massive sequencing. Our group has participated in several projects related with gene expression analysis and has wide experience in experimental techniques and data analysis. We are specialized in microarrays data analysis, using different platforms, such as Affymetrix or NimbleGene. We have developed several projects in collaboration with other groups in tomato and melon.</p>
</section>
<section id="ecotilling">
<h2>EcoTILLING<a class="headerlink" href="#ecotilling" title="Permalink to this headline">¶</a></h2>
<p>We have developed an EcoTILLING platform in pepper sp., this platform contains more than two hundred accessions of Capsicum annum, C. baccatum, C. chinnense, C. frutescens and C. pubescens. We have our own source of CEL I activity. The platform have been successfully used to identified new allelic variants of EIF4E and EIF4Eiso factors related with virus resistance. This platform can be screened using DNA or RNA as starting material, so it is not necessary to analyze the genes exon by exon and the reaction number can be reduced.</p>
</section>
<section id="molecular-markers">
<h2>Molecular markers<a class="headerlink" href="#molecular-markers" title="Permalink to this headline">¶</a></h2>
<p>Our group have extensive experience in molecular markers development and analysis. We have used markers for gene mapping, molecular breeding, phylogenetic and population studies, etc. The development of high-throughput sequencing and genotyping technologies has made easier the identification and use of SNPs in any species, even in those without previous genomic information. We have used these technologies to identify SNPs in tomato and other species, and  we have verified and employed the markers using several techniques such as Veracode, HRM or CELI.</p>
</section>
</section>



        

    </main>

</div>
    <footer>
            <a rel="license" href="https://creativecommons.org/licenses/by/4.0/"><img alt="Licencia de Creative Commons" style="border-width:0" src="https://i.creativecommons.org/l/by/4.0/88x31.png" /></a><br /><span xmlns:dct="https://purl.org/dc/terms/" property="dct:title">Bioinformatics at COMAV</span> is licensed under a <a rel="license" href="https://creativecommons.org/licenses/by/4.0/">Creative Commons Reconocimiento 4.0 Internacional License</a>.
    </footer>

<script type="text/javascript">
var gaJsHost = (("https:" == document.location.protocol) ? "https://ssl." : "http://www.");
document.write(unescape("%3Cscript src='" + gaJsHost + "google-analytics.com/ga.js' type='text/javascript'%3E%3C/script%3E"));
</script>
<script type="text/javascript">
try {
var pageTracker = _gat._getTracker("UA-16108274-1");
pageTracker._trackPageview();
} catch(err) {}</script>

  </body>
</html>