<!DOCTYPE html>

<html lang="en" data-content_root="../../">
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="viewport" content="width=device-width, initial-scale=1" />

    <title>Coalescence simulations &#8212; Pop lab  documentation</title>
    <link rel="stylesheet" type="text/css" href="../../_static/pygments.css?v=5ecbeea2" />
    <link rel="stylesheet" type="text/css" href="../../_static/basic.css?v=b08954a9" />
    <link rel="stylesheet" type="text/css" href="../../_static/alabaster.css?v=27fed22d" />
    <link rel="stylesheet" type="text/css" href="../../_static/sphinx-design.min.css?v=95c83b7e" />
    <script src="../../_static/documentation_options.js?v=5929fcd5"></script>
    <script src="../../_static/doctools.js?v=fd6eb6e6"></script>
    <script src="../../_static/sphinx_highlight.js?v=6ffebe34"></script>
    <script src="../../_static/design-tabs.js?v=f930bc37"></script>
    <script defer="defer" src="https://cdn.jsdelivr.net/npm/mathjax@4/tex-mml-chtml.js"></script>
    <link rel="index" title="Index" href="../../genindex.html" />
    <link rel="search" title="Search" href="../../search.html" />
   
  <link rel="stylesheet" href="../../_static/custom.css" type="text/css" />
  

  
  

  </head><body>
  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          

          <div class="body" role="main">
            
  <section id="coalescence-simulations">
<h1>Coalescence simulations<a class="headerlink" href="#coalescence-simulations" title="Link to this heading">¶</a></h1>
<p>These applications simulate a genomic region evolving through time with some mutation and recombination rates.
Unlike the case with one locus and two alleles, in these cases the simulations are not calculated forward in time, but by a <a class="reference external" href="https://en.wikipedia.org/wiki/Coalescent_theory">coalescence</a> approach.</p>
<p><a class="reference external" href="https://tskit.dev/msprime/docs/stable/intro.html">msprime</a> is the library used to run these simulations. Once the simulation is completed the software generates a genotypic matrix with the genotype for each individual and marker.</p>
<p>The different parameters are calculated using the <a class="reference external" href="https://github.com/JoseBlanca/pynei">pyNei</a> library.</p>
<section id="expected-heterozygosity">
<h2>Expected heterozygosity<a class="headerlink" href="#expected-heterozygosity" title="Link to this heading">¶</a></h2>
<p>The expected heterozygosity, a measure of genetic diversity, is calculated using the Unbiased expected heterozygosity <a class="reference external" href="https://biology-assets.anu.edu.au/GenAlEx/Welcome.html">Genalex</a> formula.</p>
<div class="math notranslate nohighlight">
\[    uh = {\frac{n}{(n - 1)}}  (1 - \sum p_i^2)
\]</div>
</section>
<section id="polymorphic-markers">
<h2>Polymorphic markers<a class="headerlink" href="#polymorphic-markers" title="Link to this heading">¶</a></h2>
<p>A variation/marker is considered polymorphic when its major allele has a frequency lower than 95%.
The simulations calculate the total number of polymorphic variants and the proportion of polymorphic variants calculated over the total number of variations.</p>
</section>
<section id="allele-frequency-spectrum">
<h2>Allele Frequency Spectrum<a class="headerlink" href="#allele-frequency-spectrum" title="Link to this heading">¶</a></h2>
<p>The Allele Frequency Spectrum is the distribution of allelic frequencies of the major allele, the most abundant allele. So, in the chart the number of polymorphic alleles for different major allele frequencies is plotted.</p>
</section>
<section id="principal-component-analysis-pca">
<h2>Principal Component Analysis (PCA)<a class="headerlink" href="#principal-component-analysis-pca" title="Link to this heading">¶</a></h2>
<p>The Principal Component Analysis is not calculated using all markers.
Before the calculation the non-polymorphic (95%) and the highly associated (r² &gt; 0.1) markers are filtered out.
With the remaining markers the genotypes are coded in a array with 0 for the major allele homozygote, 2 for a homozygote of any minor allele and 1 for the heterozygotes.
Finally, this array is used to carry out the <a class="reference external" href="https://en.wikipedia.org/wiki/Principal_component_analysis">Principal Component Analysis</a>.</p>
</section>
<section id="population-distances">
<h2>Population distances<a class="headerlink" href="#population-distances" title="Link to this heading">¶</a></h2>
<p>Distances between populations are calculated following the Jost’s Differentiation estimation (Dest) suggested by Genalex.</p>
<div class="math notranslate nohighlight">
\[D_est = \left( \frac{k}{k - 1} \right)\left( \frac{cH_T - cH_s}{1 - cH_s} \right)
\]</div>
<p>Observed Heterozygosity.</p>
<div class="math notranslate nohighlight">
\[H_o = \frac{Num.ofhets}{N}
\]</div>
<p>Observed heterozygosity, averaged across populations. The average observed heterozygosity of a collection of populations.
Here, HOs is the observed heterozygosity in the s-th population; k is the number of populations.</p>
<div class="math notranslate nohighlight">
\[\bar{H_o} = \frac{\sum H_{Os}}{k}
\]</div>
<p>Average within population heterozygosity. Identical to the mean He, being the average of the within population
expected heterozygosity across populations.</p>
<div class="math notranslate nohighlight">
\[H_s = \bar{H_E} = \sum \frac{H_{ES}}{k}
\]</div>
<p>Corrected Hs. HS for a given locus is adjusted for small population size and inbreeding by the correction of Nei and Cheeser [30], where n̂ is the harmonic mean population size for k populations, and HO is the average
observed within-population heterozygosity for the populations.</p>
<div class="math notranslate nohighlight">
\[cH_s = \frac{\hat{n}}{\hat{n} - 1} \left[ H_s - \frac{\bar{H_o}}{2\hat{n}} \right]
\]</div>
<p>Total expected heterozygosity calculated as if all populations were pooled (no subdivision).</p>
<div class="math notranslate nohighlight">
\[H_T = 1 - \sum_{i=1}^{h}\bar{p}_i^2
\]</div>
<p>Corrected Ht, adjusted for small population size and inbreeding, using the correction of Nei and Cheeser. The harmonic mean of population size over the k populations is n̂.</p>
<div class="math notranslate nohighlight">
\[cH_T = H_T + \frac{cH_S}{\hat{n}k} - \frac{\bar{H_O}}{2\hat{n}k}
\]</div>
</section>
</section>


          </div>
          
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="Main">
        <div class="sphinxsidebarwrapper">
<h1 class="logo"><a href="../../index.html">Pop lab</a></h1>









<search id="searchbox" style="display: none" role="search">
    <div class="searchformwrapper">
    <form class="search" action="../../search.html" method="get">
      <input type="text" name="q" aria-labelledby="searchlabel" autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" placeholder="Search"/>
      <input type="submit" value="Go" />
    </form>
    </div>
</search>
<script>document.getElementById('searchbox').style.display = "block"</script><h3>Navigation</h3>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../fwd/drift.html">Drift</a></li>
<li class="toctree-l1"><a class="reference internal" href="../fwd/selection.html">Selection</a></li>
<li class="toctree-l1"><a class="reference internal" href="../fwd/balancing_selection.html">Balancing selection</a></li>
<li class="toctree-l1"><a class="reference internal" href="../fwd/one_locus.html">One locus foward in time</a></li>
<li class="toctree-l1"><a class="reference internal" href="bottleneck.html">Bottlenecks and Genetic Diversity</a></li>
<li class="toctree-l1"><a class="reference internal" href="founder.html">Founder Effect and Genetic Diversity</a></li>
<li class="toctree-l1"><a class="reference internal" href="drifting_pops.html">Drifting populations</a></li>
<li class="toctree-l1"><a class="reference internal" href="admixture.html">Admixture</a></li>
</ul>

<div class="relations">
<h3>Related Topics</h3>
<ul>
  <li><a href="../../index.html">Documentation overview</a><ul>
  </ul></li>
</ul>
</div>








        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="footer">
      &#169;2024, Jose Blanca & Joaquin Cañizares.
      
      |
      Powered by <a href="https://www.sphinx-doc.org/">Sphinx 9.1.0</a>
      &amp; <a href="https://alabaster.readthedocs.io">Alabaster 1.0.0</a>
      
      |
      <a href="../../_sources/practices/coalescence/coalescence.md.txt"
          rel="nofollow">Page source</a>
    </div>

    

    
  </body>
</html>