<!DOCTYPE html>
<html lang="en">

  <head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1">

  <title>Biological Databases</title>
  <meta name="description" content="Introduction to Bioinformatics for the Biotechnology degree.">

  <link rel="stylesheet" href="/courses/biotech3/css/main.css">
  <link rel="canonical" href="/courses/biotech3/theory/databases.html">
  <link rel="alternate" type="application/rss+xml" title="Bioinformatics" href="/courses/biotech3/feed.xml">
</head>


  <body>

    <header class="site-header" role="banner">

  <div class="wrapper">

    <a class="site-title" href="/courses/biotech3/">Bioinformatics</a>

    <nav class="site-nav">
      <a href="#" class="menu-icon">
        <svg viewBox="0 0 18 15">
          <path fill="#424242" d="M18,1.484c0,0.82-0.665,1.484-1.484,1.484H1.484C0.665,2.969,0,2.304,0,1.484l0,0C0,0.665,0.665,0,1.484,0 h15.031C17.335,0,18,0.665,18,1.484L18,1.484z"/>
          <path fill="#424242" d="M18,7.516C18,8.335,17.335,9,16.516,9H1.484C0.665,9,0,8.335,0,7.516l0,0c0-0.82,0.665-1.484,1.484-1.484 h15.031C17.335,6.031,18,6.696,18,7.516L18,7.516z"/>
          <path fill="#424242" d="M18,13.516C18,14.335,17.335,15,16.516,15H1.484C0.665,15,0,14.335,0,13.516l0,0 c0-0.82,0.665-1.484,1.484-1.484h15.031C17.335,12.031,18,12.696,18,13.516L18,13.516z"/>
        </svg>
      </a>

      <div class="trigger">
        
          
          <a class="page-link" href="/courses/biotech3/practice/alignment.html">Sequence Alignment Practice</a>
          
        
          
          <a class="page-link" href="/courses/biotech3/practice/assembly.html">Assembly</a>
          
        
          
          <a class="page-link" href="/courses/biotech3/theory/blast.html">Blast</a>
          
        
          
          <a class="page-link" href="/courses/biotech3/practice/blast.html">Blast Practice</a>
          
        
          
          <a class="page-link" href="/courses/biotech3/covid.html">COVID19</a>
          
        
          
          <a class="page-link" href="/courses/biotech3/theory/databases.html">Biological Databases</a>
          
        
          
          <a class="page-link" href="/courses/biotech3/practice/emboss.html">Introduction to EMBOSS</a>
          
        
          
          <a class="page-link" href="/courses/biotech3/practice/entrez.html">NCBI database search Practice</a>
          
        
          
        
          
          <a class="page-link" href="/courses/biotech3/">Home</a>
          
        
          
          <a class="page-link" href="/courses/biotech3/practice/linux.html">Unix CLI</a>
          
        
          
        
          
          <a class="page-link" href="/courses/biotech3/practice/multiple.html">Multiple Sequence Alignment Practice</a>
          
        
          
          <a class="page-link" href="/courses/biotech3/theory/multiple.html">Multiple Sequence Alignment</a>
          
        
          
          <a class="page-link" href="/courses/biotech3/norms.html">Norms</a>
          
        
          
          <a class="page-link" href="/courses/biotech3/practice/phylogeny.html">Phylogeny practice</a>
          
        
          
          <a class="page-link" href="/courses/biotech3/theory/phylogeny.html">Phylogeny and evolution</a>
          
        
          
          <a class="page-link" href="/courses/biotech3/prev_knowledge.html">Previous Knowledge</a>
          
        
          
          <a class="page-link" href="/courses/biotech3/theory/sequence_alignment.html">Sequence Alignment</a>
          
        
          
          <a class="page-link" href="/courses/biotech3/theory/structural_annotation.html">Structural Annotation</a>
          
        
          
          <a class="page-link" href="/courses/biotech3/practice/structural_annotation.html">Structural Annotation</a>
          
        
      </div>
    </nav>

  </div>

</header>


    <main class="page-content" aria-label="Content">
      <div class="wrapper">
        <article class="post">

  <header class="post-header">
    <h1 class="post-title">Biological Databases</h1>
  </header>

  <div class="post-content">
    <h1 id="introduction">Introduction</h1>

<p>A <a href="https://en.wikipedia.org/wiki/Database">database</a> is an organized collection of data.
For instance, a list with some of the movies that we like would be a movie database:</p>

<table>
  <thead>
    <tr>
      <th style="text-align: center">ID</th>
      <th style="text-align: center">title</th>
      <th style="text-align: center">year</th>
      <th style="text-align: center">director</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td style="text-align: center">movie1</td>
      <td style="text-align: center">The player</td>
      <td style="text-align: center">1992</td>
      <td style="text-align: center">Robert Altman</td>
    </tr>
    <tr>
      <td style="text-align: center">movie2</td>
      <td style="text-align: center">Cookie’s fortune</td>
      <td style="text-align: center">1999</td>
      <td style="text-align: center">Robert Altman</td>
    </tr>
    <tr>
      <td style="text-align: center">movie 3</td>
      <td style="text-align: center">The man who shot Liberty Valance</td>
      <td style="text-align: center">1962</td>
      <td style="text-align: center">John Ford</td>
    </tr>
  </tbody>
</table>

<p>Vocabulary:</p>

<ul>
  <li><em>Entities</em>: The kind of things that we want to store in a database. E.g.: Genes, DNA sequences, bibliographical references.</li>
  <li><em>Records</em>: The particular things stored in the database. E.g.: The gene BRCA1</li>
  <li><em>Identifiers</em> or <em>key</em>: The unique name that identifies a record</li>
  <li><em>Fields</em>: The properties that an entity has. E.g.: The name, sequence and mutations of the gene</li>
</ul>

<p>In the previous movie examples the entities stored were movies, the records stored were: The player, Cookie’s fortune and The man who shot Liberty Valance.
The unique idenfiers were: movie1, movie2 and movie3.</p>

<p>So if we think on the database as a table, the table would store information about one entity, the fields would be the column headers and the records would be the table rows.</p>

<p>It is quite common to store different entities in a database.
For instance we could store movies, actors and directors or genes, sequences and mutations.
In that case, the different entities could be stored in different tables and the records on those tables would be related by their unique identifiers.
That structure would comprise a <a href="https://en.wikipedia.org/wiki/Relational_database">relational database</a>.</p>

<p>The databases usually provide mechanisms to store, search, retrieve and modify the data.</p>

<h1 id="biological-databases">Biological Databases</h1>

<p>The data repositories more relevant to the biological sciences include:</p>

<ul>
  <li>nucleotide and protein sequences</li>
  <li>protein structures</li>
  <li>genomes</li>
  <li>genetic expression</li>
  <li>bibliography</li>
</ul>

<p>Main sequence databases:</p>

<ul>
  <li><a href="http://www.ncbi.nlm.nih.gov/">NCBI</a></li>
  <li><a href="http://www.ebi.ac.uk/">EMBL</a></li>
</ul>

<p>Main protein databases:</p>

<ul>
  <li><a href="http://www.uniprot.org/">Uniprot</a></li>
  <li><a href="http://www.ebi.ac.uk/pdbe/">PDB</a></li>
  <li><a href="http://www.ncbi.nlm.nih.gov/Structure/MMDB/mmdb.shtml">MMDB</a></li>
</ul>

<p>Some genome databases:</p>

<ul>
  <li><a href="http://www.ensembl.org/index.html">ENSEMBL</a> (Human, mouse and others)</li>
  <li><a href="http://www.yeastgenome.org/">SGD</a> (Yeast)</li>
  <li><a href="https://www.arabidopsis.org/">TAIR</a> (Arabidopsis)</li>
</ul>

<p>Bibliography:</p>

<ul>
  <li><a href="http://www.ncbi.nlm.nih.gov/pubmed">Pubmed</a></li>
  <li><a href="https://apps.webofknowledge.com/">Web of Science</a></li>
</ul>

<p>Human diseases:</p>

<ul>
  <li><a href="http://www.omim.org/">OMIM</a></li>
</ul>

<p>Metabolic pathways:</p>

<ul>
  <li><a href="http://www.genome.jp/kegg/">KEGG</a></li>
</ul>

<h1 id="sequence-databases">Sequence databases</h1>

<p>A sequence database is a collection of DNA or protein sequences with some extra relevant information.
The main sequence databases are <a href="http://www.ncbi.nlm.nih.gov/genbank/">Genbank</a> and <a href="http://www.ebi.ac.uk/">EMBL</a>.
Originally they were just sequence collections, but they have grown to store different biological databases heavily interconnected and they provide powerful interfaces to search and browse the stored information.</p>

<p>The sequences submitted to any of those databases are shared between them, so any sequence could be retrieved in the european or the american database.
But they differ in the tools to search and browse the data and in some databases that provide extra information to the raw sequences like: mutations, coded proteins, bibliographical references, etc.</p>

<p>These databases are <a href="http://www.ncbi.nlm.nih.gov/genbank/statistics/">growing</a> at an ever increasing fast pace. In June of 2007 there were 73 million sequences in Genbank and in August of 2015 there were 187 millions.</p>

<p><a href="http://www.ncbi.nlm.nih.gov/genbank/statistics/"><img src="../static/genbank_growth.png" alt="Genbank growth" /></a></p>

<p>The sequences are split in these databases in different sections to ease the search.
Among others, there are sections for mRNAs, publised nucleotide sequences, genomes, and genes.</p>

<h1 id="genbank">Genbank</h1>

<p><a href="http://www.ncbi.nlm.nih.gov/genbank/">Genbank</a> is a public collection of annotated sequences hosted by the <a href="http://www.ncbi.nlm.nih.gov/">NCBI</a>.
Among other kinds of sequences Genbank includes messenger RNAs, genomic DNAs and ribosomic RNA.</p>

<p>Some characteristics:</p>

<ul>
  <li>It is a public repository, any one can send sequences to it.</li>
  <li>There are sequences of different qualities, anything submitted is stored.</li>
  <li>There could be multiple sequences for the same gene or for the same mRNA</li>
  <li>A sequence can have several versions that represent the modifications done by the authors.</li>
</ul>

<p>Due to the huge amount of sequences stored to ease the search the databases are split in different <a href="http://www.ncbi.nlm.nih.gov/HTGS/divisions.html">divisions</a>.
These divisions follow two criteria: the species and type of sequence.
Among the taxonomical divisions you can find: primate, rodent, other mammalian, invertebrate an others.
The other divisions are related to the kind of sequences like: <a href="http://www.ncbi.nlm.nih.gov/nucest/">EST</a>, <a href="https://www.ncbi.nlm.nih.gov/genbank/wgs/">WGS</a>, <a href="http://www.ncbi.nlm.nih.gov/genbank/htgs/">HTGS</a>, and many others.
If you are looking for reads comming from the Next Generation Sequencing Technologies they are stored in a special division called <a href="http://www.ncbi.nlm.nih.gov/sra">SRA</a>.</p>

<p><a href="https://trace.ncbi.nlm.nih.gov/Traces/sra/sra.cgi"><img src="../static/sra_growth.png" alt="SRA growth" /></a></p>

<p>You can find more information about GenBank in its <a href="http://www.ncbi.nlm.nih.gov/books/NBK143764/">handbook</a>.</p>

<h1 id="refseq">RefSeq</h1>

<p><a href="http://www.ncbi.nlm.nih.gov/refseq/">RefSeq</a> is a reference database curated by <a href="http://www.ncbi.nlm.nih.gov/">NCBI</a>.</p>

<p>In RefSeq there are only well annotated and good quality sequences.
It stores genomic, transcript and protein sequences and links the sequences that belong to a gene.
It just has one representative sequence for each mRNA in a particular organism and, thus, it will have as many sequences as different transcripts and proteins coded for a particular gene in a particular organism.</p>

<p>It is not the aim of ReqSeq to have any sequence, but just to have a collection of well curated sequences.
It is a secondary database.
Since RefSeq requires extra curation work it is not available for all organisms, but only for those with good quality sequences.
As of July of 2016 it has 65M proteins and 15M transcripts for 60K organisms.</p>

<h1 id="uniprot">UniProt</h1>

<p><a href="http://www.uniprot.org/">UniProt</a> is a protein database that includes information divided in two sections: Swiss-Prot and TrEMBL.
UniProt aims to store sequence and functional information for the proteins.</p>

<p>TrEMBL is automatically annotated while Swiss-Prot is reviewed manually by humans that add information by reviewing the literature.
Due to this effort Swiss-Prot has information of a higher quality, but it has less sequences than TrEMBL.</p>

<p>UniProt also hosts <a href="http://www.uniprot.org/uniref/">Uniref</a>.
This database aims to store one representative sequence for each protein without taking into account the species of origin.
It clusters all the similar proteins and picks one for every cluster as a representative.
There are clusters created at 100%, 90% and 50% identities.</p>

<h1 id="pubmed">PubMed</h1>

<p><a href="http://www.ncbi.nlm.nih.gov/pubmed">PubMed</a> is a bibliographical database that comprises biomedical literature (MEDLINE), life science journals and on-line books.
It is a good collection of publications related to biochemistry, cellular biology and medicine.
As of 2016 PubMed stores 26 million citations.</p>

<p>For each record it stores:</p>
<ul>
  <li>title</li>
  <li>authors</li>
  <li>abstract</li>
</ul>

<p>There is a related database named PubMed Cental (<a href="http://www.ncbi.nlm.nih.gov/pmc/">PMC</a>) that only includes citations of Free Access Journals.
These citations include the complete text for the papers stored.</p>

<h1 id="pdb-protein-data-bank">PDB, Protein Data Bank</h1>

<p><a href="http://www.rcsb.org/pdb/home/home.do">PDB</a> stores 3D structures for proteins and nucleic acids.</p>

<h1 id="access-to-the-information-in-genbank">Access to the information in Genbank</h1>

<p>Every database provides one or more methods to search and query the data.
It is quite common to provide a web interface in which to do text searches with some keyword, author, ID or any other text.
Genbank has a powerful <a href="http://www.ncbi.nlm.nih.gov/gquery/">query web interface</a>.</p>

<p><a href="http://www.ncbi.nlm.nih.gov/gquery/"><img src="../static/genbank_query_interface.png" alt="Genbank query" /></a></p>

<p>Each database shows the results in one or several formats.
For instance, the Genbank sequences can be obtained in several formats.</p>

<p>Genbank format:</p>

<div class="highlighter-rouge"><pre class="highlight"><code>LOCUS       EC750390                 558 bp    mRNA    linear   EST 	03-JUL-2006
DEFINITION  POE00005652 PL(light) Polytomella parva cDNA similar to frataxin protein
                  -related, mRNA sequence.
ACCESSION   EC750390
VERSION     EC750390.1  GI:110064507
KEYWORDS    EST.
SOURCE      Polytomella parva
ORGANISM    Polytomella parva
            Eukaryota; Viridiplantae; Chlorophyta; Chlorophyceae;Chlamydomonadales;
                  Chlamydomonadaceae; Polytomella.
REFERENCE   1  (bases 1 to 558)
  AUTHORS   Lee,R.W. and Borza,T.
  TITLE     The colorless plastid of the green alga Polytomella parva: a repertoire of its functions
  JOURNAL   Unpublished (2006)
COMMENT     Contact: TBestDB
            Departement de Biochimie, Universite de Montreal
            Montreal, Canada
            Email: tbestdb-curator@bch.umontreal.ca
            Plate: 4065.
FEATURES             Location/Qualifiers
     source          1..558
                     /organism="Polytomella parva"
                     /mol_type="mRNA"
                     /db_xref="taxon:51329"
                     /clone_lib="PL(light)"
ORIGIN
        1 gcggccgctt tttttttttt tttttttttt ttttcgtccg ttatttcttt tttaagaatg
       61 cagtcatctg tacatcgtca agtattcgga gtgttatctc gttttgtggg aaacaaagcg
      121 ggtattttta caaagcataa tcatggtgtc tcaaggttgt cttcatgcac ttcgtcatgc
      181 gtaaagatgt atactagcaa caaggccccc gaggatcttc aaacgttcca ccggcaagca
      241 gacgaaactc tagagcaagt cactgaagcc cttgaaaact atgtagatga gcatgaagtg
      301 gaaggcagcg acattgagca tacgcaagga gtgcttacta ttaagcttgg aactcttgga
      361 agttatgtaa ttaataaaca gactcctaat aagcagatat ggttatcctc tcccgtcagt
      421 ggacccttcc gatatgatct taaagaaggt gcctgggttt atgaacgggc tggcgaggct
      481 cggcgcgagc ttatttctca attagaaaca gaaatttcgg atttagttgg tgtcgaatta
      541 aagataagta actgaacg
</code></pre>
</div>

<p>EMBL format:</p>

<div class="highlighter-rouge"><pre class="highlight"><code>ID   EC750390; SV 1; linear; mRNA; EST; PLN; 558 BP.
XX
AC   EC750390;
XX
DT   04-JUL-2006 (Rel. 88, Created)
DT   04-JUL-2006 (Rel. 88, Last updated, Version 1)
XX
DE   POE00005652 PL(light) Polytomella parva cDNA similar to frataxin
DE   protein-related, mRNA sequence.
XX
KW   EST.
XX
OS   Polytomella parva
OC   Eukaryota; Viridiplantae; Chlorophyta; Chlorophyceae; Chlamydomonadales;
OC   Chlamydomonadaceae; Polytomella.
XX
RN   [1]
RP   1-558
RA   Lee R.W., Borza T.;
RT   "The colorless plastid of the green alga Polytomella parva: a repertoire of
RT   its functions";
RL   Unpublished.
XX
DR   UNILIB; 42732; 19932.
XX
CC   Contact: TBestDB
CC   Departement de Biochimie, Universite de Montrealhttp://es.wikipedia.org/wiki/Base_de_datos
CC   Montreal, Canada
CC   Email: tbestdb-curator@bch.umontreal.ca
CC   Plate: 4065.
XX
FH   Key             Location/Qualifiers
FH
FT   source          1..558
FT                   /organism="Polytomella parva"
FT                   /mol_type="mRNA"
FT                   /clone_lib="PL(light)"
FT                   /db_xref="taxon:51329"
FT                   /db_xref="UNILIB:42732"
XX
SQ   Sequence 558 BP; 153 A; 105 C; 127 G; 173 T; 0 other;
     gcggccgctt tttttttttt tttttttttt ttttcgtccg ttatttcttt tttaagaatg        60
     cagtcatctg tacatcgtca agtattcgga gtgttatctc gttttgtggg aaacaaagcg       120
     ggtattttta caaagcataa tcatggtgtc tcaaggttgt cttcatgcac ttcgtcatgc       180
     gtaaagatgt atactagcaa caaggccccc gaggatcttc aaacgttcca ccggcaagca       240
     gacgaaactc tagagcaagt cactgaagcc cttgaaaact atgtagatga gcatgaagtg       300
     gaaggcagcg acattgagca tacgcaagga gtgcttacta ttaagcttgg aactcttgga       360
     agttatgtaa ttaataaaca gactcctaat aagcagatat ggttatcctc tcccgtcagt       420
     ggacccttcc gatatgatct taaagaaggt gcctgggttt atgaacgggc tggcgaggct       480
     cggcgcgagc ttatttctca attagaaaca gaaatttcgg atttagttgg tgtcgaatta       540
     aagataagta actgaacg                                                     558
</code></pre>
</div>

<p>Main fields in the <a href="http://www.ncbi.nlm.nih.gov/Sitemap/samplerecord.html">Genbank format</a>.</p>

<table>
  <thead>
    <tr>
      <th>Field</th>
      <th>Description</th>
      <th>Search in Entrez</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>Locus name</td>
      <td>Unique sequence name</td>
      <td>[ACCN]</td>
    </tr>
    <tr>
      <td>Sequence length</td>
      <td>Sequence Length</td>
      <td>[SLEN]</td>
    </tr>
    <tr>
      <td>Molecule Type</td>
      <td>DNA, genomic, mRNA, etc.</td>
      <td>[PROP]</td>
    </tr>
    <tr>
      <td>Genbank Division</td>
      <td>Division for the sequence</td>
      <td>[PROP]</td>
    </tr>
    <tr>
      <td>Modification Date</td>
      <td>Date for the last edit</td>
      <td>[MDAT]</td>
    </tr>
    <tr>
      <td>Definition</td>
      <td>Brief description</td>
      <td>[TITL]</td>
    </tr>
    <tr>
      <td>Accession</td>
      <td>Unique accession ID. It does not changes with modifications</td>
      <td>[ACCN]</td>
    </tr>
    <tr>
      <td>Version</td>
      <td>Version number of the sequence</td>
      <td>All fields</td>
    </tr>
    <tr>
      <td>Keywords</td>
      <td>keywords that describe the sequence</td>
      <td>[KYWD]</td>
    </tr>
    <tr>
      <td>Source</td>
      <td>Common name for the source species</td>
      <td>[ORGN]</td>
    </tr>
    <tr>
      <td>Organism</td>
      <td>Oficial name for the source species</td>
      <td>[ORGN]</td>
    </tr>
    <tr>
      <td>Reference</td>
      <td>Related publications</td>
      <td>[TITL] [AUTH] [JOUR]</td>
    </tr>
    <tr>
      <td>Features</td>
      <td>Regions of interest</td>
      <td>[FKEY]</td>
    </tr>
    <tr>
      <td>CDS</td>
      <td>Coding Sequence</td>
      <td>[FKEY]</td>
    </tr>
  </tbody>
</table>

<p>The <a href="http://www.ncbi.nlm.nih.gov/Sitemap/samplerecord.html#AccessionA">Accession</a> is the unique identifier for a sequence record.
An accession number applies to the complete record and is usually a combination of a letter(s) and numbers, such as a single letter followed by five digits (e.g., U12345) or two letters followed by six digits (e.g., AF123456).</p>

<p>The records in GenBank can be updated by an author request, accession numbers do not change, even if information in the record is changed.
So, a sequence can have several versions in GenBank.
<a href="http://www.ncbi.nlm.nih.gov/Sitemap/samplerecord.html#VersionA">Version</a> is an unique identifier that represents a single, specific sequence in the GenBank database.
If there is any change to the sequence data (even a single base), the version number will be increased, e.g., U12345.1 → U12345.2, but the accession portion will remain stable.</p>

<p><a href="http://www.ncbi.nlm.nih.gov/Sitemap/samplerecord.html#FeaturesA">Features</a> holds information about genes and gene products, as well as regions of biological significance reported in the sequence.
These can include regions of the sequence that code for proteins and RNA molecules.</p>

<p>Features example:</p>

<div class="highlighter-rouge"><pre class="highlight"><code> FEATURES             Location/Qualifiers
 source          1..12401
                 /organism="Homo sapiens"
                 /mol_type="genomic DNA"
                 /db_xref="taxon:9606"
                 /chromosome="17"
                 /map="17q24.3"
 gene            complement(&lt;1..4774)
                 /gene="SOX9-AS1"
                 /note="SOX9 antisense RNA 1"
                 /db_xref="GeneID:400618"
                 /db_xref="HGNC:HGNC:49321"
 ncRNA           complement(&lt;4744..4774)
                 /ncRNA_class="lncRNA"
                 /gene="SOX9-AS1"
                 /product="SOX9 antisense RNA 1, transcript variant 2"
                 /inference="similar to RNA sequence (same
                 species):RefSeq:NR_103737.1"
                 /exception="annotated by transcript or proteomic data"
                 /transcript_id="NR_103737.1"
                 /db_xref="GeneID:400618"
                 /db_xref="HGNC:HGNC:49321"
 gene            5001..10401
                 /gene="SOX9"
                 /gene_synonym="CMD1; CMPD1; SRA1; SRXX2; SRXY10"
                 /note="SRY-box 9"
                 /db_xref="GeneID:6662"
                 /db_xref="HGNC:HGNC:11204"
                 /db_xref="MIM:608160"
 mRNA            join(5001..5803,6700..6953,7524..10401)
                 /gene="SOX9"
                 /gene_synonym="CMD1; CMPD1; SRA1; SRXX2; SRXY10"
                 /product="SRY-box 9"
                 /transcript_id="NM_000346.3"
                 /db_xref="GeneID:6662"
                 /db_xref="HGNC:HGNC:11204"
                 /db_xref="MIM:608160"
 exon            5001..5803
                 /gene="SOX9"
                 /gene_synonym="CMD1; CMPD1; SRA1; SRXX2; SRXY10"
                 /inference="alignment:Splign:2.1.0"
                 /number=1
 CDS             join(5373..5803,6700..6953,7524..8368)
                 /gene="SOX9"
                 /gene_synonym="CMD1; CMPD1; SRA1; SRXX2; SRXY10"
                 /note="SRY (sex-determining region Y)-box 9 protein;
                 SRY-related HMG-box, gene 9; SRY (sex determining region
                 Y)-box9"
                 /codon_start=1
                 /product="transcription factor SOX-9"
                 /protein_id="NP_000337.1"
                 /db_xref="CCDS:CCDS11689.1"
                 /db_xref="GeneID:6662"
                 /db_xref="HGNC:HGNC:11204"
                 /db_xref="MIM:608160"
                 /translation="MNLLDPFMKMTDEQEKGLSGAPSPTMSEDSAGSPCPSGSGSDTE
                 NTRPQENTFPKGEPDLKKESEEDKFPVCIREAVSQVLKGYDWTLVPMPVRVNGSSKNK
                 PHVKRPMNAFMVWAQAARRKLADQYPHLHNAELSKTLGKLWRLLNESEKRPFVEEAER
                 LRVQHKKDHPDYKYQPRRRKSVKNGQAEAEEATEQTHISPNAIFKALQADSPHSSSGM
                 SEVHSPGEHSGQSQGPPTPPTTPKTDVQPGKADLKREGRPLPEGGRQPPIDFRDVDIG
                 ELSSDVISNIETFDVNEFDQYLPPNGHPGVPATHGQVTYTGSYGISSTAATPASAGHV
                 WMSKQQAPPPPPQQPPQAPPAPQAPPQPQAAPPQQPAAPPQQPQAHTLTTLSSEPGQS
                 QRTHIKTEQLSPSHYSEQQQHSPQQIAYSPFNLPHYSPSYPPITRSQYDYTDHQNSSS
                 YYSHAAGQGTGLYSTFTYMNPAQRPMYTPIADTSGVPSIPQTHSPQHWEQPVYTQLTR
                 P"
 STS             5374..5670
                 /gene="SOX9"
                 /gene_synonym="CMD1; CMPD1; SRA1; SRXX2; SRXY10"
                 /standard_name="PMC34415P1"
                 /db_xref="UniSTS:273201"
 exon            6700..6953
                 /gene="SOX9"
                 /gene_synonym="CMD1; CMPD1; SRA1; SRXX2; SRXY10"
                 /inference="alignment:Splign:2.1.0"
                 /number=2
 STS             6765..7647
                 /gene="SOX9"
                 /gene_synonym="CMD1; CMPD1; SRA1; SRXX2; SRXY10"
                 /standard_name="PMC351321P1"
                 /db_xref="UniSTS:273242"
 STS             6771..7605
                 /gene="SOX9"
                 /gene_synonym="CMD1; CMPD1; SRA1; SRXX2; SRXY10"
                 /standard_name="MARC_71570-71571:1249593890:3"
                 /db_xref="UniSTS:528379"
 STS             6876..8104
                 /gene="SOX9"
                 /gene_synonym="CMD1; CMPD1; SRA1; SRXX2; SRXY10"
                 /standard_name="Sox9"
                 /db_xref="UniSTS:502689"
 exon            7524..10401
                 /gene="SOX9"
                 /gene_synonym="CMD1; CMPD1; SRA1; SRXX2; SRXY10"
                 /inference="alignment:Splign:2.1.0"
                 /number=3
</code></pre>
</div>

<h1 id="sequence-formats">Sequence formats</h1>

<h2 id="text-files">Text files</h2>

<p>There are different formats to store sequences in a <a href="https://en.wikipedia.org/wiki/Text_file">text file</a>.
Text files should only include <a href="https://en.wikipedia.org/wiki/Plain_text">Plain text</a>.
Graphics or any other binary information are not allowed in text files.</p>

<p>Microsoft Word files are not text files, they are binary files that happen to represent documents.
These documents can include text among many other things like images, charts or formats.</p>

<h2 id="sequences-in-plain-files">Sequences in plain files</h2>

<p>We could store the sequence in a text file by just writing the sequence.
This files would had to include only <a href="https://en.wikipedia.org/wiki/FASTA_format#Sequence_representation">IUPAC</a> characters.</p>

<div class="highlighter-rouge"><pre class="highlight"><code>ACAAGATGCCATTGTCCCCCGGCCTCCTGCTGCTGCTGCTCTCCGGGGCCACGGCCACCGCTGCCCTGCCCCTGGAGGGTAC
GGCCCCACCGGCCGAGACAGCGAGCATATGCAGGAAGCGGCAGGAATAAGGAAAAGCAGCCTCCTGACTTTCCTCGCTTGGT
AGTGGACCTCCCAGGCCAGTGCCGGGCCCCTCATAGGAGAGGAAGCTCGGGAGGTGGCCAGGCGGCAGGAAGGCGCACCCCC
ATCCGCGCGCCGGGACAGAATGCCCTGCAGGAACTTCTTCTGGAAGACCTTCTCCTCCTGCAAATAAAA
</code></pre>
</div>

<p>This kind of file is seldom used because it lacks any metadata to identify the sequence.</p>

<h2 id="fasta-format">Fasta format</h2>

<p>The <a href="https://en.wikipedia.org/wiki/FASTA_format">Fasta</a> file includes a name for the sequence and, optionally, some description.
The sequence should be preceded by a line that starts with the symbol <em>&gt;</em>.
The name will be written after that symbol.
Spaces are not allowed in the sequence name.
If there is a description it will be found after a space in the same line.</p>

<p>Several sequences can be included in the same file.</p>

<p>It is one of the most common formats.</p>

<div class="highlighter-rouge"><pre class="highlight"><code>&gt;sequence1_name description
ACAAGATGCCATTGTCCCCCGGCCTCCTGCTGCTGCTGCTCTCCGGGGCCACGGCCACCGCTGCCCTGCC
CCTGGAGGGTGGCCCCACCGGCCGAGACAGCGAGCATATGCAGGAAGCGGCAGGAATAAGGAAAAGCAGC
CTCCTGACTTTCCTCGCTTGGTGGTTTGAGTGGACCTCCCAGGCCAGTGCCGGGCCCCTCATAGGAGAGG
AAGCTCGGGAGGTGGCCAGGCGGCAGGAAGGCGCACCCCCCCAGCAATCCGCGCGCCGGGACAGAATGCC
CTGCAGGAACTTCTTCTGGAAGACCTTCTCCTCCTGCAAATAAAACCTCACCCATGAATGCTCACGC
&gt;sequence2_name description
ACAAGATGCCATTGTCCCCCGGCCTCCTGCTGCTGCTGCTCTCCGGGGCCACGGCCACCGCTGCCCTGCC
CCTGGAGGGTGGCCCCACCGGCCGAGACAGCGAGCATATGCAGGAAGCGGCAGGA
</code></pre>
</div>

<p>If we want to include more information we could use the GenBank or EMBL formats.
It is also very common in the sequences that come directly from a sequencing machine to include the quality information, for that purpose the most common format is <a href="https://en.wikipedia.org/wiki/FASTQ_format">FASTQ</a>.</p>

<h1 id="ontologies">Ontologies</h1>

<p>Wikipedia defines <a href="https://en.wikipedia.org/wiki/Ontology_%28information_science%29">ontology</a> as:</p>

<blockquote>
  <p>an ontology is a formal naming and definition of the types, properties, and interrelationships of the entities that really or fundamentally exist for a particular domain of discourse.</p>
</blockquote>

<p>An ontology is a way of structure the knowledge by dividing it in the entities relevant to a particular field.
For instance, we have been talking about sequences, so a term in our ontology could be <em>sequence</em>.</p>

<p>It is also quite common to create hierarchical ontologies.
For instance, <em>nucleotide sequence</em> and <em>protein sequence</em> could be subterms of sequence.
In an ontology the terms are precisely defined and, usually, there are no synonyms.</p>

<p>Defining the terms relevant to a field is very useful, specially if those terms are discussed and adopted by the whole community.
Standard ontologies became powerful tools that enable automatic analyses and searches.
Imagine that we want to look for all enzymes related to lipid metabolism in a database.
Which term should we use for the search? Lipid metabolism? Fat degradation? Lipid catabolism?</p>

<p>There are different biological ontologies, but the main ones are maintained by the <a href="http://geneontology.org/">Gene Ontology Consortium</a>.
The GO terms are used to define gene functions.
Each GO term has an unique ID and a definition.
There are three aspects covered by three hierarchical ontologies:</p>

<ul>
  <li><em>molecular function</em>: molecular activities of gene products</li>
  <li><em>cellular component</em>: where gene products are active</li>
  <li><em>biological process</em>: pathways and larger processes made up of the activities of multiple gene products.</li>
</ul>

<p><img src="../static/go_hierarchy.png" alt="GO hieralchy" /></p>

<p>You can browse the GO hierarchies at a <a href="http://www.informatics.jax.org/searches/GO_form.shtml">GO browser</a></p>

<p>A gene can have several GO terms associated with each ontology.
A gene can be annotated with terms from different levels of the hierarchy.</p>

<p>The GO ontologies ease the search for information and allow complex automated analyses.</p>

<h2 id="plant-related-ontologies">Plant related ontologies</h2>

<p>The <a href="http://planteome.org/">Planteome</a> project curates some plant related ontologies.</p>

<p>The <a href="http://wiki.plantontology.org/index.php/PO_Project_Overview">Plant Ontology</a> deals with Plant Anatomical Entities and Plant Developmental Stages.</p>

<p>The [Plant Trait Ontology] curates terms related to measurable traits, and the [Plant Experimental Condition] deals with experimental conditions.</p>


  </div>

</article>

      </div>
    </main>

    

  </body>

</html>
