<!DOCTYPE html>
<html lang="en">

  <head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1">

  <title>Sequence Alignment Practice</title>
  <meta name="description" content="Introduction to Bioinformatics for the Biotechnology degree.">

  <link rel="stylesheet" href="/courses/biotech3/css/main.css">
  <link rel="canonical" href="https://bioinf.comav.upv.es/courses/biotech3/practice/alignment.html">
  <link rel="alternate" type="application/rss+xml" title="Bioinformatics" href="https://bioinf.comav.upv.es/courses/biotech3/feed.xml">
</head>


  <body>

    <header class="site-header" role="banner">

  <div class="wrapper">

    <a class="site-title" href="/courses/biotech3/">Bioinformatics</a>

    <nav class="site-nav">
      <a href="#" class="menu-icon">
        <svg viewBox="0 0 18 15">
          <path fill="#424242" d="M18,1.484c0,0.82-0.665,1.484-1.484,1.484H1.484C0.665,2.969,0,2.304,0,1.484l0,0C0,0.665,0.665,0,1.484,0 h15.031C17.335,0,18,0.665,18,1.484L18,1.484z"/>
          <path fill="#424242" d="M18,7.516C18,8.335,17.335,9,16.516,9H1.484C0.665,9,0,8.335,0,7.516l0,0c0-0.82,0.665-1.484,1.484-1.484 h15.031C17.335,6.031,18,6.696,18,7.516L18,7.516z"/>
          <path fill="#424242" d="M18,13.516C18,14.335,17.335,15,16.516,15H1.484C0.665,15,0,14.335,0,13.516l0,0 c0-0.82,0.665-1.484,1.484-1.484h15.031C17.335,12.031,18,12.696,18,13.516L18,13.516z"/>
        </svg>
      </a>

      <div class="trigger">
        
          
          <a class="page-link" href="/courses/biotech3/practice/alignment.html">Sequence Alignment Practice</a>
          
        
          
          <a class="page-link" href="/courses/biotech3/practice/assembly.html">Assembly</a>
          
        
          
          <a class="page-link" href="/courses/biotech3/theory/blast.html">Blast</a>
          
        
          
          <a class="page-link" href="/courses/biotech3/practice/blast.html">Blast Practice</a>
          
        
          
          <a class="page-link" href="/courses/biotech3/covid.html">COVID19</a>
          
        
          
          <a class="page-link" href="/courses/biotech3/theory/databases.html">Biological Databases</a>
          
        
          
          <a class="page-link" href="/courses/biotech3/practice/emboss.html">Introduction to EMBOSS</a>
          
        
          
          <a class="page-link" href="/courses/biotech3/practice/entrez.html">Entrez and NCBI databases Practice</a>
          
        
          
        
          
          <a class="page-link" href="/courses/biotech3/">Home</a>
          
        
          
          <a class="page-link" href="/courses/biotech3/practice/linux.html">Unix CLI</a>
          
        
          
        
          
          <a class="page-link" href="/courses/biotech3/practice/multiple.html">Multiple Sequence Alignment Practice</a>
          
        
          
          <a class="page-link" href="/courses/biotech3/theory/multiple.html">Multiple Sequence Alignment</a>
          
        
          
          <a class="page-link" href="/courses/biotech3/norms.html">Norms</a>
          
        
          
          <a class="page-link" href="/courses/biotech3/practice/phylogeny.html">Phylogeny practice</a>
          
        
          
          <a class="page-link" href="/courses/biotech3/theory/phylogeny.html">Phylogeny and evolution</a>
          
        
          
          <a class="page-link" href="/courses/biotech3/prev_knowledge.html">Previous Knowledge</a>
          
        
          
          <a class="page-link" href="/courses/biotech3/theory/sequence_alignment.html">Sequence Alignment</a>
          
        
          
          <a class="page-link" href="/courses/biotech3/theory/structural_annotation.html">Structural Annotation</a>
          
        
          
          <a class="page-link" href="/courses/biotech3/practice/structural_annotation.html">Structural Annotation</a>
          
        
      </div>
    </nav>

  </div>

</header>


    <main class="page-content" aria-label="Content">
      <div class="wrapper">
        <article class="post">

  <header class="post-header">
    <h1 class="post-title">Sequence Alignment Practice</h1>
  </header>

  <div class="post-content">
    <h1 id="sequence-alignment-practice">Sequence Alignment Practice</h1>

<p>Which EMBOSS programs can we use to do sequence alignments?</p>

<div class="highlighter-rouge"><pre class="highlight"><code>$ wossname alignment
</code></pre>
</div>

<p>Which can we use to do dotplots?
And local and global alignments?</p>

<p>Which program is specific to align mRNA?</p>

<h1 id="dotplot">Dotplot</h1>

<h2 id="gene-structure">Gene structure</h2>

<p>We can study the exon and intron gene structure graphically by using the graphical alignment program <a href="http://emboss.sourceforge.net/apps/release/6.3/emboss/apps/dotmatcher.html">dotmatcher</a>.
dotmatcher can create dotplots in EMBOSS.</p>

<p>Let’s compare the Arabidopsis <a href="../static/alignments/AT3G4905.1.genomico.fasta">genomic region AT3G52905.1</a> with one <a href="../static/alignments/AT3G4905.1.cdna.fasta">cDNA</a>.</p>

<div class="highlighter-rouge"><pre class="highlight"><code>$ dotmatcher AT3G4905.1.genomico.fasta AT3G4905.1.cdna.fasta
</code></pre>
</div>

<p>Why is there so much noise in the result?</p>

<p>Is there any parameter that we can use to improve the result?</p>

<div class="highlighter-rouge"><pre class="highlight"><code>$ dotmatcher -windowsize 15 -threshold 23 AT3G4905.1.genomico.fasta AT3G4905.1.cdna.fasta
</code></pre>
</div>

<p>Or even more restrictive</p>

<div class="highlighter-rouge"><pre class="highlight"><code>$ dotmatcher -windowsize 15 -threshold 35 AT3G4905.1.genomico.fasta AT3G4905.1.cdna.fasta
</code></pre>
</div>

<p>Why is the main diagonal split in several segments?</p>

<h2 id="genes-in-close-species">Genes in close species</h2>

<p>We can study homologous regions between two proteins, one from human and another from dog.
Using dotmatcher we will compare the CDS the <a href="../static/alignments/CDKL4.human.fasta">human</a> and <a href="../static/alignments/cyclin.dog.fasta">dog</a> genes.</p>

<ul>
  <li>
    <p>Are the genes similar along all the CDS?</p>
  </li>
  <li>
    <p>Why some regions could be more similar than others?</p>
  </li>
</ul>

<h2 id="tandem-repeats">Tandem repeats</h2>

<p>Graphical alignments are particularly well suited to study repeated or inverted regions.
As an example we will study the repeats found in the human zinc finger <a href="../static/alignments/Q9P255.fasta">Q9P255</a></p>

<p>Align the protein against itself using dotmatcher.</p>

<ul>
  <li>
    <p>How do you interpret the resulting pattern?</p>
  </li>
  <li>
    <p>How are the repeats distributed? How many repeats are there?</p>
  </li>
  <li>
    <p>Change the threshold value and study how it affects the graphical alignment.</p>
  </li>
  <li>
    <p>Look for the domain structure of the <a href="http://www.uniprot.org/uniprot/Q9P255">Q9P255</a> in Uniprot.</p>
  </li>
</ul>

<h2 id="genomic-level-dotplots">Genomic level dotplots</h2>

<p>Dotplots can be used to detect long genomic <a href="https://en.wikipedia.org/wiki/Structural_variation">structural variants</a>.
There are several software tools to do it, for instance the web service <a href="https://genomevolution.org/coge/SynMap.pl">SynMap</a>.</p>

<p>Running these analyses is slow, so we have already run them and you have here the results.</p>

<p>Alignment the complete genome of two <em>E. coli</em> K12 substrains DH10B and VSW3110</p>

<p><img src="../static/alignments/synmap_ecoli_k12.dh10b_vsw3110.png" alt="Ecoli K12 DH10B vs VSW3110" /></p>

<ul>
  <li>
    <p>Which is the most striking difference?</p>
  </li>
  <li>
    <p>Are there any other differences?</p>
  </li>
  <li>
    <p>How can you interpret the dots outside the main diagonals?</p>
  </li>
</ul>

<p>Ploplar genome aligned against itself by comparing just the CDSs.
The black vertial and horizontal lines divide the chromosomes.</p>

<p><img src="../static/alignments/synmap_poplar_vs_itself.png" alt="Ploplar genome vs itself" /></p>

<ul>
  <li>How do you interpret the result?</li>
</ul>

<p>A graphical alignment of the maize vs sorgum genomes.
Vertical and horizontal lines divide the chromosomes.</p>

<p><img src="../static/alignments/synmap_maize_vs_sorgum.png" alt="Maize vs Sorgum" /></p>

<ul>
  <li>How do you interpret the result?</li>
</ul>

<h2 id="virus-repeats-dotplot">Virus repeats dotplot</h2>

<p>An endogenous retrovirus 12 kb long and has an inverted repeat of 250 pb located at both ends.
Draw a dotplot comparing its sequence against itself.</p>

<h2 id="interpret-the-result-of-the-dotplot-between-the-sequences-a-and-b">Interpret the result of the dotplot between the sequences A and B</h2>

<p><img src="../static/alignments/dotplot_reversed.png" alt="dotplot" /></p>

<h2 id="pcr-primer-dotplot">PCR primer dotplot</h2>

<p>You have amplified a 200 pb PCR fragment using two primers (18 and 20 pb). Draw the dotplot between the sequences of the fragment and the primers</p>

<h2 id="gene-structure-dotplot">Gene structure dotplot</h2>

<p><img src="../static/alignments/gene_structure.png" alt="dotplot" /></p>

<ul>
  <li>
    <p>Draw a dotplot between the genomic sequence and a full length cDNA of this gene.</p>
  </li>
  <li>
    <p>Draw a dotplot between the  genomic sequence and the CDS of this gene</p>
  </li>
</ul>

<h1 id="local-alignments">Local alignments</h1>

<h2 id="dna-sequence-alignment">DNA sequence alignment</h2>

<p>We are going to do a local alignment between two DNA sequences: <a href="../static/alignments/sequence1.fasta">sequence1</a> and <a href="../static/alignments/sequence2.fasta">sequence2</a>.</p>

<p>In the command line we would write:</p>

<div class="highlighter-rouge"><pre class="highlight"><code>$ matcher sequence1.fasta sequence2.fasta stdout
</code></pre>
</div>

<ul>
  <li>
    <p>How similar are the sequences?</p>
  </li>
  <li>
    <p>Are they similar in all its length?</p>
  </li>
</ul>

<p>Let’s align the reverse and complementary sequence:</p>

<div class="highlighter-rouge"><pre class="highlight"><code>$ matcher -sreverse1 sequence1.fasta sequence2.fasta stdout
</code></pre>
</div>

<p>How similar are the sequences?
Why is different from the previous attempt?</p>

<p>What would happen if we were to run the following command (Try to answer it before executing the command)?</p>

<div class="highlighter-rouge"><pre class="highlight"><code>$ matcher -sreverse1 -sreverse2 sequence1.fasta sequence2.fasta stdout
</code></pre>
</div>

<h2 id="homologous-proteins">Homologous proteins</h2>

<p>We can create the best alignment between two homologous proteins from two different species, once we have decided a score system.
We also could use these methods to align homologous proteins within a species.</p>

<p>As an example we will align a <a href="../static/alignments/CDKL4.human.fasta">human</a> and <a href="../static/alignments/cyclin.dog.fasta">canine</a> cyclin using <a href="http://emboss.sourceforge.net/apps/release/6.3/emboss/apps/matcher.html">matcher</a> or <a href="http://emboss.sourceforge.net/apps/release/6.3/emboss/apps/water.html">water</a>.</p>

<div class="highlighter-rouge"><pre class="highlight"><code>$ matcher CDKL4.human.fasta cyclin.dog.fasta stdout
</code></pre>
</div>

<p>How similar are the sequences?</p>

<p>Is this alignment local or global? Why?</p>

<h2 id="conserved-regions">Conserved regions</h2>

<p>Usually not all regions of two homologous proteins vary in the same degree.
There are regions that are more conversed and regions that are less conserved.</p>

<p>We can look for the more conserved regions by doing a local alignment.</p>

<p>Align the <a href="../static/alignments/frataxin.human.gb">human</a> and <a href="../static/alignments/frataxin.aedes.gb">Aedes mosquito</a> frataxins using  <a href="http://emboss.sourceforge.net/apps/release/6.3/emboss/apps/matcher.html">matcher</a> or <a href="http://emboss.sourceforge.net/apps/release/6.3/emboss/apps/water.html">water</a>.</p>

<div class="highlighter-rouge"><pre class="highlight"><code>$ matcher frataxin.human.gb frataxin.aedes.gb stdout
</code></pre>
</div>

<ul>
  <li>
    <p>Is the alignment local?</p>
  </li>
  <li>
    <p>Why are there regions that has not been aligned?</p>
  </li>
</ul>

<p>Why some regions are more similar than others?</p>

<ul>
  <li>Is there anything special with aminoacids 92, 100, 101, 104, 108, 111, 112, 122 and 124 in the human frataxin?</li>
</ul>

<p>Open in a text editor the genbank files for the human and mosquito proteins and check the annotation of those aminoacids.</p>

<p>Let’s try to modify the parameters given to the aligner.
Let’s do the alignment with the parameter gapopen set to 4.</p>

<div class="highlighter-rouge"><pre class="highlight"><code>$ matcher frataxin.human.gb frataxin.aedes.gb stdout -gapopen 4
</code></pre>
</div>

<ul>
  <li>Is this alignment better or worse?</li>
</ul>

<p>Every score set will produce more than one alignment, but matcher will only show, by default, only one alignment.
Ask matcher to show 3 alternative alignments</p>

<div class="highlighter-rouge"><pre class="highlight"><code>$ matcher frataxin.human.gb frataxin.aedes.gb stdout -alternatives 3
</code></pre>
</div>

<p>Why the first alignment is the best one?</p>

<h1 id="gobal-alignments">Gobal alignments</h1>

<p>We could use an algorithm that aligns both sequences in its whole length.</p>

<p>Redo the alignments from the last two exercises using <a href="http://emboss.sourceforge.net/apps/release/6.3/emboss/apps/needle.html">needle</a>.</p>

<div class="highlighter-rouge"><pre class="highlight"><code>$ needle frataxin.human.gb frataxin.aedes.gb stdout -auto
</code></pre>
</div>

<ul>
  <li>Is this alignment better or worse than the previous one?</li>
</ul>

<p>The global alignments should only be used when we are sure that the two sequences are similar in all its length.
These algorithms will give a very bad result if there are regions that are not similar between the sequences.</p>

<h1 id="genic-structure-1">Genic structure 1</h1>

<p>We have aligned a cDNA and the  genomic sequence of a gene with matcher. Do you think that it is a good approach? Could you think on any problems regarding the approach?</p>

<div class="highlighter-rouge"><pre class="highlight"><code>             560       570       580       590       600 
cDNA TTCAGGAGTATTTAGATTACGCGAACAGGTATGTAAACACTGAGCCAGCA
     : ::::::::::::::::::::::::::::::::::::::::::::::::
GENO TGCAGGAGTATTTAGATTACGCGAACAGGTATGTAAACACTGAGCCAGCA
     1550      1560      1570      1580      1590        

             610       620       630       640       650 
cDNA GAGTAAAAGGAAACGGGTAATGACAAAATAGGGAAGCCGCTAGTTTCGGA
     ::::::::::::::::::::::::::::::::::::::::::::::::::
GENO GAGTAAAAGGAAACGGGTAATGACAAAATAGGGAAGCCGCTAGTTTCGGA
     1600      1610      1620      1630      1640        

             660       670       680       690       700 
cDNA TATATGAATATAATGATGCTTGATTTGTTTTGACATTATCTATGTGTACA
     ::::::::::::::::::::::::::::::::::::::::::::::::::
GENO TATATGAATATAATGATGCTTGATTTGTTTTGACATTATCTATGTGTACA
     1650      1660      1670      1680      1690        

             710       720       730       740 
cDNA AAGACATCTTGCATCTTTTAAGAAATTCGTTTTTCGCTGC
     ::::::::::::::::::::::::::::::::::::::::
GENO AAGACATCTTGCATCTTTTAAGAAATTCGTTTTTCGCTGC
     1700      1710      1720      1730        

</code></pre>
</div>

<h1 id="genic-structure-2">Genic structure 2</h1>

<p>Determine the genic structure of the AT3G4905 <em>Arabidopsis</em> gene.
You need to dermine where a the exons and introns located and which are the conding sequences.
You have part of the <a href="../static/alignments/AT3G4905.1.genomico.fasta">genomic</a> sequence, a <a href="../static/alignments/AT3G4905.1.cdna.fasta">cDNA</a> and the <a href="../static/alignments/AT3G4905.1.cds.fasta">CDS</a> for that cDNA.</p>

<p>EMBOSS has a program created just with the aim of alignming messenger RNA with a genomic sequences.
These kind of alingments are special:</p>

<ul>
  <li>They are very similar in the regions share by the genome and the mRNA</li>
  <li>They have long gaps (the introns)</li>
  <li>The exons should be surronded by some specific sequence <a href="https://en.wikipedia.org/wiki/RNA_splicing#Spliceosomal">intronic sequences</a>.</li>
</ul>

<p>Let’s align the genomic sequence of an <em>Arabidopsis</em> gene <a href="../static/alignments/AT3G4905.1.genomico.fasta">AT3G4905</a> with a <a href="../static/alignments/AT3G4905.1.cdna.fasta">cDNA</a>.</p>

<div class="highlighter-rouge"><pre class="highlight"><code>$ est2genome AT3G4905.1.cdna.fasta AT3G4905.1.genomico.fasta stdout -auto
</code></pre>
</div>

<ul>
  <li>How many introns are there?</li>
</ul>

<p>To get the alignment report with the complete sequence:</p>

<div class="highlighter-rouge"><pre class="highlight"><code>$ est2genome AT3G4905.1.cdna.fasta AT3G4905.1.genomico.fasta stdout -align -auto
</code></pre>
</div>

<ul>
  <li>
    <p>Would needle or matcher create the same alignment?</p>
  </li>
  <li>
    <p>Compare the sequence of the <a href="../static/alignments/AT3G4905.1.cds.fasta">CDS</a> of this cDNA with the sequence of the genome.</p>
  </li>
  <li>
    <p>Is the alignment similar or different to previous one?. Why?</p>
  </li>
  <li>
    <p>Do the alignment between CDS and mRNA. Which conclusions could you reach?</p>
  </li>
</ul>

<h1 id="protein-alignment">Protein alignment</h1>

<p>The alignment between the SOX9 and NHE-RF2 proteins gave this result. could you infer that this proteins are homologues? Is this region a conserved domain? Check your answers in the databases.</p>

<div class="highlighter-rouge"><pre class="highlight"><code>
#=======================================
#
# Aligned_sequences: 2
# 1: AHA92033.1
# 2: XP_016879383.1
# Matrix: EBLOSUM62
# Gap_penalty: 14
# Extend_penalty: 4
#
# Length: 33
# Identity:      12/33 (36.4%)
# Similarity:    13/33 (39.4%)
# Gaps:           0/33 ( 0.0%)
# Score: 49
# 
#
#=======================================

           330       340       350      
AHA920 TAATPAGAGHVWMSKQQAPPPPPPPQQPPPAPP
       : ::      :      : ::   :  :: .::
XP_016 TPATCCHCAAVMARSGSATPPARAPGAPPRSPP
      150       160       170       180 

</code></pre>
</div>

<h1 id="bl2seq">bl2seq</h1>

<p>An often used program to align two sequences is bl2seq.
We are going to use a web service to align with <a href="http://blast.ncbi.nlm.nih.gov/Blast.cgi?PAGE_TYPE=BlastSearch&amp;PROG_DEF=blastn&amp;BLAST_PROG_DEF=megaBlast&amp;BLAST_SPEC=blast2seq">bl2seq</a>.</p>

<p>Blast uses a different algorithm to the software used before in this practice.
This algorithm is faster, but it is not guaranteed that we will get the best alignment.</p>

<p>Bl2seq is a good alignment method when the sequences are large and the other methods won’t produce a solution in a reasonable time.
Blast is used more often to compare one sequence with a whole database of sequences.</p>

<p>Compare the blast result with the previous <a href="../static/alignments/AT3G4905.1.genomico.fasta">genomic</a> vs <a href="../static/alignments/AT3G4905.1.cdna.fasta">cDNA</a> alignment.</p>

<ul>
  <li>How many alignments have we obtained?</li>
</ul>

<h1 id="matcher-vs-bl2seq">matcher vs bl2seq</h1>

<p>We have aligned using both matcher and bl2seq a sequence against itself. Could you suggest an interpretacion of the results?</p>

<div class="highlighter-rouge"><pre class="highlight"><code>
########################################
# Program: matcher
# Rundate: Mon 16 Oct 2017 14:57:26
# Commandline: matcher
#    -auto
#    -asequence /var/lib/emboss-explorer/output/783917/.asequence
#    -bsequence /var/lib/emboss-explorer/output/783917/.bsequence
#    -alternatives 1
#    -outfile outfile
#    -aformat3 markx0
# Align_format: markx0
# Report_file: outfile
########################################

#=======================================
#
# Aligned_sequences: 2
# 1: seq_bi
# 2: seq_bi
# Matrix: EDNAFULL
# Gap_penalty: 16
# Extend_penalty: 4
#
# Length: 250
# Identity:     250/250 (100.0%)
# Similarity:   250/250 (100.0%)
# Gaps:           0/250 ( 0.0%)
# Score: 1250
# 
#
#=======================================

               10        20        30        40        50
seq_bi TCGCGAAGACATTTAGCAGAACCATTTGTCGTTTCCTCTCGCGATCAGGC
       ::::::::::::::::::::::::::::::::::::::::::::::::::
seq_bi TCGCGAAGACATTTAGCAGAACCATTTGTCGTTTCCTCTCGCGATCAGGC
               10        20        30        40        50

               60        70        80        90       100
seq_bi CAGATTCCGTGCTCCGGAGCCAGAAAGTTTCCGCCTTTAAAGAAATGAAG
       ::::::::::::::::::::::::::::::::::::::::::::::::::
seq_bi CAGATTCCGTGCTCCGGAGCCAGAAAGTTTCCGCCTTTAAAGAAATGAAG
               60        70        80        90       100

              110       120       130       140       150
seq_bi TAGCTATCTCAGATCCTTCAAATATGGTTGCTTCTCCATTGAGTGTTTTG
       ::::::::::::::::::::::::::::::::::::::::::::::::::
seq_bi TAGCTATCTCAGATCCTTCAAATATGGTTGCTTCTCCATTGAGTGTTTTG
              110       120       130       140       150

              160       170       180       190       200
seq_bi CAGATTCCGTGCTCCGGAGCCAGAAAGTTTCCGCCTTTAAAGAAATGAAG
       ::::::::::::::::::::::::::::::::::::::::::::::::::
seq_bi CAGATTCCGTGCTCCGGAGCCAGAAAGTTTCCGCCTTTAAAGAAATGAAG
              160       170       180       190       200

              210       220       230       240       250
seq_bi CTCAGAAAGAAATCAAACATTGACCTGATGGCTACAGATTTCCAGAACCT
       ::::::::::::::::::::::::::::::::::::::::::::::::::
seq_bi CTCAGAAAGAAATCAAACATTGACCTGATGGCTACAGATTTCCAGAACCT
              210       220       230       240       250

</code></pre>
</div>

<div class="highlighter-rouge"><pre class="highlight"><code>
BLASTN 2.7.0+
Reference: Zheng Zhang, Scott Schwartz, Lukas Wagner, and
Webb Miller (2000), "A greedy algorithm for aligning DNA
sequences", J Comput Biol 2000; 7(1-2):203-14.


RID: Y9B74S2U114


Query= seq_bi

Length=250


                                                                   Score     E
Sequences producing significant alignments:                       (Bits)  Value

Query_163477  seq_bi                                               462     3e-135

ALIGNMENTS
&gt;Query_163477 seq_bi
Length=250

 Score = 462 bits (250),  Expect = 3e-135
 Identities = 250/250 (100%), Gaps = 0/250 (0%)
 Strand=Plus/Plus

Query  1    TCGCGAAGACATTTAGCAGAACCATTTGTCGTTTCCTCTCGCGATCAGGCCAGATTCCGT  60
            ||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||
Sbjct  1    TCGCGAAGACATTTAGCAGAACCATTTGTCGTTTCCTCTCGCGATCAGGCCAGATTCCGT  60

Query  61   GCTCCGGAGCCAGAAAGTTTCCGCCTTTAAAGAAATGAAGTAGCTATCTCAGATCCTTCA  120
            ||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||
Sbjct  61   GCTCCGGAGCCAGAAAGTTTCCGCCTTTAAAGAAATGAAGTAGCTATCTCAGATCCTTCA  120

Query  121  AATATGGTTGCTTCTCCATTGAGTGTTTTGCAGATTCCGTGCTCCGGAGCCAGAAAGTTT  180
            ||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||
Sbjct  121  AATATGGTTGCTTCTCCATTGAGTGTTTTGCAGATTCCGTGCTCCGGAGCCAGAAAGTTT  180

Query  181  CCGCCTTTAAAGAAATGAAGCTCAGAAAGAAATCAAACATTGACCTGATGGCTACAGATT  240
            ||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||
Sbjct  181  CCGCCTTTAAAGAAATGAAGCTCAGAAAGAAATCAAACATTGACCTGATGGCTACAGATT  240

Query  241  TCCAGAACCT  250
            ||||||||||
Sbjct  241  TCCAGAACCT  250


 Score = 93.5 bits (50),  Expect = 4e-24
 Identities = 50/50 (100%), Gaps = 0/50 (0%)
 Strand=Plus/Plus

Query  151  CAGATTCCGTGCTCCGGAGCCAGAAAGTTTCCGCCTTTAAAGAAATGAAG  200
            ||||||||||||||||||||||||||||||||||||||||||||||||||
Sbjct  51   CAGATTCCGTGCTCCGGAGCCAGAAAGTTTCCGCCTTTAAAGAAATGAAG  100


 Score = 93.5 bits (50),  Expect = 4e-24
 Identities = 50/50 (100%), Gaps = 0/50 (0%)
 Strand=Plus/Plus

Query  51   CAGATTCCGTGCTCCGGAGCCAGAAAGTTTCCGCCTTTAAAGAAATGAAG  100
            ||||||||||||||||||||||||||||||||||||||||||||||||||
Sbjct  151  CAGATTCCGTGCTCCGGAGCCAGAAAGTTTCCGCCTTTAAAGAAATGAAG  200

</code></pre>
</div>

<h1 id="primer-search">Primer search</h1>

<p>We have developed two PCR primers: <a href="../static/alignments/primerf.fasta">primerf</a> and <a href="../static/alignments/primerr.fasta">primerr</a> to amplify the <a href="../static/alignments/CDKL4.human.fasta">cyclin</a> CDS.</p>

<p>Can you locate the primers in the cyclin sequence?</p>

  </div>

</article>

      </div>
    </main>

    

  </body>

</html>
