<!DOCTYPE html>

<html lang="en" data-content_root="../">
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="viewport" content="width=device-width, initial-scale=1" />

    <title>Sequence Alignment &#8212; Bioinformatics at COMAV 0.1 documentation</title>
    <link rel="stylesheet" type="text/css" href="../_static/pygments.css?v=fa44fd50" />
    <link rel="stylesheet" type="text/css" href="../_static/alabaster.css?v=039e1c02" />
    <script src="../_static/documentation_options.js?v=2709fde1"></script>
    <script src="../_static/doctools.js?v=888ff710"></script>
    <script src="../_static/sphinx_highlight.js?v=dc90522c"></script>
    <link rel="icon" href="../_static/favicon.ico"/>
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Blast Practice" href="blast.html" />
    <link rel="prev" title="Introduction to EMBOSS" href="emboss.html" />
   
  <link rel="stylesheet" href="../_static/custom.css" type="text/css" />
  
  
  <meta name="viewport" content="width=device-width, initial-scale=0.9, maximum-scale=0.9" />

  </head><body>
  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          

          <div class="body" role="main">
            
  <section id="sequence-alignment">
<h1>Sequence Alignment<a class="headerlink" href="#sequence-alignment" title="Link to this heading">¶</a></h1>
<section id="sequence-alignment-practice">
<h2>Sequence Alignment Practice<a class="headerlink" href="#sequence-alignment-practice" title="Link to this heading">¶</a></h2>
<p>Which EMBOSS programs can we use to do sequence alignments?</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ wossname alignment
</pre></div>
</div>
<p>Which can we use to do dotplots?
And local and global alignments?</p>
<p>Which program is specific to align mRNA?</p>
</section>
<section id="dotplot">
<h2>Dotplot<a class="headerlink" href="#dotplot" title="Link to this heading">¶</a></h2>
<section id="gene-structure">
<h3>Gene structure<a class="headerlink" href="#gene-structure" title="Link to this heading">¶</a></h3>
<p>We can study the exon and intron gene structure graphically by using the graphical alignment program <a class="reference external" href="http://emboss.sourceforge.net/apps/release/6.3/emboss/apps/dotmatcher.html">dotmatcher</a>.
dotmatcher can create dotplots in EMBOSS.</p>
<p>Let’s compare the Arabidopsis <a class="reference download internal" download="" href="../_downloads/7f470d3314118d610ad8e7220f183715/AT3G4905.1.genomico.fasta"><code class="xref download docutils literal notranslate"><span class="pre">genomic</span> <span class="pre">region</span> <span class="pre">AT3G52905.1</span></code></a> with one <a class="reference download internal" download="" href="../_downloads/1b09e08e6c6ee7cd9cf575a950388591/AT3G4905.1.cdna.fasta"><code class="xref download docutils literal notranslate"><span class="pre">cDNA</span></code></a>.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ dotmatcher AT3G4905.1.genomico.fasta AT3G4905.1.cdna.fasta
</pre></div>
</div>
<p>Why is there so much noise in the result?</p>
<p>Is there any parameter that we can use to improve the result?</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ dotmatcher -windowsize 15 -threshold 23 AT3G4905.1.genomico.fasta AT3G4905.1.cdna.fasta
</pre></div>
</div>
<p>Or even more restrictive</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ dotmatcher -windowsize 15 -threshold 35 AT3G4905.1.genomico.fasta AT3G4905.1.cdna.fasta
</pre></div>
</div>
<p>Why is the main diagonal split in several segments?</p>
</section>
<section id="genes-in-close-species">
<h3>Genes in close species<a class="headerlink" href="#genes-in-close-species" title="Link to this heading">¶</a></h3>
<p>We can study homologous regions between two proteins, one from human and another from dog.
Using dotmatcher we will compare the CDS the <a class="reference download internal" download="" href="../_downloads/099955f3cecea2f5195f1ec9d1766cfb/CDKL4.human.fasta"><code class="xref download docutils literal notranslate"><span class="pre">human</span></code></a> and <a class="reference download internal" download="" href="../_downloads/8e30a19254c18a3dd039a4cbb19ab02c/cyclin.dog.fasta"><code class="xref download docutils literal notranslate"><span class="pre">dog</span></code></a> genes.</p>
<ul class="simple">
<li><p>Are the genes similar along all the CDS?</p></li>
<li><p>Why some regions could be more similar than others?</p></li>
</ul>
</section>
<section id="tandem-repeats">
<h3>Tandem repeats<a class="headerlink" href="#tandem-repeats" title="Link to this heading">¶</a></h3>
<p>Graphical alignments are particularly well suited to study repeated or inverted regions.
As an example we will study the repeats found in the human zinc finger <a class="reference download internal" download="" href="../_downloads/f722ff911adc4a2ad75e076119db36a5/Q9P255.fasta"><code class="xref download docutils literal notranslate"><span class="pre">Q9P255</span></code></a></p>
<p>Align the protein against itself using dotmatcher.</p>
<ul class="simple">
<li><p>How do you interpret the resulting pattern?</p></li>
<li><p>How are the repeats distributed? How many repeats are there?</p></li>
<li><p>Change the threshold value and study how it affects the graphical alignment.</p></li>
<li><p>Look for the domain structure of the <a class="reference external" href="http://www.uniprot.org/uniprot/Q9P255">Q9P255</a> in Uniprot.</p></li>
</ul>
</section>
<section id="genomic-level-dotplots">
<h3>Genomic level dotplots<a class="headerlink" href="#genomic-level-dotplots" title="Link to this heading">¶</a></h3>
<p>Dotplots can be used to detect long genomic <a class="reference external" href="https://en.wikipedia.org/wiki/Structural_variation">structural variants</a>.
There are several software tools to do it, for instance the web service <a class="reference external" href="https://genomevolution.org/coge/SynMap.pl">SynMap</a>.</p>
<p>Running these analyses is slow, so we have already run them and you have here the results.</p>
<p>Alignment the complete genome of two <em>E. coli</em> K12 substrains DH10B and VSW3110</p>
<a class="reference external image-reference" href="../static/alignments/synmap_ecoli_k12.dh10b_vsw3110.png"><img alt="Ecoli K12 DH10B vs VSW3110" src="../_images/synmap_ecoli_k12.dh10b_vsw3110.png" /></a>
<ul class="simple">
<li><p>Which is the most striking difference?</p></li>
<li><p>Are there any other differences?</p></li>
<li><p>How can you interpret the dots outside the main diagonals?</p></li>
</ul>
<p>Ploplar genome aligned against itself by comparing just the CDSs.
The black vertial and horizontal lines divide the chromosomes.</p>
<a class="reference external image-reference" href="../static/alignments/synmap_poplar_vs_itself.png"><img alt="Ploplar genome vs itself" src="../_images/synmap_poplar_vs_itself.png" /></a>
<ul class="simple">
<li><p>How do you interpret the result?</p></li>
</ul>
<p>A graphical alignment of the maize vs sorgum genomes.
Vertical and horizontal lines divide the chromosomes.</p>
<a class="reference external image-reference" href="../static/alignments/synmap_maize_vs_sorgum.png"><img alt="Maize vs Sorgum" src="../_images/synmap_maize_vs_sorgum.png" /></a>
<ul class="simple">
<li><p>How do you interpret the result?</p></li>
</ul>
</section>
<section id="virus-repeats-dotplot">
<h3>Virus repeats dotplot<a class="headerlink" href="#virus-repeats-dotplot" title="Link to this heading">¶</a></h3>
<p>An endogenous retrovirus 12 kb long and has an inverted repeat of 250 pb located at both ends.
Draw a dotplot comparing its sequence against itself.</p>
</section>
<section id="interpret-the-result-of-the-dotplot-between-the-sequences-a-and-b">
<h3>Interpret the result of the dotplot between the sequences A and B<a class="headerlink" href="#interpret-the-result-of-the-dotplot-between-the-sequences-a-and-b" title="Link to this heading">¶</a></h3>
<a class="reference external image-reference" href="../static/alignments/dotplot_reversed.png"><img alt="dotplot" src="../_images/dotplot_reversed.png" /></a>
</section>
<section id="pcr-primer-dotplot">
<h3>PCR primer dotplot<a class="headerlink" href="#pcr-primer-dotplot" title="Link to this heading">¶</a></h3>
<p>You have amplified a 200 pb PCR fragment using two primers (18 and 20 pb). Draw the dotplot between the sequences of the fragment and the primers</p>
</section>
<section id="gene-structure-dotplot">
<h3>Gene structure dotplot<a class="headerlink" href="#gene-structure-dotplot" title="Link to this heading">¶</a></h3>
<a class="reference external image-reference" href="../static/alignments/gene_structure.png"><img alt="dotplot" src="../_images/gene_structure.png" /></a>
<ul class="simple">
<li><p>Draw a dotplot between the genomic sequence and a full length cDNA of this gene.</p></li>
<li><p>Draw a dotplot between the  genomic sequence and the CDS of this gene</p></li>
</ul>
</section>
</section>
<section id="local-alignments">
<h2>Local alignments<a class="headerlink" href="#local-alignments" title="Link to this heading">¶</a></h2>
<section id="dna-sequence-alignment">
<h3>DNA sequence alignment<a class="headerlink" href="#dna-sequence-alignment" title="Link to this heading">¶</a></h3>
<p>We are going to do a local alignment between two DNA sequences: <a class="reference download internal" download="" href="../_downloads/f38457255433c3b5ead4cdbece176c80/sequence1.fasta"><code class="xref download docutils literal notranslate"><span class="pre">sequence1</span></code></a> and <a class="reference download internal" download="" href="../_downloads/c93471efc76d4ca85b669e5501850bed/sequence2.fasta"><code class="xref download docutils literal notranslate"><span class="pre">sequence2</span></code></a>.</p>
<p>In the command line we would write:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ matcher sequence1.fasta sequence2.fasta stdout
</pre></div>
</div>
<ul class="simple">
<li><p>How similar are the sequences?</p></li>
<li><p>Are they similar in all its length?</p></li>
</ul>
<p>Let’s align the reverse and complementary sequence:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ matcher -sreverse1 sequence1.fasta sequence2.fasta stdout
</pre></div>
</div>
<p>How similar are the sequences?
Why is different from the previous attempt?</p>
<p>What would happen if we were to run the following command (Try to answer it before executing the command)?</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ matcher -sreverse1 -sreverse2 sequence1.fasta sequence2.fasta stdout
</pre></div>
</div>
</section>
<section id="homologous-proteins">
<h3>Homologous proteins<a class="headerlink" href="#homologous-proteins" title="Link to this heading">¶</a></h3>
<p>We can create the best alignment between two homologous proteins from two different species, once we have decided a score system.
We also could use these methods to align homologous proteins within a species.</p>
<p>As an example we will align a <a class="reference download internal" download="" href="../_downloads/099955f3cecea2f5195f1ec9d1766cfb/CDKL4.human.fasta"><code class="xref download docutils literal notranslate"><span class="pre">human</span></code></a> and <a class="reference download internal" download="" href="../_downloads/8e30a19254c18a3dd039a4cbb19ab02c/cyclin.dog.fasta"><code class="xref download docutils literal notranslate"><span class="pre">canine</span></code></a> cyclin using <a class="reference external" href="http://emboss.sourceforge.net/apps/release/6.3/emboss/apps/matcher.html">matcher</a> or <cite>water &lt;http://emboss.sourceforge.net/apps/release/6.3/emboss/apps/water.html&gt;</cite>.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ matcher CDKL4.human.fasta cyclin.dog.fasta stdout
</pre></div>
</div>
<p>How similar are the sequences?</p>
<p>Is this alignment local or global? Why?</p>
</section>
<section id="conserved-regions">
<h3>Conserved regions<a class="headerlink" href="#conserved-regions" title="Link to this heading">¶</a></h3>
<p>Usually not all regions of two homologous proteins vary in the same degree.
There are regions that are more conversed and regions that are less conserved.</p>
<p>We can look for the more conserved regions by doing a local alignment.</p>
<p>Align the <a class="reference download internal" download="" href="../_downloads/c2e644862730299d4e27f550a6130394/frataxin.human.gb"><code class="xref download docutils literal notranslate"><span class="pre">human</span></code></a> and <a class="reference download internal" download="" href="../_downloads/7b71c5f333968fe2b04795ec34981b94/frataxin.aedes.gb"><code class="xref download docutils literal notranslate"><span class="pre">Aedes</span> <span class="pre">mosquito</span></code></a> frataxins using  <a class="reference external" href="http://emboss.sourceforge.net/apps/release/6.3/emboss/apps/matcher.html">matcher</a> or <cite>water &lt;http://emboss.sourceforge.net/apps/release/6.3/emboss/apps/water.html&gt;</cite>.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ matcher frataxin.human.gb frataxin.aedes.gb stdout
</pre></div>
</div>
<ul class="simple">
<li><p>Is the alignment local?</p></li>
<li><p>Why are there regions that has not been aligned?</p></li>
</ul>
<p>Why some regions are more similar than others?</p>
<ul class="simple">
<li><p>Is there anything special with aminoacids 92, 100, 101, 104, 108, 111, 112, 122 and 124 in the human frataxin?</p></li>
</ul>
<p>Open in a text editor the genbank files for the human and mosquito proteins and check the annotation of those aminoacids.</p>
<p>Let’s try to modify the parameters given to the aligner.
Let’s do the alignment with the parameter gapopen set to 4.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ matcher frataxin.human.gb frataxin.aedes.gb stdout -gapopen 4
</pre></div>
</div>
<ul class="simple">
<li><p>Is this alignment better or worse?</p></li>
</ul>
<p>Every score set will produce more than one alignment, but matcher will only show, by default, only one alignment.
Ask matcher to show 3 alternative alignments</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ matcher frataxin.human.gb frataxin.aedes.gb stdout -alternatives 3
</pre></div>
</div>
<p>Why the first alignment is the best one?</p>
</section>
</section>
<section id="gobal-alignments">
<h2>Gobal alignments<a class="headerlink" href="#gobal-alignments" title="Link to this heading">¶</a></h2>
<p>We could use an algorithm that aligns both sequences in its whole length.</p>
<p>Redo the alignments from the last two exercises using <a class="reference external" href="http://emboss.sourceforge.net/apps/release/6.3/emboss/apps/needle.html">needle</a>.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ needle frataxin.human.gb frataxin.aedes.gb stdout -auto
</pre></div>
</div>
<ul class="simple">
<li><p>Is this alignment better or worse than the previous one?</p></li>
</ul>
<p>The global alignments should only be used when we are sure that the two sequences are similar in all its length.
These algorithms will give a very bad result if there are regions that are not similar between the sequences.</p>
</section>
<section id="genic-structure-1">
<h2>Genic structure 1<a class="headerlink" href="#genic-structure-1" title="Link to this heading">¶</a></h2>
<p>We have aligned a cDNA and the  genomic sequence of a gene with matcher. Do you think that it is a good approach? Could you think on any problems regarding the approach?</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>             <span class="mi">560</span>       <span class="mi">570</span>       <span class="mi">580</span>       <span class="mi">590</span>       <span class="mi">600</span>
<span class="n">cDNA</span> <span class="n">TTCAGGAGTATTTAGATTACGCGAACAGGTATGTAAACACTGAGCCAGCA</span>
     <span class="p">:</span> <span class="p">::::::::::::::::::::::::::::::::::::::::::::::::</span>
<span class="n">GENO</span> <span class="n">TGCAGGAGTATTTAGATTACGCGAACAGGTATGTAAACACTGAGCCAGCA</span>
     <span class="mi">1550</span>      <span class="mi">1560</span>      <span class="mi">1570</span>      <span class="mi">1580</span>      <span class="mi">1590</span>

             <span class="mi">610</span>       <span class="mi">620</span>       <span class="mi">630</span>       <span class="mi">640</span>       <span class="mi">650</span>
<span class="n">cDNA</span> <span class="n">GAGTAAAAGGAAACGGGTAATGACAAAATAGGGAAGCCGCTAGTTTCGGA</span>
     <span class="p">::::::::::::::::::::::::::::::::::::::::::::::::::</span>
<span class="n">GENO</span> <span class="n">GAGTAAAAGGAAACGGGTAATGACAAAATAGGGAAGCCGCTAGTTTCGGA</span>
     <span class="mi">1600</span>      <span class="mi">1610</span>      <span class="mi">1620</span>      <span class="mi">1630</span>      <span class="mi">1640</span>

             <span class="mi">660</span>       <span class="mi">670</span>       <span class="mi">680</span>       <span class="mi">690</span>       <span class="mi">700</span>
<span class="n">cDNA</span> <span class="n">TATATGAATATAATGATGCTTGATTTGTTTTGACATTATCTATGTGTACA</span>
     <span class="p">::::::::::::::::::::::::::::::::::::::::::::::::::</span>
<span class="n">GENO</span> <span class="n">TATATGAATATAATGATGCTTGATTTGTTTTGACATTATCTATGTGTACA</span>
     <span class="mi">1650</span>      <span class="mi">1660</span>      <span class="mi">1670</span>      <span class="mi">1680</span>      <span class="mi">1690</span>

             <span class="mi">710</span>       <span class="mi">720</span>       <span class="mi">730</span>       <span class="mi">740</span>
<span class="n">cDNA</span> <span class="n">AAGACATCTTGCATCTTTTAAGAAATTCGTTTTTCGCTGC</span>
     <span class="p">::::::::::::::::::::::::::::::::::::::::</span>
<span class="n">GENO</span> <span class="n">AAGACATCTTGCATCTTTTAAGAAATTCGTTTTTCGCTGC</span>
     <span class="mi">1700</span>      <span class="mi">1710</span>      <span class="mi">1720</span>      <span class="mi">1730</span>
</pre></div>
</div>
</section>
<section id="genic-structure-2">
<h2>Genic structure 2<a class="headerlink" href="#genic-structure-2" title="Link to this heading">¶</a></h2>
<p>Determine the genic structure of the AT3G4905 <em>Arabidopsis</em> gene.
You need to dermine where a the exons and introns located and which are the conding sequences.
You have part of the <a class="reference download internal" download="" href="../_downloads/7f470d3314118d610ad8e7220f183715/AT3G4905.1.genomico.fasta"><code class="xref download docutils literal notranslate"><span class="pre">genomic</span></code></a> sequence, a <a class="reference download internal" download="" href="../_downloads/1b09e08e6c6ee7cd9cf575a950388591/AT3G4905.1.cdna.fasta"><code class="xref download docutils literal notranslate"><span class="pre">cDNA</span></code></a> and the <a class="reference download internal" download="" href="../_downloads/4530a2a2f3be76030111a329915b5826/AT3G4905.1.cds.fasta"><code class="xref download docutils literal notranslate"><span class="pre">CDS</span></code></a> for that cDNA.</p>
<p>EMBOSS has a program created just with the aim of alignming messenger RNA with a genomic sequences.
These kind of alingments are special:</p>
<ul class="simple">
<li><p>They are very similar in the regions share by the genome and the mRNA</p></li>
<li><p>They have long gaps (the introns)</p></li>
<li><p>The exons should be surronded by some specific sequence <a class="reference external" href="https://en.wikipedia.org/wiki/RNA_splicing#Spliceosomal">intronic sequences</a>.</p></li>
</ul>
<p>Let’s align the genomic sequence of an <em>Arabidopsis</em> gene <a class="reference download internal" download="" href="../_downloads/7f470d3314118d610ad8e7220f183715/AT3G4905.1.genomico.fasta"><code class="xref download docutils literal notranslate"><span class="pre">AT3G4905</span></code></a> with a <a class="reference download internal" download="" href="../_downloads/1b09e08e6c6ee7cd9cf575a950388591/AT3G4905.1.cdna.fasta"><code class="xref download docutils literal notranslate"><span class="pre">cDNA</span></code></a>.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ est2genome AT3G4905.1.cdna.fasta AT3G4905.1.genomico.fasta stdout -auto
</pre></div>
</div>
<ul class="simple">
<li><p>How many introns are there?</p></li>
</ul>
<p>To get the alignment report with the complete sequence:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ est2genome AT3G4905.1.cdna.fasta AT3G4905.1.genomico.fasta stdout -align -auto
</pre></div>
</div>
<ul class="simple">
<li><p>Would needle or matcher create the same alignment?</p></li>
<li><p>Compare the sequence of the <a class="reference download internal" download="" href="../_downloads/4530a2a2f3be76030111a329915b5826/AT3G4905.1.cds.fasta"><code class="xref download docutils literal notranslate"><span class="pre">CDS</span></code></a> of this cDNA with the sequence of the genome.</p></li>
<li><p>Is the alignment similar or different to previous one?. Why?</p></li>
<li><p>Do the alignment between CDS and mRNA. Which conclusions could you reach?</p></li>
</ul>
</section>
<section id="protein-alignment">
<h2>Protein alignment<a class="headerlink" href="#protein-alignment" title="Link to this heading">¶</a></h2>
<p>The alignment between the SOX9 and NHE-RF2 proteins gave this result. could you infer that this proteins are homologues? Is this region a conserved domain? Check your answers in the databases.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="c1">#=======================================</span>
<span class="c1">#</span>
<span class="c1"># Aligned_sequences: 2</span>
<span class="c1"># 1: AHA92033.1</span>
<span class="c1"># 2: XP_016879383.1</span>
<span class="c1"># Matrix: EBLOSUM62</span>
<span class="c1"># Gap_penalty: 14</span>
<span class="c1"># Extend_penalty: 4</span>
<span class="c1">#</span>
<span class="c1"># Length: 33</span>
<span class="c1"># Identity:      12/33 (36.4%)</span>
<span class="c1"># Similarity:    13/33 (39.4%)</span>
<span class="c1"># Gaps:           0/33 ( 0.0%)</span>
<span class="c1"># Score: 49</span>
<span class="c1">#</span>
<span class="c1">#</span>
<span class="c1">#=======================================</span>

           <span class="mi">330</span>       <span class="mi">340</span>       <span class="mi">350</span>
<span class="n">AHA920</span> <span class="n">TAATPAGAGHVWMSKQQAPPPPPPPQQPPPAPP</span>
       <span class="p">:</span> <span class="p">::</span>      <span class="p">:</span>      <span class="p">:</span> <span class="p">::</span>   <span class="p">:</span>  <span class="p">::</span> <span class="o">.</span><span class="p">::</span>
<span class="n">XP_016</span> <span class="n">TPATCCHCAAVMARSGSATPPARAPGAPPRSPP</span>
      <span class="mi">150</span>       <span class="mi">160</span>       <span class="mi">170</span>       <span class="mi">180</span>
</pre></div>
</div>
</section>
<section id="bl2seq">
<h2>bl2seq<a class="headerlink" href="#bl2seq" title="Link to this heading">¶</a></h2>
<p>An often used program to align two sequences is bl2seq.
We are going to use a web service to align with <a class="reference external" href="http://blast.ncbi.nlm.nih.gov/Blast.cgi?PAGE_TYPE=BlastSearch&amp;PROG_DEF=blastn&amp;BLAST_PROG_DEF=megaBlast&amp;BLAST_SPEC=blast2seq">bl2seq</a>.</p>
<p>Blast uses a different algorithm to the software used before in this practice.
This algorithm is faster, but it is not guaranteed that we will get the best alignment.</p>
<p>Bl2seq is a good alignment method when the sequences are large and the other methods won’t produce a solution in a reasonable time.
Blast is used more often to compare one sequence with a whole database of sequences.</p>
<p>Compare the blast result with the previous <a class="reference download internal" download="" href="../_downloads/7f470d3314118d610ad8e7220f183715/AT3G4905.1.genomico.fasta"><code class="xref download docutils literal notranslate"><span class="pre">genomic</span></code></a> vs <a class="reference download internal" download="" href="../_downloads/1b09e08e6c6ee7cd9cf575a950388591/AT3G4905.1.cdna.fasta"><code class="xref download docutils literal notranslate"><span class="pre">cDNA</span></code></a> alignment.</p>
<ul class="simple">
<li><p>How many alignments have we obtained?</p></li>
</ul>
</section>
<section id="matcher-vs-bl2seq">
<h2>matcher vs bl2seq<a class="headerlink" href="#matcher-vs-bl2seq" title="Link to this heading">¶</a></h2>
<p>We have aligned using both matcher and bl2seq a sequence against itself. Could you suggest an interpretacion of the results?</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="c1">########################################</span>
<span class="c1"># Program: matcher</span>
<span class="c1"># Rundate: Mon 16 Oct 2017 14:57:26</span>
<span class="c1"># Commandline: matcher</span>
<span class="c1">#    -auto</span>
<span class="c1">#    -asequence /var/lib/emboss-explorer/output/783917/.asequence</span>
<span class="c1">#    -bsequence /var/lib/emboss-explorer/output/783917/.bsequence</span>
<span class="c1">#    -alternatives 1</span>
<span class="c1">#    -outfile outfile</span>
<span class="c1">#    -aformat3 markx0</span>
<span class="c1"># Align_format: markx0</span>
<span class="c1"># Report_file: outfile</span>
<span class="c1">########################################</span>

<span class="c1">#=======================================</span>
<span class="c1">#</span>
<span class="c1"># Aligned_sequences: 2</span>
<span class="c1"># 1: seq_bi</span>
<span class="c1"># 2: seq_bi</span>
<span class="c1"># Matrix: EDNAFULL</span>
<span class="c1"># Gap_penalty: 16</span>
<span class="c1"># Extend_penalty: 4</span>
<span class="c1">#</span>
<span class="c1"># Length: 250</span>
<span class="c1"># Identity:     250/250 (100.0%)</span>
<span class="c1"># Similarity:   250/250 (100.0%)</span>
<span class="c1"># Gaps:           0/250 ( 0.0%)</span>
<span class="c1"># Score: 1250</span>
<span class="c1">#</span>
<span class="c1">#</span>
<span class="c1">#=======================================</span>

               <span class="mi">10</span>        <span class="mi">20</span>        <span class="mi">30</span>        <span class="mi">40</span>        <span class="mi">50</span>
<span class="n">seq_bi</span> <span class="n">TCGCGAAGACATTTAGCAGAACCATTTGTCGTTTCCTCTCGCGATCAGGC</span>
       <span class="p">::::::::::::::::::::::::::::::::::::::::::::::::::</span>
<span class="n">seq_bi</span> <span class="n">TCGCGAAGACATTTAGCAGAACCATTTGTCGTTTCCTCTCGCGATCAGGC</span>
               <span class="mi">10</span>        <span class="mi">20</span>        <span class="mi">30</span>        <span class="mi">40</span>        <span class="mi">50</span>

               <span class="mi">60</span>        <span class="mi">70</span>        <span class="mi">80</span>        <span class="mi">90</span>       <span class="mi">100</span>
<span class="n">seq_bi</span> <span class="n">CAGATTCCGTGCTCCGGAGCCAGAAAGTTTCCGCCTTTAAAGAAATGAAG</span>
       <span class="p">::::::::::::::::::::::::::::::::::::::::::::::::::</span>
<span class="n">seq_bi</span> <span class="n">CAGATTCCGTGCTCCGGAGCCAGAAAGTTTCCGCCTTTAAAGAAATGAAG</span>
               <span class="mi">60</span>        <span class="mi">70</span>        <span class="mi">80</span>        <span class="mi">90</span>       <span class="mi">100</span>

              <span class="mi">110</span>       <span class="mi">120</span>       <span class="mi">130</span>       <span class="mi">140</span>       <span class="mi">150</span>
<span class="n">seq_bi</span> <span class="n">TAGCTATCTCAGATCCTTCAAATATGGTTGCTTCTCCATTGAGTGTTTTG</span>
       <span class="p">::::::::::::::::::::::::::::::::::::::::::::::::::</span>
<span class="n">seq_bi</span> <span class="n">TAGCTATCTCAGATCCTTCAAATATGGTTGCTTCTCCATTGAGTGTTTTG</span>
              <span class="mi">110</span>       <span class="mi">120</span>       <span class="mi">130</span>       <span class="mi">140</span>       <span class="mi">150</span>

              <span class="mi">160</span>       <span class="mi">170</span>       <span class="mi">180</span>       <span class="mi">190</span>       <span class="mi">200</span>
<span class="n">seq_bi</span> <span class="n">CAGATTCCGTGCTCCGGAGCCAGAAAGTTTCCGCCTTTAAAGAAATGAAG</span>
       <span class="p">::::::::::::::::::::::::::::::::::::::::::::::::::</span>
<span class="n">seq_bi</span> <span class="n">CAGATTCCGTGCTCCGGAGCCAGAAAGTTTCCGCCTTTAAAGAAATGAAG</span>
              <span class="mi">160</span>       <span class="mi">170</span>       <span class="mi">180</span>       <span class="mi">190</span>       <span class="mi">200</span>

              <span class="mi">210</span>       <span class="mi">220</span>       <span class="mi">230</span>       <span class="mi">240</span>       <span class="mi">250</span>
<span class="n">seq_bi</span> <span class="n">CTCAGAAAGAAATCAAACATTGACCTGATGGCTACAGATTTCCAGAACCT</span>
       <span class="p">::::::::::::::::::::::::::::::::::::::::::::::::::</span>
<span class="n">seq_bi</span> <span class="n">CTCAGAAAGAAATCAAACATTGACCTGATGGCTACAGATTTCCAGAACCT</span>
              <span class="mi">210</span>       <span class="mi">220</span>       <span class="mi">230</span>       <span class="mi">240</span>       <span class="mi">250</span>
</pre></div>
</div>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">BLASTN</span> <span class="mf">2.7.0</span><span class="o">+</span>
<span class="n">Reference</span><span class="p">:</span> <span class="n">Zheng</span> <span class="n">Zhang</span><span class="p">,</span> <span class="n">Scott</span> <span class="n">Schwartz</span><span class="p">,</span> <span class="n">Lukas</span> <span class="n">Wagner</span><span class="p">,</span> <span class="ow">and</span>
<span class="n">Webb</span> <span class="n">Miller</span> <span class="p">(</span><span class="mi">2000</span><span class="p">),</span> <span class="s2">&quot;A greedy algorithm for aligning DNA</span>
<span class="n">sequences</span><span class="s2">&quot;, J Comput Biol 2000; 7(1-2):203-14.</span>


<span class="n">RID</span><span class="p">:</span> <span class="n">Y9B74S2U114</span>


<span class="n">Query</span><span class="o">=</span> <span class="n">seq_bi</span>

<span class="n">Length</span><span class="o">=</span><span class="mi">250</span>


                                                                   <span class="n">Score</span>     <span class="n">E</span>
<span class="n">Sequences</span> <span class="n">producing</span> <span class="n">significant</span> <span class="n">alignments</span><span class="p">:</span>                       <span class="p">(</span><span class="n">Bits</span><span class="p">)</span>  <span class="n">Value</span>

<span class="n">Query_163477</span>  <span class="n">seq_bi</span>                                               <span class="mi">462</span>     <span class="mf">3e-135</span>

<span class="n">ALIGNMENTS</span>
<span class="o">&gt;</span><span class="n">Query_163477</span> <span class="n">seq_bi</span>
<span class="n">Length</span><span class="o">=</span><span class="mi">250</span>

 <span class="n">Score</span> <span class="o">=</span> <span class="mi">462</span> <span class="n">bits</span> <span class="p">(</span><span class="mi">250</span><span class="p">),</span>  <span class="n">Expect</span> <span class="o">=</span> <span class="mf">3e-135</span>
 <span class="n">Identities</span> <span class="o">=</span> <span class="mi">250</span><span class="o">/</span><span class="mi">250</span> <span class="p">(</span><span class="mi">100</span><span class="o">%</span><span class="p">),</span> <span class="n">Gaps</span> <span class="o">=</span> <span class="mi">0</span><span class="o">/</span><span class="mi">250</span> <span class="p">(</span><span class="mi">0</span><span class="o">%</span><span class="p">)</span>
 <span class="n">Strand</span><span class="o">=</span><span class="n">Plus</span><span class="o">/</span><span class="n">Plus</span>

<span class="n">Query</span>  <span class="mi">1</span>    <span class="n">TCGCGAAGACATTTAGCAGAACCATTTGTCGTTTCCTCTCGCGATCAGGCCAGATTCCGT</span>  <span class="mi">60</span>
            <span class="o">||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||</span>
<span class="n">Sbjct</span>  <span class="mi">1</span>    <span class="n">TCGCGAAGACATTTAGCAGAACCATTTGTCGTTTCCTCTCGCGATCAGGCCAGATTCCGT</span>  <span class="mi">60</span>

<span class="n">Query</span>  <span class="mi">61</span>   <span class="n">GCTCCGGAGCCAGAAAGTTTCCGCCTTTAAAGAAATGAAGTAGCTATCTCAGATCCTTCA</span>  <span class="mi">120</span>
            <span class="o">||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||</span>
<span class="n">Sbjct</span>  <span class="mi">61</span>   <span class="n">GCTCCGGAGCCAGAAAGTTTCCGCCTTTAAAGAAATGAAGTAGCTATCTCAGATCCTTCA</span>  <span class="mi">120</span>

<span class="n">Query</span>  <span class="mi">121</span>  <span class="n">AATATGGTTGCTTCTCCATTGAGTGTTTTGCAGATTCCGTGCTCCGGAGCCAGAAAGTTT</span>  <span class="mi">180</span>
            <span class="o">||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||</span>
<span class="n">Sbjct</span>  <span class="mi">121</span>  <span class="n">AATATGGTTGCTTCTCCATTGAGTGTTTTGCAGATTCCGTGCTCCGGAGCCAGAAAGTTT</span>  <span class="mi">180</span>

<span class="n">Query</span>  <span class="mi">181</span>  <span class="n">CCGCCTTTAAAGAAATGAAGCTCAGAAAGAAATCAAACATTGACCTGATGGCTACAGATT</span>  <span class="mi">240</span>
            <span class="o">||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||</span>
<span class="n">Sbjct</span>  <span class="mi">181</span>  <span class="n">CCGCCTTTAAAGAAATGAAGCTCAGAAAGAAATCAAACATTGACCTGATGGCTACAGATT</span>  <span class="mi">240</span>

<span class="n">Query</span>  <span class="mi">241</span>  <span class="n">TCCAGAACCT</span>  <span class="mi">250</span>
            <span class="o">||||||||||</span>
<span class="n">Sbjct</span>  <span class="mi">241</span>  <span class="n">TCCAGAACCT</span>  <span class="mi">250</span>


 <span class="n">Score</span> <span class="o">=</span> <span class="mf">93.5</span> <span class="n">bits</span> <span class="p">(</span><span class="mi">50</span><span class="p">),</span>  <span class="n">Expect</span> <span class="o">=</span> <span class="mf">4e-24</span>
 <span class="n">Identities</span> <span class="o">=</span> <span class="mi">50</span><span class="o">/</span><span class="mi">50</span> <span class="p">(</span><span class="mi">100</span><span class="o">%</span><span class="p">),</span> <span class="n">Gaps</span> <span class="o">=</span> <span class="mi">0</span><span class="o">/</span><span class="mi">50</span> <span class="p">(</span><span class="mi">0</span><span class="o">%</span><span class="p">)</span>
 <span class="n">Strand</span><span class="o">=</span><span class="n">Plus</span><span class="o">/</span><span class="n">Plus</span>

<span class="n">Query</span>  <span class="mi">151</span>  <span class="n">CAGATTCCGTGCTCCGGAGCCAGAAAGTTTCCGCCTTTAAAGAAATGAAG</span>  <span class="mi">200</span>
            <span class="o">||||||||||||||||||||||||||||||||||||||||||||||||||</span>
<span class="n">Sbjct</span>  <span class="mi">51</span>   <span class="n">CAGATTCCGTGCTCCGGAGCCAGAAAGTTTCCGCCTTTAAAGAAATGAAG</span>  <span class="mi">100</span>


 <span class="n">Score</span> <span class="o">=</span> <span class="mf">93.5</span> <span class="n">bits</span> <span class="p">(</span><span class="mi">50</span><span class="p">),</span>  <span class="n">Expect</span> <span class="o">=</span> <span class="mf">4e-24</span>
 <span class="n">Identities</span> <span class="o">=</span> <span class="mi">50</span><span class="o">/</span><span class="mi">50</span> <span class="p">(</span><span class="mi">100</span><span class="o">%</span><span class="p">),</span> <span class="n">Gaps</span> <span class="o">=</span> <span class="mi">0</span><span class="o">/</span><span class="mi">50</span> <span class="p">(</span><span class="mi">0</span><span class="o">%</span><span class="p">)</span>
 <span class="n">Strand</span><span class="o">=</span><span class="n">Plus</span><span class="o">/</span><span class="n">Plus</span>

<span class="n">Query</span>  <span class="mi">51</span>   <span class="n">CAGATTCCGTGCTCCGGAGCCAGAAAGTTTCCGCCTTTAAAGAAATGAAG</span>  <span class="mi">100</span>
            <span class="o">||||||||||||||||||||||||||||||||||||||||||||||||||</span>
<span class="n">Sbjct</span>  <span class="mi">151</span>  <span class="n">CAGATTCCGTGCTCCGGAGCCAGAAAGTTTCCGCCTTTAAAGAAATGAAG</span>  <span class="mi">200</span>
</pre></div>
</div>
</section>
<section id="primer-search">
<h2>Primer search<a class="headerlink" href="#primer-search" title="Link to this heading">¶</a></h2>
<p>We have developed two PCR primers: <a class="reference download internal" download="" href="../_downloads/e066e195736e0ca893074bbec9c60c63/primerf.fasta"><code class="xref download docutils literal notranslate"><span class="pre">primerf</span></code></a> and <a class="reference download internal" download="" href="../_downloads/95d47cfd2e7e934d1103b22bf6832873/primerr.fasta"><code class="xref download docutils literal notranslate"><span class="pre">primerr</span></code></a> to amplify the <a class="reference download internal" download="" href="../_downloads/099955f3cecea2f5195f1ec9d1766cfb/CDKL4.human.fasta"><code class="xref download docutils literal notranslate"><span class="pre">cyclin</span></code></a> CDS.</p>
<p>Can you locate the primers in the cyclin sequence?</p>
</section>
</section>


          </div>
          
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
            <p class="logo"><a href="../index.html">
              <img class="logo" src="../_static/comav.png" alt="Logo"/>
            </a></p>
<h1 class="logo"><a href="../index.html">Bioinformatics at COMAV</a></h1>








<h3>Navigation</h3>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../norms.html">Norms</a></li>
<li class="toctree-l1"><a class="reference internal" href="../norms.html#tutoring">Tutoring</a></li>
<li class="toctree-l1"><a class="reference internal" href="../norms.html#id1">Norms</a></li>
</ul>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../prev_knowledge.html">Previous knowledge</a></li>
</ul>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../theory/databases.html">Databases</a></li>
<li class="toctree-l1"><a class="reference internal" href="../theory/sequence_alignment.html">Sequence alignment</a></li>
<li class="toctree-l1"><a class="reference internal" href="../theory/blast.html">Blast</a></li>
<li class="toctree-l1"><a class="reference internal" href="../theory/multiple.html">Multiple</a></li>
<li class="toctree-l1"><a class="reference internal" href="../theory/structural_annotation.html">Structural annotation</a></li>
<li class="toctree-l1"><a class="reference internal" href="../theory/phylogeny.html">Phylogeny</a></li>
<li class="toctree-l1"><a class="reference internal" href="../theory/sequence_analysis.html">Sequence analysis</a></li>
</ul>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="entrez.html">NCBI search</a></li>
<li class="toctree-l1"><a class="reference internal" href="emboss.html">Emboss</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Sequence Alignment</a></li>
<li class="toctree-l1"><a class="reference internal" href="blast.html">Blast</a></li>
<li class="toctree-l1"><a class="reference internal" href="multiple.html">Multiple Sequence Alignment</a></li>
<li class="toctree-l1"><a class="reference internal" href="structural_annotation.html">Structural annotation</a></li>
<li class="toctree-l1"><a class="reference internal" href="phylogeny.html">Phylogeny</a></li>
</ul>

<div class="relations">
<h3>Related Topics</h3>
<ul>
  <li><a href="../index.html">Documentation overview</a><ul>
      <li>Previous: <a href="emboss.html" title="previous chapter">Introduction to EMBOSS</a></li>
      <li>Next: <a href="blast.html" title="next chapter">Blast Practice</a></li>
  </ul></li>
</ul>
</div>
<div id="searchbox" style="display: none" role="search">
  <h3 id="searchlabel">Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="../search.html" method="get">
      <input type="text" name="q" aria-labelledby="searchlabel" autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false"/>
      <input type="submit" value="Go" />
    </form>
    </div>
</div>
<script>document.getElementById('searchbox').style.display = "block"</script>








        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="footer">
      &copy;2010, Jose Blanca & Joaquin Cañizares.
      
      |
      Powered by <a href="http://sphinx-doc.org/">Sphinx 7.2.6</a>
      &amp; <a href="https://github.com/bitprophet/alabaster">Alabaster 0.7.13</a>
      
    </div>

    

    
  </body>
</html>