<!DOCTYPE html>

<html lang="en" data-content_root="../">
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="viewport" content="width=device-width, initial-scale=1" />

    <title>Introduction to EMBOSS &#8212; Bioinformatics at COMAV 0.1 documentation</title>
    <link rel="stylesheet" type="text/css" href="../_static/pygments.css?v=fa44fd50" />
    <link rel="stylesheet" type="text/css" href="../_static/basic.css?v=c058f7c8" />
    <link rel="stylesheet" type="text/css" href="../_static/alabaster.css?v=27fed22d" />
    <script src="../_static/documentation_options.js?v=2709fde1"></script>
    <script src="../_static/doctools.js?v=9a2dae69"></script>
    <script src="../_static/sphinx_highlight.js?v=dc90522c"></script>
    <link rel="icon" href="../_static/favicon.ico"/>
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Sequence Alignment" href="alignment.html" />
    <link rel="prev" title="NCBI database search" href="entrez.html" />
   
  <link rel="stylesheet" href="../_static/custom.css" type="text/css" />
  

  
  

  </head><body>
  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          

          <div class="body" role="main">
            
  <section id="introduction-to-emboss">
<h1>Introduction to EMBOSS<a class="headerlink" href="#introduction-to-emboss" title="Link to this heading">¶</a></h1>
<section id="emboss">
<h2>EMBOSS<a class="headerlink" href="#emboss" title="Link to this heading">¶</a></h2>
<p><a class="reference external" href="http://emboss.sourceforge.net/">EMBOSS</a> is a suite of programs developed to do molecular biology by EMBnet.
EMBOSS is <a class="reference external" href="https://en.wikipedia.org/wiki/Free_software">free software</a> and although it is primarily developed for Linux it also runs in MS Windows and Mac OS X.</p>
<p>EMBOSS is comprised by hunders of programs and includes utilities to align sequences, translate RNA to proteins, identify protien motifs, analyze repetitions, calculate codon usage statistics and many more analyses.</p>
<section id="running-emboss-commands">
<h3>Running EMBOSS commands<a class="headerlink" href="#running-emboss-commands" title="Link to this heading">¶</a></h3>
<p>The EMBOSS utilities have a <a class="reference external" href="https://en.wikipedia.org/wiki/Command-line_interface">Command Line Interface</a> and all can be run from a text terminal.
The <a class="reference external" href="http://emboss.sourceforge.net/apps/release/6.3/emboss/apps/groups.html">documentation</a> for the EMBOSS aplications is written taking into account mainly the Command Line Interface.</p>
<p>But there are other ways to run the EMBOSS software.</p>
<p>You can use <a class="reference external" href="https://www.bioinformatics.nl/cgi-bin/emboss/">emboss online</a> in the Wageningen server.</p>
<p>There are web <a class="reference external" href="http://emboss.sourceforge.net/servers/#explorer">servers</a> that provide the EMBOSS tools as a service.
In those servers the EMBOSS utilities can be accessed as web pages and the server runs the terminal command and return the result.</p>
<p>Another interface to EMBOSS is <a class="reference external" href="http://emboss.sourceforge.net/Jemboss/">Jemboss</a>.
Jemboss is a Graphical User Interface application, like any of the other common desktop applications.</p>
</section>
</section>
<section id="identify-the-command">
<h2>Identify the command<a class="headerlink" href="#identify-the-command" title="Link to this heading">¶</a></h2>
<p>When we do not know which of the EMBOSS commands to use we can use another EMBOSS command, <a class="reference external" href="http://emboss.sourceforge.net/apps/release/6.3/emboss/apps/wossname.html">wossname</a>, to search in the EMBOSS command list.</p>
<p>Get a list with all commands:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ wossname
</pre></div>
</div>
<p>Look for command to do alignments:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ wossname alignment
</pre></div>
</div>
</section>
<section id="getting-help-for-a-command">
<h2>Getting help for a command<a class="headerlink" href="#getting-help-for-a-command" title="Link to this heading">¶</a></h2>
<p>The <a class="reference external" href="http://emboss.sourceforge.net/apps/release/6.3/emboss/apps/groups.html">command</a> manuals can be found in the web.</p>
<p>They are also available in any EMBOSS installation by using the <a class="reference external" href="http://emboss.sourceforge.net/apps/release/6.3/emboss/apps/tfm.html">tfm</a> command.
For instance, to read the manual for the <a class="reference external" href="http://emboss.sourceforge.net/apps/release/6.3/emboss/apps/water.html">water</a> command we could run:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ tfm water
</pre></div>
</div>
<p>The manual reader is similar to the Unix less command.</p>
<p>It is also possible to get a brief list of parameters available with the option “-help” and the complete list with “-help -verbose”</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ water -help -verbose
</pre></div>
</div>
</section>
<section id="example-change-a-sequence-format">
<h2>Example: change a sequence format<a class="headerlink" href="#example-change-a-sequence-format" title="Link to this heading">¶</a></h2>
<p>To get familiar with the EMBOSS way of running the analyse we will change the format of a sequence file.
EMBOSS has a program to modify sequence files: <a class="reference external" href="http://emboss.sourceforge.net/apps/release/6.3/emboss/apps/seqret.html">seqret</a></p>
<p>seqret can change the sequence format, trim the sequences and reverse-complement them.
Let’s change the format of fasta <a class="reference external" href="../static/emboss/sequence.fasta">sequence file</a> to the GenBank format.</p>
<p>If we run seqset in the command line it will ask for the input file and for the output file:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ seqret
</pre></div>
</div>
<p>By default seqret won’t change the sequence format unless we specify the output format that we want.
To do it we have to write <em>genbank::secuencia.gb</em>.
We can also tell seqret which is our input file:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ seqret secuencia.fasta genbank::secuencia.gb
</pre></div>
</div>
<p>We could sent the result to standard output instead of a file.
This is a general feature of all EMBOSS programs.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ seqret secuencia.fasta genbank::stdout
</pre></div>
</div>
<p>To reverse and complement a sequence we can use the parameter -sreverse1:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ seqret secuencia.fasta embl::stdout -sreverse1
</pre></div>
</div>
<p>Would the result sequence be different in this case?</p>
<p>seqret can also trim the input sequence.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ seqret secuencia.fasta embl::stdout -sbegin1 10 -send1 30
</pre></div>
</div>
</section>
</section>


          </div>
          
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="Main">
        <div class="sphinxsidebarwrapper">
            <p class="logo"><a href="../index.html">
              <img class="logo" src="../_static/comav.png" alt="Logo of Bioinformatics at COMAV"/>
            </a></p>
<h1 class="logo"><a href="../index.html">Bioinformatics at COMAV</a></h1>









<search id="searchbox" style="display: none" role="search">
    <div class="searchformwrapper">
    <form class="search" action="../search.html" method="get">
      <input type="text" name="q" aria-labelledby="searchlabel" autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" placeholder="Search"/>
      <input type="submit" value="Go" />
    </form>
    </div>
</search>
<script>document.getElementById('searchbox').style.display = "block"</script><h3>Navigation</h3>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../norms.html">Norms</a></li>
<li class="toctree-l1"><a class="reference internal" href="../norms.html#tutoring">Tutoring</a></li>
<li class="toctree-l1"><a class="reference internal" href="../norms.html#id1">Norms</a></li>
</ul>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../prev_knowledge.html">Previous knowledge</a></li>
</ul>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../theory/databases.html">Databases</a></li>
<li class="toctree-l1"><a class="reference internal" href="../theory/sequence_alignment.html">Sequence alignment</a></li>
<li class="toctree-l1"><a class="reference internal" href="../theory/blast.html">Blast</a></li>
<li class="toctree-l1"><a class="reference internal" href="../theory/multiple.html">Multiple</a></li>
<li class="toctree-l1"><a class="reference internal" href="../theory/structural_annotation.html">Structural annotation</a></li>
<li class="toctree-l1"><a class="reference internal" href="../theory/phylogeny.html">Phylogeny</a></li>
<li class="toctree-l1"><a class="reference internal" href="../theory/sequence_analysis.html">Sequence analysis</a></li>
</ul>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="entrez.html">NCBI search</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Emboss</a></li>
<li class="toctree-l1"><a class="reference internal" href="alignment.html">Sequence Alignment</a></li>
<li class="toctree-l1"><a class="reference internal" href="blast.html">Blast</a></li>
<li class="toctree-l1"><a class="reference internal" href="multiple.html">Multiple Sequence Alignment</a></li>
<li class="toctree-l1"><a class="reference internal" href="structural_annotation.html">Structural annotation</a></li>
<li class="toctree-l1"><a class="reference internal" href="phylogeny.html">Phylogeny</a></li>
</ul>

<div class="relations">
<h3>Related Topics</h3>
<ul>
  <li><a href="../index.html">Documentation overview</a><ul>
      <li>Previous: <a href="entrez.html" title="previous chapter">NCBI database search</a></li>
      <li>Next: <a href="alignment.html" title="next chapter">Sequence Alignment</a></li>
  </ul></li>
</ul>
</div>








        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="footer">
      &#169;2010, Jose Blanca & Joaquin Cañizares.
      
      |
      Powered by <a href="https://www.sphinx-doc.org/">Sphinx 8.0.2</a>
      &amp; <a href="https://alabaster.readthedocs.io">Alabaster 1.0.0</a>
      
    </div>

    

    
  </body>
</html>