
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    <meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

    <title>Introduction to EMBOSS &mdash; Bioinformatics at COMAV 0.1 documentation</title>
    <link rel="stylesheet" href="../_static/style.css" type="text/css" />
    <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    <link rel="stylesheet" href="../_static/style.css" type="text/css" />
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    '../',
        VERSION:     '0.1',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="../_static/documentation_options.js"></script>
    <script type="text/javascript" src="../_static/jquery.js"></script>
    <script type="text/javascript" src="../_static/underscore.js"></script>
    <script type="text/javascript" src="../_static/doctools.js"></script>
    <link rel="shortcut icon" href="../_static/favicon.ico"/>
    <link rel="search" title="Search" href="../search.html" />
    <link rel="top" title="Bioinformatics at COMAV 0.1 documentation" href="../index.html" />
    <link rel="next" title="Sequence Alignment" href="alignment.html" />
    <link rel="prev" title="Unix CLI" href="linux.html" /> 
  </head>
  <body>

<div class="body">
    <header>
        <h1><a href="index.html">Bioinformatics at COMAV</a></h1>
    </header>


    <nav class="horizontal_nav page_nav">
        <ul>
            <li><a href="index.html">Bioinformatics & genomics</a>
                <ul>
                    <li><a href="/services.html">Services</a></li>
                    <li><a href="/people.html">People</a></li>
                    <li><a href="/publications.html">Publications</a></li>
                    <li><a href="/getting_here.html">Getting here</a></li>
<!--                    <li><a href="http://bioinf.comav.upv.es/notes/">Wiki</a></li>-->
               </ul>
            </li>
            <li><a href="/software.html">Software</a>
                <ul>
                    <li><a href="https://github.com/bioinfcomav/">Bioinf COMAV</a></li>
                    <li><a href="https://github.com/JoseBlanca/">Jose Blanca</a></li>
                    <!--
                    <li><a href="/seq_crumbs/index.html">seq_crumbs</a></li>
                    <li><a href="/ngs_backbone/index.html">ngs_backbone</a></li>
                    <li><a href="/clean_reads/index.html">clean_reads</a></li>
                    <li><a href="/sff_extract/index.html">sff_extract</a></li>
                    <li><a href="/psubprocess/index.html">psubprocess</a></li>-->
               </ul>
            </li>
            <li><a href="/courses.html">Courses</a>
                <ul>
                    <li><a href="/courses/intro_bioinf/index.html">Bionformática básica</a></li>
                    <li><a href="/courses/sequence_analysis/index.html">NGS sequence analysis</a></li>
                    <li><a href="/courses/linux/index.html">Linux y Python</a></li>
                    <li><a href="/courses/unix/index.html">Linux para Biologos</a></li>
               </ul>
        </ul>
    </nav>

    <main>
                
    
  <section id="introduction-to-emboss">
<h1>Introduction to EMBOSS<a class="headerlink" href="#introduction-to-emboss" title="Permalink to this headline">¶</a></h1>
<section id="emboss">
<h2>EMBOSS<a class="headerlink" href="#emboss" title="Permalink to this headline">¶</a></h2>
<p><a class="reference external" href="http://emboss.sourceforge.net/">EMBOSS</a> is a suite of programs developed to do molecular biology by EMBnet.
EMBOSS is <a class="reference external" href="https://en.wikipedia.org/wiki/Free_software">free software</a> and although it is primarily developed for Linux it also runs in MS Windows and Mac OS X.</p>
<p>EMBOSS is comprised by hunders of programs and includes utilities to align sequences, translate RNA to proteins, identify protien motifs, analyze repetitions, calculate codon usage statistics and many more analyses.</p>
<section id="running-emboss-commands">
<h3>Running EMBOSS commands<a class="headerlink" href="#running-emboss-commands" title="Permalink to this headline">¶</a></h3>
<p>The EMBOSS utilities have a <a class="reference external" href="https://en.wikipedia.org/wiki/Command-line_interface">Command Line Interface</a> and all can be run from a text terminal.
The <a class="reference external" href="http://emboss.sourceforge.net/apps/release/6.3/emboss/apps/groups.html">documentation</a> for the EMBOSS aplications is written taking into account mainly the Command Line Interface.</p>
<p>But there are other ways to run the EMBOSS software.</p>
<p>There are web <a class="reference external" href="http://emboss.sourceforge.net/servers/#explorer">servers</a> that provide the EMBOSS tools as a service.
In those servers the EMBOSS utilities can be accessed as web pages and the server runs the terminal command and return the result.</p>
<p>Another interface to EMBOSS is <a class="reference external" href="http://emboss.sourceforge.net/Jemboss/">Jemboss</a>.
Jemboss is a Graphical User Interface application, like any of the other common desktop applications.</p>
</section>
</section>
<section id="identify-the-command">
<h2>Identify the command<a class="headerlink" href="#identify-the-command" title="Permalink to this headline">¶</a></h2>
<p>When we do not know which of the EMBOSS commands to use we can use another EMBOSS command, <a class="reference external" href="http://emboss.sourceforge.net/apps/release/6.3/emboss/apps/wossname.html">wossname</a>, to search in the EMBOSS command list.</p>
<p>Get a list with all commands:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ wossname
</pre></div>
</div>
<p>Look for command to do alignments:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ wossname alignment
</pre></div>
</div>
</section>
<section id="getting-help-for-a-command">
<h2>Getting help for a command<a class="headerlink" href="#getting-help-for-a-command" title="Permalink to this headline">¶</a></h2>
<p>The <a class="reference external" href="http://emboss.sourceforge.net/apps/release/6.3/emboss/apps/groups.html">command</a> manuals can be found in the web.</p>
<p>They are also available in any EMBOSS installation by using the <a class="reference external" href="http://emboss.sourceforge.net/apps/release/6.3/emboss/apps/tfm.html">tfm</a> command.
For instance, to read the manual for the <a class="reference external" href="http://emboss.sourceforge.net/apps/release/6.3/emboss/apps/water.html">water</a> command we could run:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ tfm water
</pre></div>
</div>
<p>The manual reader is similar to the Unix less command.</p>
<p>It is also possible to get a brief list of parameters available with the option “-help” and the complete list with “-help -verbose”</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ water -help -verbose
</pre></div>
</div>
</section>
<section id="example-change-a-sequence-format">
<h2>Example: change a sequence format<a class="headerlink" href="#example-change-a-sequence-format" title="Permalink to this headline">¶</a></h2>
<p>To get familiar with the EMBOSS way of running the analyse we will change the format of a sequence file.
EMBOSS has a program to modify sequence files: <a class="reference external" href="http://emboss.sourceforge.net/apps/release/6.3/emboss/apps/seqret.html">seqret</a></p>
<p>seqret can change the sequence format, trim the sequences and reverse-complement them.
Let’s change the format of fasta <a class="reference external" href="../static/emboss/sequence.fasta">sequence file</a> to the GenBank format.</p>
<p>If we run seqset in the command line it will ask for the input file and for the output file:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ seqret
</pre></div>
</div>
<p>By default seqret won’t change the sequence format unless we specify the output format that we want.
To do it we have to write <em>genbank::secuencia.gb</em>.
We can also tell seqret which is our input file:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ seqret secuencia.fasta genbank::secuencia.gb
</pre></div>
</div>
<p>We could sent the result to standard output instead of a file.
This is a general feature of all EMBOSS programs.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ seqret secuencia.fasta genbank::stdout
</pre></div>
</div>
<p>To reverse and complement a sequence we can use the parameter -sreverse1:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ seqret secuencia.fasta embl::stdout -sreverse1
</pre></div>
</div>
<p>Would the result sequence be different in this case?</p>
<p>seqret can also trim the input sequence.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ seqret secuencia.fasta embl::stdout -sbegin1 10 -send1 30
</pre></div>
</div>
</section>
</section>



        

    </main>

</div>
    <footer>
            <a rel="license" href="https://creativecommons.org/licenses/by/4.0/"><img alt="Licencia de Creative Commons" style="border-width:0" src="https://i.creativecommons.org/l/by/4.0/88x31.png" /></a><br /><span xmlns:dct="https://purl.org/dc/terms/" property="dct:title">Bioinformatics at COMAV</span> is licensed under a <a rel="license" href="https://creativecommons.org/licenses/by/4.0/">Creative Commons Reconocimiento 4.0 Internacional License</a>.
    </footer>

<script type="text/javascript">
var gaJsHost = (("https:" == document.location.protocol) ? "https://ssl." : "http://www.");
document.write(unescape("%3Cscript src='" + gaJsHost + "google-analytics.com/ga.js' type='text/javascript'%3E%3C/script%3E"));
</script>
<script type="text/javascript">
try {
var pageTracker = _gat._getTracker("UA-16108274-1");
pageTracker._trackPageview();
} catch(err) {}</script>

  </body>
</html>