<!DOCTYPE html>
<html lang="en">

  <head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1">

  <title>Introduction to EMBOSS</title>
  <meta name="description" content="Introduction to Bioinformatics for the Biotechnology degree.">

  <link rel="stylesheet" href="/courses/biotech3/css/main.css">
  <link rel="canonical" href="/courses/biotech3/practice/emboss.html">
  <link rel="alternate" type="application/rss+xml" title="Bioinformatics" href="/courses/biotech3/feed.xml">
</head>


  <body>

    <header class="site-header" role="banner">

  <div class="wrapper">

    <a class="site-title" href="/courses/biotech3/">Bioinformatics</a>

    <nav class="site-nav">
      <a href="#" class="menu-icon">
        <svg viewBox="0 0 18 15">
          <path fill="#424242" d="M18,1.484c0,0.82-0.665,1.484-1.484,1.484H1.484C0.665,2.969,0,2.304,0,1.484l0,0C0,0.665,0.665,0,1.484,0 h15.031C17.335,0,18,0.665,18,1.484L18,1.484z"/>
          <path fill="#424242" d="M18,7.516C18,8.335,17.335,9,16.516,9H1.484C0.665,9,0,8.335,0,7.516l0,0c0-0.82,0.665-1.484,1.484-1.484 h15.031C17.335,6.031,18,6.696,18,7.516L18,7.516z"/>
          <path fill="#424242" d="M18,13.516C18,14.335,17.335,15,16.516,15H1.484C0.665,15,0,14.335,0,13.516l0,0 c0-0.82,0.665-1.484,1.484-1.484h15.031C17.335,12.031,18,12.696,18,13.516L18,13.516z"/>
        </svg>
      </a>

      <div class="trigger">
        
          
          <a class="page-link" href="/courses/biotech3/practice/alignment.html">Sequence Alignment Practice</a>
          
        
          
          <a class="page-link" href="/courses/biotech3/practice/assembly.html">Assembly</a>
          
        
          
          <a class="page-link" href="/courses/biotech3/theory/blast.html">Blast</a>
          
        
          
          <a class="page-link" href="/courses/biotech3/practice/blast.html">Blast Practice</a>
          
        
          
          <a class="page-link" href="/courses/biotech3/covid.html">COVID19</a>
          
        
          
          <a class="page-link" href="/courses/biotech3/theory/databases.html">Biological Databases</a>
          
        
          
          <a class="page-link" href="/courses/biotech3/practice/emboss.html">Introduction to EMBOSS</a>
          
        
          
          <a class="page-link" href="/courses/biotech3/practice/entrez.html">NCBI database search Practice</a>
          
        
          
        
          
          <a class="page-link" href="/courses/biotech3/">Home</a>
          
        
          
          <a class="page-link" href="/courses/biotech3/practice/linux.html">Unix CLI</a>
          
        
          
        
          
          <a class="page-link" href="/courses/biotech3/practice/multiple.html">Multiple Sequence Alignment Practice</a>
          
        
          
          <a class="page-link" href="/courses/biotech3/theory/multiple.html">Multiple Sequence Alignment</a>
          
        
          
          <a class="page-link" href="/courses/biotech3/norms.html">Norms</a>
          
        
          
          <a class="page-link" href="/courses/biotech3/practice/phylogeny.html">Phylogeny practice</a>
          
        
          
          <a class="page-link" href="/courses/biotech3/theory/phylogeny.html">Phylogeny and evolution</a>
          
        
          
          <a class="page-link" href="/courses/biotech3/prev_knowledge.html">Previous Knowledge</a>
          
        
          
          <a class="page-link" href="/courses/biotech3/theory/sequence_alignment.html">Sequence Alignment</a>
          
        
          
          <a class="page-link" href="/courses/biotech3/theory/structural_annotation.html">Structural Annotation</a>
          
        
          
          <a class="page-link" href="/courses/biotech3/practice/structural_annotation.html">Structural Annotation</a>
          
        
      </div>
    </nav>

  </div>

</header>


    <main class="page-content" aria-label="Content">
      <div class="wrapper">
        <article class="post">

  <header class="post-header">
    <h1 class="post-title">Introduction to EMBOSS</h1>
  </header>

  <div class="post-content">
    <h1 id="emboss">EMBOSS</h1>

<p><a href="http://emboss.sourceforge.net/">EMBOSS</a> is a suite of programs developed to do molecular biology by EMBnet.
EMBOSS is <a href="https://en.wikipedia.org/wiki/Free_software">free software</a> and although it is primarily developed for Linux it also runs in MS Windows and Mac OS X.</p>

<p>EMBOSS is comprised by hunders of programs and includes utilities to align sequences, translate RNA to proteins, identify protien motifs, analyze repetitions, calculate codon usage statistics and many more analyses.</p>

<h2 id="running-emboss-commands">Running EMBOSS commands</h2>

<p>The EMBOSS utilities have a <a href="https://en.wikipedia.org/wiki/Command-line_interface">Command Line Interface</a> and all can be run from a text terminal.
The <a href="http://emboss.sourceforge.net/apps/release/6.3/emboss/apps/groups.html">documentation</a> for the EMBOSS aplications is written taking into account mainly the Command Line Interface.</p>

<p>But there are other ways to run the EMBOSS software.</p>

<p>There are web <a href="http://emboss.sourceforge.net/servers/#explorer">servers</a> that provide the EMBOSS tools as a service.
In those servers the EMBOSS utilities can be accessed as web pages and the server runs the terminal command and return the result.</p>

<p>Another interface to EMBOSS is <a href="http://emboss.sourceforge.net/Jemboss/">Jemboss</a>.
Jemboss is a Graphical User Interface application, like any of the other common desktop applications.</p>

<h1 id="identify-the-command">Identify the command</h1>

<p>When we do not know which of the EMBOSS commands to use we can use another EMBOSS command, <a href="http://emboss.sourceforge.net/apps/release/6.3/emboss/apps/wossname.html">wossname</a>, to search in the EMBOSS command list.</p>

<p>Get a list with all commands:</p>

<div class="highlighter-rouge"><pre class="highlight"><code>$ wossname
</code></pre>
</div>

<p>Look for command to do alignments:</p>

<div class="highlighter-rouge"><pre class="highlight"><code>$ wossname alignment
</code></pre>
</div>

<h1 id="getting-help-for-a-command">Getting help for a command</h1>

<p>The <a href="http://emboss.sourceforge.net/apps/release/6.3/emboss/apps/groups.html">command</a> manuals can be found in the web.</p>

<p>They are also available in any EMBOSS installation by using the <a href="http://emboss.sourceforge.net/apps/release/6.3/emboss/apps/tfm.html">tfm</a> command.
For instance, to read the manual for the <a href="http://emboss.sourceforge.net/apps/release/6.3/emboss/apps/water.html">water</a> command we could run:</p>

<div class="highlighter-rouge"><pre class="highlight"><code>$ tfm water
</code></pre>
</div>

<p>The manual reader is similar to the Unix less command.</p>

<p>It is also possible to get a brief list of parameters available with the option “-help” and the complete list with “-help -verbose”</p>

<div class="highlighter-rouge"><pre class="highlight"><code>$ water -help -verbose
</code></pre>
</div>

<h1 id="example-change-a-sequence-format">Example: change a sequence format</h1>

<p>To get familiar with the EMBOSS way of running the analyse we will change the format of a sequence file.
EMBOSS has a program to modify sequence files: <a href="http://emboss.sourceforge.net/apps/release/6.3/emboss/apps/seqret.html">seqret</a></p>

<p>seqret can change the sequence format, trim the sequences and reverse-complement them.
Let’s change the format of fasta <a href="../static/emboss/sequence.fasta">sequence file</a> to the GenBank format.</p>

<p>If we run seqset in the command line it will ask for the input file and for the output file:</p>

<div class="highlighter-rouge"><pre class="highlight"><code>$ seqret
</code></pre>
</div>

<p>By default seqret won’t change the sequence format unless we specify the output format that we want.
To do it we have to write <em>genbank::secuencia.gb</em>.
We can also tell seqret which is our input file:</p>

<div class="highlighter-rouge"><pre class="highlight"><code>$ seqret secuencia.fasta genbank::secuencia.gb
</code></pre>
</div>

<p>We could sent the result to standard output instead of a file.
This is a general feature of all EMBOSS programs.</p>

<div class="highlighter-rouge"><pre class="highlight"><code>$ seqret secuencia.fasta genbank::stdout
</code></pre>
</div>

<p>To reverse and complement a sequence we can use the parameter -sreverse1:</p>

<div class="highlighter-rouge"><pre class="highlight"><code>$ seqret secuencia.fasta embl::stdout -sreverse1
</code></pre>
</div>

<p>Would the result sequence be different in this case?</p>

<p>seqret can also trim the input sequence.</p>

<div class="highlighter-rouge"><pre class="highlight"><code>$ seqret secuencia.fasta embl::stdout -sbegin1 10 -send1 30
</code></pre>
</div>

  </div>

</article>

      </div>
    </main>

    

  </body>

</html>
