<!DOCTYPE html>

<html lang="en" data-content_root="../">
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="viewport" content="width=device-width, initial-scale=1" />

    <title>NCBI database search &#8212; Bioinformatics at COMAV 0.1 documentation</title>
    <link rel="stylesheet" type="text/css" href="../_static/pygments.css?v=fa44fd50" />
    <link rel="stylesheet" type="text/css" href="../_static/alabaster.css?v=039e1c02" />
    <script src="../_static/documentation_options.js?v=2709fde1"></script>
    <script src="../_static/doctools.js?v=888ff710"></script>
    <script src="../_static/sphinx_highlight.js?v=dc90522c"></script>
    <link rel="icon" href="../_static/favicon.ico"/>
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Introduction to EMBOSS" href="emboss.html" />
    <link rel="prev" title="Sequence analysis" href="../theory/sequence_analysis.html" />
   
  <link rel="stylesheet" href="../_static/custom.css" type="text/css" />
  
  
  <meta name="viewport" content="width=device-width, initial-scale=0.9, maximum-scale=0.9" />

  </head><body>
  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          

          <div class="body" role="main">
            
  <section id="ncbi-database-search">
<h1>NCBI database search<a class="headerlink" href="#ncbi-database-search" title="Link to this heading">¶</a></h1>
<section id="id1">
<h2>NCBI database search<a class="headerlink" href="#id1" title="Link to this heading">¶</a></h2>
<p>This is a tutorial based on the <a class="reference external" href="http://www.ncbi.nlm.nih.gov/entrez/query/static/help/entrez_tutorial_BIB.pdf">NCBI’s search</a> tutorial.</p>
<p>We can use the <a class="reference external" href="https://www.ncbi.nlm.nih.gov/search">NCBI search</a> interface to access databases related, among other things, to:</p>
<ul class="simple">
<li><p>Bibliography</p></li>
<li><p>Nucleotide sequences</p></li>
<li><p>Protein sequences</p></li>
<li><p>Complete genomes</p></li>
<li><p>Protein structures</p></li>
</ul>
<p>To show how NCBI search works we are going to search for information about the gene MLH1, a human gene involved in human colon cancer.</p>
<p>We want to look for:</p>
<ul class="simple">
<li><p>mRNA that better represents the gene</p></li>
<li><p>Most relevant bibliographic references</p></li>
<li><p>Protein sequences</p></li>
<li><p>Conserved domains</p></li>
<li><p>Similar proteins</p></li>
<li><p>Mutations found in the gene</p></li>
<li><p>Protein structure.</p></li>
<li><p>Genome region in which the gene is located</p></li>
</ul>
<p>Go to the <a class="reference external" href="https://www.ncbi.nlm.nih.gov/search">NCBI search</a> site and look for the gene name: MLH1</p>
<p>You will get results for the search in many of the NCBI’s databases.</p>
<p>There are tools and filters to reduce the number of results obtained:</p>
<ul class="simple">
<li><p>You can look for results that have more than one search term. For instance, you can look for MLH1 and Human using <em>AND</em>. Look for “MLH1 AND Human”</p></li>
<li><p>You can also restrict the search to particular fields, otherwise the search terms can occur at any place in the found records. To restrict the search you have to write the field ID between square brackets. Try with “MLH1 AND Human[ORGN]”.
You can also build this complex queries in the advance query page of some of the databases, e.g. the <a class="reference external" href="http://www.ncbi.nlm.nih.gov/nuccore/advanced">advance query</a> page of the Nucleotide Database.</p></li>
</ul>
<p>Main GenBank fields.</p>
<table class="docutils align-default">
<thead>
<tr class="row-odd"><th class="head"><p>Field</p></th>
<th class="head"><p>Description</p></th>
<th class="head"><p>Search field</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>Locus name</p></td>
<td><p>Unique sequence name</p></td>
<td><p>[ACCN]</p></td>
</tr>
<tr class="row-odd"><td><p>Sequence length</p></td>
<td><p>Sequence Length</p></td>
<td><p>[SLEN]</p></td>
</tr>
<tr class="row-even"><td><p>Molecule Type</p></td>
<td><p>DNA, genomic, mRNA, etc.</p></td>
<td><p>[PROP]</p></td>
</tr>
<tr class="row-odd"><td><p>Genbank Division</p></td>
<td><p>Division for the sequence</p></td>
<td><p>[PROP]</p></td>
</tr>
<tr class="row-even"><td><p>Modification Date</p></td>
<td><p>Date for the last edit</p></td>
<td><p>[MDAT]</p></td>
</tr>
<tr class="row-odd"><td><p>Definition</p></td>
<td><p>Brief description</p></td>
<td><p>[TITL]</p></td>
</tr>
<tr class="row-even"><td><p>Accession</p></td>
<td><p>Unique accession ID. It does not changes with modifications</p></td>
<td><p>[ACCN]</p></td>
</tr>
<tr class="row-odd"><td><p>Version</p></td>
<td><p>Version number of the sequence</p></td>
<td><p>All fields</p></td>
</tr>
<tr class="row-even"><td><p>Keywords</p></td>
<td><p>keywords that describe the sequence</p></td>
<td><p>[KYWD]</p></td>
</tr>
<tr class="row-odd"><td><p>Source</p></td>
<td><p>Common name for the source species</p></td>
<td><p>[ORGN]</p></td>
</tr>
<tr class="row-even"><td><p>Organism</p></td>
<td><p>Oficial name for the source species</p></td>
<td><p>[ORGN]</p></td>
</tr>
<tr class="row-odd"><td><p>Reference</p></td>
<td><p>Related publications</p></td>
<td><p>[TITL] [AUTH] [JOUR]</p></td>
</tr>
<tr class="row-even"><td><p>Features</p></td>
<td><p>Regions of interest</p></td>
<td><p>[FKEY]</p></td>
</tr>
<tr class="row-odd"><td><p>CDS</p></td>
<td><p>Coding Sequence</p></td>
<td><p>[FKEY]</p></td>
</tr>
</tbody>
</table>
<p>We can go to the results in one of the databases, for instance Nucleotide, and evaluate the results.</p>
<p>In the database search result page you have aditional filters available in the left column and the search query used in the right column.
Also in the search column you can find your recent search history.</p>
<p>You can also search in the page for any of the databases without going back to the general search page.
For instance, you could search only in the Nucleotide database by searching in the <a class="reference external" href="http://www.ncbi.nlm.nih.gov/nuccore/">Nucleotide Database</a> page instead of in the general search page.</p>
</section>
<section id="mrna">
<h2>mRNA<a class="headerlink" href="#mrna" title="Link to this heading">¶</a></h2>
<p>There are different nucleotide databases in NCBI, but let’s do some searches in the <a class="reference external" href="http://www.ncbi.nlm.nih.gov/nuccore/">Nucleotide Database</a>.</p>
<ul class="simple">
<li><p>How many records do we get if we look for “colon cancer” in the Nucleotide Database?</p></li>
<li><p>How many records do you expect to get if you look for MLH1? How many do you get? Why?</p></li>
</ul>
<p>To look for the sequences that best represent the MLH1 gene it is better to look in the RefSeq database.
We can filter down the results that we got by selecting RefSeq in “Source Databases” in the left column.</p>
<p>There are still many results for MLH1 in RefSeq.
We can get less results if we restrict the search to the <em>Definition</em> field by doing the search “MLH1[TITL]”</p>
<p>We haven’t filter down the results to get only human results yet.
We could do it by using the search “MLH1[TITL] AND Human[ORGN]”, but we can also do it by going to the <a class="reference external" href="http://www.ncbi.nlm.nih.gov/nuccore/advanced">advanced query</a> page.
In the advance query form we can select the fields from drop down menus instead of using the square bracket syntax and also we can use a previous search and add terms to it by using the numbers displayed in the history table.
For isntance we could do something similar to “Human[ORGN] AND #1”</p>
<p>How many results do you get?</p>
</section>
<section id="bibliographic-references">
<h2>Bibliographic References<a class="headerlink" href="#bibliographic-references" title="Link to this heading">¶</a></h2>
<p>Once we have located the sequence of interest it is very easy to find related information in the search result page or in the page for any of the records by going to the section related information.
Thus we can look for the papers related to our search or to any of the sequence records.</p>
<p>In the Pubmed case we can differentiate the references that include the complete free text by looking for PMC.</p>
</section>
<section id="other-related-information">
<h2>Other related information<a class="headerlink" href="#other-related-information" title="Link to this heading">¶</a></h2>
<p>Most of NCBI’s databases are related in this way so we can go from sequences to papers, from papers to genes, from genes to structures, etc.</p>
<p>Look for the related proteins to the MLH1 human gene for its homologous genes.</p>
<p>In the page for any of its proteins you can look for conserved domains.</p>
<ul class="simple">
<li><p>Which are its domains?</p></li>
<li><p>Which are their functions?</p></li>
<li><p>Which other proteins share any of its domains?</p></li>
</ul>
</section>
<section id="blast-search">
<h2>Blast search<a class="headerlink" href="#blast-search" title="Link to this heading">¶</a></h2>
<p>Also in its protein page you can do a Blast search by going to the Blink and look for other proteins with similar sequences.</p>
<p>Within the Blink blast search we have Display Options that allow us to filter down the results and get only the results with a Swiss-Prot entry or the results with a PDB structure.</p>
</section>
<section id="gene">
<h2>Gene<a class="headerlink" href="#gene" title="Link to this heading">¶</a></h2>
<p>There is in NCBI a Gene database that collects all the information for the genes of some species.
You can access to the the gene page by looking in the gene database or by following a link in the related information section of any record in the other databases.</p>
</section>
<section id="mutations">
<h2>Mutations<a class="headerlink" href="#mutations" title="Link to this heading">¶</a></h2>
<p>In the gene page you can access to the related SNPs found in that gene in SNP Geneview.</p>
<p>In the Geneview page you can look for the SNPs with clinical information in Clinical Source.</p>
<ul class="simple">
<li><p>Are there any mutation with a related 3D structure?</p></li>
<li><p>Any mutation with links to OMIM? (Look in the column Clinically Associated)</p></li>
</ul>
</section>
<section id="genomic-region">
<h2>Genomic region<a class="headerlink" href="#genomic-region" title="Link to this heading">¶</a></h2>
<p>In the protein, mRNA or RefSeqGene page you have a Graphics link to show the record information in a genome graphical browser instead of in the Genbank format.</p>
<p>You can also get a representation of the genomic region using the related information link “Map viewer”.</p>
</section>
<section id="exercise-1">
<h2>Exercise 1<a class="headerlink" href="#exercise-1" title="Link to this heading">¶</a></h2>
<p>Look for the most relevant information for any of the human genes FXN, OCA2 or FOXP2.</p>
<ul class="simple">
<li><p>Molecular function</p></li>
<li><p>Related diseases</p></li>
<li><p>Genomic structure</p></li>
<li><p>Protein sequence</p></li>
<li><p>Homologous sequences in other species</p></li>
</ul>
</section>
<section id="exercise-2">
<h2>Exercise 2<a class="headerlink" href="#exercise-2" title="Link to this heading">¶</a></h2>
<p>Are there any genes described for the HYPERCHOLESTEROLEMIA?</p>
<p>Which would be the most convenient database to look for human diseases?</p>
</section>
<section id="exercise-3">
<h2>Exercise 3<a class="headerlink" href="#exercise-3" title="Link to this heading">¶</a></h2>
<p>Look for information related to the possible carcinogenic effects of: nicotine, milk, cannabis and banana.</p>
<p>Look for information in Google and Pubmed?</p>
<ul class="simple">
<li><p>How many references have you found in both searches? Are the references consistent? Can you get solid conclusion? Which evidences would you value the most?</p></li>
<li><p>Look in Pubmed, but only in the “meta-analysis” references. Which conclusions can you reach now?</p></li>
</ul>
</section>
</section>


          </div>
          
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
            <p class="logo"><a href="../index.html">
              <img class="logo" src="../_static/comav.png" alt="Logo"/>
            </a></p>
<h1 class="logo"><a href="../index.html">Bioinformatics at COMAV</a></h1>








<h3>Navigation</h3>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../norms.html">Norms</a></li>
<li class="toctree-l1"><a class="reference internal" href="../norms.html#tutoring">Tutoring</a></li>
<li class="toctree-l1"><a class="reference internal" href="../norms.html#id1">Norms</a></li>
</ul>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../prev_knowledge.html">Previous knowledge</a></li>
</ul>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../theory/databases.html">Databases</a></li>
<li class="toctree-l1"><a class="reference internal" href="../theory/sequence_alignment.html">Sequence alignment</a></li>
<li class="toctree-l1"><a class="reference internal" href="../theory/blast.html">Blast</a></li>
<li class="toctree-l1"><a class="reference internal" href="../theory/multiple.html">Multiple</a></li>
<li class="toctree-l1"><a class="reference internal" href="../theory/structural_annotation.html">Structural annotation</a></li>
<li class="toctree-l1"><a class="reference internal" href="../theory/phylogeny.html">Phylogeny</a></li>
<li class="toctree-l1"><a class="reference internal" href="../theory/sequence_analysis.html">Sequence analysis</a></li>
</ul>
<ul class="current">
<li class="toctree-l1 current"><a class="current reference internal" href="#">NCBI search</a></li>
<li class="toctree-l1"><a class="reference internal" href="emboss.html">Emboss</a></li>
<li class="toctree-l1"><a class="reference internal" href="alignment.html">Sequence Alignment</a></li>
<li class="toctree-l1"><a class="reference internal" href="blast.html">Blast</a></li>
<li class="toctree-l1"><a class="reference internal" href="multiple.html">Multiple Sequence Alignment</a></li>
<li class="toctree-l1"><a class="reference internal" href="structural_annotation.html">Structural annotation</a></li>
<li class="toctree-l1"><a class="reference internal" href="phylogeny.html">Phylogeny</a></li>
</ul>

<div class="relations">
<h3>Related Topics</h3>
<ul>
  <li><a href="../index.html">Documentation overview</a><ul>
      <li>Previous: <a href="../theory/sequence_analysis.html" title="previous chapter">Sequence analysis</a></li>
      <li>Next: <a href="emboss.html" title="next chapter">Introduction to EMBOSS</a></li>
  </ul></li>
</ul>
</div>
<div id="searchbox" style="display: none" role="search">
  <h3 id="searchlabel">Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="../search.html" method="get">
      <input type="text" name="q" aria-labelledby="searchlabel" autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false"/>
      <input type="submit" value="Go" />
    </form>
    </div>
</div>
<script>document.getElementById('searchbox').style.display = "block"</script>








        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="footer">
      &copy;2010, Jose Blanca & Joaquin Cañizares.
      
      |
      Powered by <a href="http://sphinx-doc.org/">Sphinx 7.2.6</a>
      &amp; <a href="https://github.com/bitprophet/alabaster">Alabaster 0.7.13</a>
      
    </div>

    

    
  </body>
</html>