
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    <meta name="generator" content="Docutils 0.18.1: http://docutils.sourceforge.net/" />

    <title>Multiple Sequence Alignment &mdash; Bioinformatics at COMAV 0.1 documentation</title>
    <link rel="stylesheet" href="../_static/style.css" type="text/css" />
    <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    <link rel="stylesheet" href="../_static/style.css" type="text/css" />
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    '../',
        VERSION:     '0.1',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="../_static/documentation_options.js"></script>
    <script type="text/javascript" src="../_static/jquery.js"></script>
    <script type="text/javascript" src="../_static/underscore.js"></script>
    <script type="text/javascript" src="../_static/_sphinx_javascript_frameworks_compat.js"></script>
    <script type="text/javascript" src="../_static/doctools.js"></script>
    <link rel="shortcut icon" href="../_static/favicon.ico"/>
    <link rel="search" title="Search" href="../search.html" />
    <link rel="top" title="Bioinformatics at COMAV 0.1 documentation" href="../index.html" />
    <link rel="next" title="Structural Annotation" href="structural_annotation.html" />
    <link rel="prev" title="Blast Practice" href="blast.html" /> 
  </head>
  <body>

<div class="body">
    <header>
        <h1><a href="index.html">Bioinformatics at COMAV</a></h1>
    </header>


    <nav class="horizontal_nav page_nav">
        <ul>
            <li><a href="/index.html">Bioinformatics & genomics</a>
                <ul>
                    <li><a href="/services.html">Services</a></li>
                    <li><a href="/people.html">People</a></li>
                    <li><a href="/publications.html">Publications</a></li>
                    <li><a href="/getting_here.html">Getting here</a></li>
<!--                    <li><a href="http://bioinf.comav.upv.es/notes/">Wiki</a></li>-->
               </ul>
            </li>
            <li><a href="/software.html">Software</a>
                <ul>
                    <li><a href="https://github.com/bioinfcomav/">Bioinf COMAV</a></li>
                    <li><a href="https://github.com/JoseBlanca/">Jose Blanca</a></li>
                    <!--
                    <li><a href="/seq_crumbs/index.html">seq_crumbs</a></li>
                    <li><a href="/ngs_backbone/index.html">ngs_backbone</a></li>
                    <li><a href="/clean_reads/index.html">clean_reads</a></li>
                    <li><a href="/sff_extract/index.html">sff_extract</a></li>
                    <li><a href="/psubprocess/index.html">psubprocess</a></li>-->
               </ul>
            </li>
            <li><a href="/courses.html">Courses</a>
                <ul>
                    <li><a href="/courses/intro_bioinf/index.html">Bionformática básica</a></li>
                    <li><a href="/courses/sequence_analysis/index.html">NGS sequence analysis</a></li>
                    <li><a href="/courses/linux/index.html">Linux y Python</a></li>
                    <li><a href="/courses/unix/index.html">Linux para Biologos</a></li>
               </ul>
        </ul>
    </nav>

    <main>
                
    
  <section id="multiple-sequence-alignment">
<h1>Multiple Sequence Alignment<a class="headerlink" href="#multiple-sequence-alignment" title="Permalink to this heading">¶</a></h1>
<section id="clustal-omega-protein-alignment">
<h2>Clustal Omega protein alignment<a class="headerlink" href="#clustal-omega-protein-alignment" title="Permalink to this heading">¶</a></h2>
<p>We are going to use the clustal omega program to analyze some homologous proteins to the human frataxin in <a class="reference download internal" download="" href="../_downloads/695321845f3acbd024acad468cf1edd6/frataxin_mamiferos.fasta"><code class="xref download docutils literal notranslate"><span class="pre">mammals</span></code></a>.</p>
<p>You can use clustal <a class="reference external" href="https://www.ebi.ac.uk/Tools/msa/clustalo/">omega</a> in the ebi server.
There are also on-line servers that create clustalw alignments and you can read its</p>
<p>Create an alignment check the result.
Is there any problem?</p>
<section id="what-would-happen-if-we-remove-the-most-dissimilar-sequences">
<h3>What would happen if we remove the most dissimilar sequences<a class="headerlink" href="#what-would-happen-if-we-remove-the-most-dissimilar-sequences" title="Permalink to this heading">¶</a></h3>
<p>Redo the analysis only with the <a class="reference download internal" download="" href="../_downloads/fcbee16caac02e50b56e9fb40a418669/frataxin_mamiferos_cercanos.fasta"><code class="xref download docutils literal notranslate"><span class="pre">closest</span> <span class="pre">mammals</span></code></a> and compare the result with the previous one.</p>
</section>
<section id="what-would-happen-with-more-dissimilar-sequences">
<h3>What would happen with more dissimilar sequences<a class="headerlink" href="#what-would-happen-with-more-dissimilar-sequences" title="Permalink to this heading">¶</a></h3>
<p>Try to do the alignment with more different <a class="reference download internal" download="" href="../_downloads/7ef0b6cf2089123525c930f52e9523c7/frataxin_prot.fasta"><code class="xref download docutils literal notranslate"><span class="pre">sequences</span></code></a>.</p>
</section>
<section id="t-coffee">
<h3>T-Coffee<a class="headerlink" href="#t-coffee" title="Permalink to this heading">¶</a></h3>
<p>Compare the result with the one given by <a class="reference external" href="http://tcoffee.crg.cat/apps/tcoffee/do:regular">T-Coffee</a></p>
<p>Once you have the result you can use the TCS tool in the same server to get an idea of which sites are reliable which are not.</p>
</section>
<section id="conserved-domains">
<h3>Conserved domains<a class="headerlink" href="#conserved-domains" title="Permalink to this heading">¶</a></h3>
<p>Compare the result with the <a class="reference external" href="https://www.ncbi.nlm.nih.gov/Structure/cdd/cddsrv.cgi?uid=29607">domain</a> stored in NCBI for the frataxin.</p>
<p>Is this domain similar to any of our alignments?</p>
</section>
</section>
<section id="dna-sequence-alignment">
<h2>DNA sequence alignment<a class="headerlink" href="#dna-sequence-alignment" title="Permalink to this heading">¶</a></h2>
<p>We can also align DNA sequences.
This is useful to do phylogenetic analysis and to look for mutations in the sequences.
We are going to look for the SNPs in the <a class="reference download internal" download="" href="../_downloads/42a4d4f309f3c79baf7ea0dfdf1d00c9/seq_dna.fasta"><code class="xref download docutils literal notranslate"><span class="pre">CT099</span></code></a> gene of <em>Lycopersicum pimpinellifolium</em>. Align them using <a class="reference external" href="http://www.ebi.ac.uk/Tools/msa/mafft/">MAFFT</a>.</p>
<p>Take a look at the guide tree. Is this a reliable phylogenetic tree?</p>
</section>
</section>



        

    </main>

</div>
    <footer>
            <a rel="license" href="https://creativecommons.org/licenses/by/4.0/"><img alt="Licencia de Creative Commons" style="border-width:0" src="https://i.creativecommons.org/l/by/4.0/88x31.png" /></a><br /><span xmlns:dct="https://purl.org/dc/terms/" property="dct:title">Bioinformatics at COMAV</span> is licensed under a <a rel="license" href="https://creativecommons.org/licenses/by/4.0/">Creative Commons Reconocimiento 4.0 Internacional License</a>.
    </footer>

<script type="text/javascript">
var gaJsHost = (("https:" == document.location.protocol) ? "https://ssl." : "http://www.");
document.write(unescape("%3Cscript src='" + gaJsHost + "google-analytics.com/ga.js' type='text/javascript'%3E%3C/script%3E"));
</script>
<script type="text/javascript">
try {
var pageTracker = _gat._getTracker("UA-16108274-1");
pageTracker._trackPageview();
} catch(err) {}</script>

  </body>
</html>