<!DOCTYPE html>

<html lang="en" data-content_root="../">
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="viewport" content="width=device-width, initial-scale=1" />

    <title>Phylogeny &#8212; Bioinformatics at COMAV 0.1 documentation</title>
    <link rel="stylesheet" type="text/css" href="../_static/pygments.css?v=fa44fd50" />
    <link rel="stylesheet" type="text/css" href="../_static/basic.css?v=c058f7c8" />
    <link rel="stylesheet" type="text/css" href="../_static/alabaster.css?v=27fed22d" />
    <script src="../_static/documentation_options.js?v=2709fde1"></script>
    <script src="../_static/doctools.js?v=9a2dae69"></script>
    <script src="../_static/sphinx_highlight.js?v=dc90522c"></script>
    <link rel="icon" href="../_static/favicon.ico"/>
    <link rel="search" title="Search" href="../search.html" />
    <link rel="prev" title="Structural Annotation" href="structural_annotation.html" />
   
  <link rel="stylesheet" href="../_static/custom.css" type="text/css" />
  

  
  

  </head><body>
  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          

          <div class="body" role="main">
            
  <section id="phylogeny">
<h1>Phylogeny<a class="headerlink" href="#phylogeny" title="Link to this heading">¶</a></h1>
<p>We are going to do different phylogenetic reconstructions using different methods.
We are going to use the software <a class="reference external" href="http://www.megasoftware.net/">Mega</a>.</p>
<section id="mega-tutorial">
<h2>Mega tutorial<a class="headerlink" href="#mega-tutorial" title="Link to this heading">¶</a></h2>
<p>Follow the Mega tutorial, especially the sections:</p>
<ul class="simple">
<li><p>Introduction</p></li>
<li><p>Mega Basics</p></li>
<li><p>Aligning sequences</p></li>
<li><p>Estimating Evolutionary Distances</p></li>
<li><p>Building Trees</p></li>
<li><p>Testing Tree Reliability</p></li>
<li><p>Constructing Likelihood Trees</p></li>
</ul>
</section>
<section id="distance-methods">
<h2>Distance methods<a class="headerlink" href="#distance-methods" title="Link to this heading">¶</a></h2>
<p>You have a <a class="reference download internal" download="" href="../_downloads/1c9236eff9ae08514e5f60d4446e7157/aprot.fasta"><code class="xref download docutils literal notranslate"><span class="pre">multiple</span> <span class="pre">sequence</span> <span class="pre">alignment</span></code></a> alignment.
Generate two trees without bootstrap (set “Test of Phylogeny” as None): one using neighbor-joining and another using UPGMA.</p>
<ul class="simple">
<li><p>Visualize the trees obtained.</p></li>
<li><p>Are they the same?</p></li>
<li><p>Which clades do they share? You can change the order in which the branches are drawn without changing their topologies to be able to compare the trees.</p></li>
<li><p>Which tree is better</p></li>
</ul>
<p>Use this other <a class="reference download internal" download="" href="../_downloads/7c55cdd19432ffcc143054362e72cebc/hprot.fasta"><code class="xref download docutils literal notranslate"><span class="pre">multiple</span> <span class="pre">sequence</span> <span class="pre">alignment</span></code></a> to generate a new neighbor-joining tree without bootstrap.</p>
<ul class="simple">
<li><p>Is this tree informative? Why?</p></li>
</ul>
</section>
<section id="bootstrap">
<h2>Bootstrap<a class="headerlink" href="#bootstrap" title="Link to this heading">¶</a></h2>
<p>Redo the trees from the previous exercise doing a boostratp with 100 replicates.</p>
<p>Do you get the same topologies as in the previous exercise?</p>
<p>Which are the significative clades?</p>
<p>What remains if we collapse all clades that are not significative?</p>
</section>
<section id="frataxin-phylogenies">
<h2>Frataxin phylogenies<a class="headerlink" href="#frataxin-phylogenies" title="Link to this heading">¶</a></h2>
<p>We are going to use a multiple sequence alignment of the <a class="reference download internal" download="" href="../_downloads/69f7a2642a81e7cea7fd48e82d9cf186/frataxin_mod.meg"><code class="xref download docutils literal notranslate"><span class="pre">frataxin</span></code></a> protein.
This is an alignment in MEGA format that can be loaded directly into MEGA.
It includes only the conserved regions without the signal peptide.</p>
<section id="distance-trees">
<h3>Distance trees<a class="headerlink" href="#distance-trees" title="Link to this heading">¶</a></h3>
<p>Do the neighbor-joining and UPGMA trees with boostrap</p>
</section>
<section id="maximum-likelihood">
<h3>Maximum likelihood<a class="headerlink" href="#maximum-likelihood" title="Link to this heading">¶</a></h3>
<p>Do the tree with boostrap</p>
</section>
<section id="compare-the-trees">
<h3>Compare the trees<a class="headerlink" href="#compare-the-trees" title="Link to this heading">¶</a></h3>
<p>Are the trees identical?</p>
<p>Are there any clades not found in both trees?</p>
<p>Does any clade with enough support differ between both trees?</p>
<p>Are your trees the same as the ones that the other students have created?</p>
</section>
</section>
<section id="phylogenies-of-the-old-world-monkeys">
<h2>Phylogenies of the old world monkeys<a class="headerlink" href="#phylogenies-of-the-old-world-monkeys" title="Link to this heading">¶</a></h2>
<p>Create a phylogeny using parsimony, neighbor-joining and maximum likelihood suing the <a class="reference download internal" download="" href="../_downloads/1be8a393a9d41f79928d1bd13e0035cb/hepcidina.fasta"><code class="xref download docutils literal notranslate"><span class="pre">hepcidine</span></code></a> sequences.</p>
<p>Compare the trees obtained by the different methods.</p>
</section>
<section id="solanum-and-lycopersicum">
<h2>Solanum and Lycopersicum<a class="headerlink" href="#solanum-and-lycopersicum" title="Link to this heading">¶</a></h2>
<p>NCBI holds <a class="reference external" href="http://www.ncbi.nlm.nih.gov/popset">Popset</a>, a database of phylogenetic and population studies.
In Popset you can find sequences used in studies carried on in different species so you can repeat those studies with the same sequences and others.</p>
<p>Get from the database the sequences for the marker <a class="reference external" href="http://www.ncbi.nlm.nih.gov/popset/224980415">COS-At5g14320</a> used in the study: “Do potatoes and tomatoes have a single evolutionary history, and what proportion of the genome supports this history?”.Rodriguez,F., Wu,F., Ane,C., Tanksley,S. and Spooner,D.M.(2009) BMC Evol. Biol. 9:191.</p>
<p>Does the tomato belong to the <em>Solanum</em> genus as the potato does?</p>
</section>
<section id="human-hemoglobins">
<h2>Human hemoglobins<a class="headerlink" href="#human-hemoglobins" title="Link to this heading">¶</a></h2>
<p>Reconstruct a phylogeny for the human hemoglobins.</p>
<p>Starting with the sequence of the <a class="reference download internal" download="" href="../_downloads/a8f1a5a255ecccbe51b7ec2714e0df86/beta.fasta"><code class="xref download docutils literal notranslate"><span class="pre">beta</span></code></a> subunit of the human hemoglobin identify using blast all the human subunits of the human hemoglobin and create its phylogeny to identify the duplication events that led to all of them.
Use the mioglobin as an outgroup.</p>
</section>
</section>


          </div>
          
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="Main">
        <div class="sphinxsidebarwrapper">
            <p class="logo"><a href="../index.html">
              <img class="logo" src="../_static/comav.png" alt="Logo of Bioinformatics at COMAV"/>
            </a></p>
<h1 class="logo"><a href="../index.html">Bioinformatics at COMAV</a></h1>









<search id="searchbox" style="display: none" role="search">
    <div class="searchformwrapper">
    <form class="search" action="../search.html" method="get">
      <input type="text" name="q" aria-labelledby="searchlabel" autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" placeholder="Search"/>
      <input type="submit" value="Go" />
    </form>
    </div>
</search>
<script>document.getElementById('searchbox').style.display = "block"</script><h3>Navigation</h3>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../norms.html">Norms</a></li>
<li class="toctree-l1"><a class="reference internal" href="../norms.html#tutoring">Tutoring</a></li>
<li class="toctree-l1"><a class="reference internal" href="../norms.html#id1">Norms</a></li>
</ul>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../prev_knowledge.html">Previous knowledge</a></li>
</ul>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../theory/databases.html">Databases</a></li>
<li class="toctree-l1"><a class="reference internal" href="../theory/sequence_alignment.html">Sequence alignment</a></li>
<li class="toctree-l1"><a class="reference internal" href="../theory/blast.html">Blast</a></li>
<li class="toctree-l1"><a class="reference internal" href="../theory/multiple.html">Multiple</a></li>
<li class="toctree-l1"><a class="reference internal" href="../theory/structural_annotation.html">Structural annotation</a></li>
<li class="toctree-l1"><a class="reference internal" href="../theory/phylogeny.html">Phylogeny</a></li>
<li class="toctree-l1"><a class="reference internal" href="../theory/sequence_analysis.html">Sequence analysis</a></li>
</ul>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="entrez.html">NCBI search</a></li>
<li class="toctree-l1"><a class="reference internal" href="emboss.html">Emboss</a></li>
<li class="toctree-l1"><a class="reference internal" href="alignment.html">Sequence Alignment</a></li>
<li class="toctree-l1"><a class="reference internal" href="blast.html">Blast</a></li>
<li class="toctree-l1"><a class="reference internal" href="multiple.html">Multiple Sequence Alignment</a></li>
<li class="toctree-l1"><a class="reference internal" href="structural_annotation.html">Structural annotation</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Phylogeny</a></li>
</ul>

<div class="relations">
<h3>Related Topics</h3>
<ul>
  <li><a href="../index.html">Documentation overview</a><ul>
      <li>Previous: <a href="structural_annotation.html" title="previous chapter">Structural Annotation</a></li>
  </ul></li>
</ul>
</div>








        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="footer">
      &#169;2010, Jose Blanca & Joaquin Cañizares.
      
      |
      Powered by <a href="https://www.sphinx-doc.org/">Sphinx 8.0.2</a>
      &amp; <a href="https://alabaster.readthedocs.io">Alabaster 1.0.0</a>
      
    </div>

    

    
  </body>
</html>