<!DOCTYPE html>
<html lang="en">

  <head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1">

  <title>Structural Annotation</title>
  <meta name="description" content="Introduction to Bioinformatics for the Biotechnology degree.">

  <link rel="stylesheet" href=".//css/main.css">
  <link rel="canonical" href=".//practice/structural_annotation.html">
  <link rel="alternate" type="application/rss+xml" title="Bioinformatics" href=".//feed.xml">
</head>


  <body>

    <header class="site-header" role="banner">

  <div class="wrapper">

    <a class="site-title" href=".//">Bioinformatics</a>

    <nav class="site-nav">
      <a href="#" class="menu-icon">
        <svg viewBox="0 0 18 15">
          <path fill="#424242" d="M18,1.484c0,0.82-0.665,1.484-1.484,1.484H1.484C0.665,2.969,0,2.304,0,1.484l0,0C0,0.665,0.665,0,1.484,0 h15.031C17.335,0,18,0.665,18,1.484L18,1.484z"/>
          <path fill="#424242" d="M18,7.516C18,8.335,17.335,9,16.516,9H1.484C0.665,9,0,8.335,0,7.516l0,0c0-0.82,0.665-1.484,1.484-1.484 h15.031C17.335,6.031,18,6.696,18,7.516L18,7.516z"/>
          <path fill="#424242" d="M18,13.516C18,14.335,17.335,15,16.516,15H1.484C0.665,15,0,14.335,0,13.516l0,0 c0-0.82,0.665-1.484,1.484-1.484h15.031C17.335,12.031,18,12.696,18,13.516L18,13.516z"/>
        </svg>
      </a>

      <div class="trigger">
        
          
          <a class="page-link" href=".//practice/alignment.html">Sequence Alignment Practice</a>
          
        
          
          <a class="page-link" href=".//practice/assembly.html">Assembly</a>
          
        
          
          <a class="page-link" href=".//theory/blast.html">Blast</a>
          
        
          
          <a class="page-link" href=".//practice/blast.html">Blast Practice</a>
          
        
          
          <a class="page-link" href=".//covid.html">COVID19</a>
          
        
          
          <a class="page-link" href=".//theory/databases.html">Biological Databases</a>
          
        
          
          <a class="page-link" href=".//practice/emboss.html">Introduction to EMBOSS</a>
          
        
          
          <a class="page-link" href=".//practice/entrez.html">NCBI database search Practice</a>
          
        
          
        
          
          <a class="page-link" href=".//">Home</a>
          
        
          
          <a class="page-link" href=".//practice/linux.html">Unix CLI</a>
          
        
          
        
          
          <a class="page-link" href=".//practice/multiple.html">Multiple Sequence Alignment Practice</a>
          
        
          
          <a class="page-link" href=".//theory/multiple.html">Multiple Sequence Alignment</a>
          
        
          
          <a class="page-link" href=".//norms.html">Norms</a>
          
        
          
          <a class="page-link" href=".//practice/phylogeny.html">Phylogeny practice</a>
          
        
          
          <a class="page-link" href=".//theory/phylogeny.html">Phylogeny and evolution</a>
          
        
          
          <a class="page-link" href=".//prev_knowledge.html">Previous Knowledge</a>
          
        
          
          <a class="page-link" href=".//theory/sequence_alignment.html">Sequence Alignment</a>
          
        
          
          <a class="page-link" href=".//theory/structural_annotation.html">Structural Annotation</a>
          
        
          
          <a class="page-link" href=".//practice/structural_annotation.html">Structural Annotation</a>
          
        
      </div>
    </nav>

  </div>

</header>


    <main class="page-content" aria-label="Content">
      <div class="wrapper">
        <article class="post">

  <header class="post-header">
    <h1 class="post-title">Structural Annotation</h1>
  </header>

  <div class="post-content">
    <h1 id="graphic-detection-of-orfs">Graphic detection of ORFs</h1>

<p>Long ORFs are a hint of a CDS.
ORFs can be plot using the EMBOSS program <a href="http://emboss.sourceforge.net/apps/release/6.6/emboss/apps/plotorf.html">plotorf</a>.</p>

<p>Detect the ORFs in the <a href="../static/annotation/genomic_frataxin.fasta">genomic</a> and <a href="../static/annotation/mRNA_frataxin.fasta">mRNA</a> of the frataxin gene.</p>

<div class="highlighter-rouge"><pre class="highlight"><code>$ plotorf 'sequence'
</code></pre>
</div>

<h1 id="orf-translation">ORF translation</h1>

<p>Translate the largest ORF of the frataxin <a href="../static/annotation/mRNA_frataxin.fasta">mRNA</a>.
<a href="http://emboss.sourceforge.net/apps/release/6.6/emboss/apps/transeq.html">transeq</a> from EMBOSS can be used to translate the ORFs.
You have to provide to transeq the frame to translate.</p>

<div class="highlighter-rouge"><pre class="highlight"><code>$ transeq -frame 'x' 'sequence'
</code></pre>
</div>

<p>transeq would have translate the whole sequence without taking into account where does the start and stop codons are.
Edit the translated peptide to keep only the aminoacid sequence between the start and stop codons.</p>

<p>How can you be sure which one is the start methionine?
One way to have an idea is to compare the translated proteion with known proteins and compare the structure of the proteins.
Do a blastp in NCBI aginst the nr database.
Determine which should be the sequence of the translated protein taking into account the blast result.</p>

<h1 id="detection-of-the-intron-exon-and-cds-in-a-genomic-sequence">Detection of the intron, exon and CDS in a genomic sequence</h1>

<p>Determine the exons and introns in a genomic sequence by using two kinds of evidences: comparisons with known proteins and comparisons with ESTs.</p>

<h2 id="comparison-with-known-proteins">Comparison with known proteins</h2>

<p>Do a blastx aginst refseq with the <a href="../static/annotation/genomic_frataxin.fasta">genomic</a> sequence in NCBI.</p>

<p>Could you propose a structure for the gene based on the blast matches obtained?</p>

<h2 id="comparison-with-ests">Comparison with ESTs</h2>

<p><a href="https://en.wikipedia.org/wiki/Expressed_sequence_tag">ESTs</a> are sequences that originate from cDNAs.
We have two ESTs: <a href="../static/annotation/est1.fasta">EST1</a> and <a href="../static/annotation/est2.fasta">EST2</a>.
Align them to the genomic sequence and propose a structure for the frataxin gene.</p>

<p>Do the evidences from the blast and mRNA analyses match?</p>

<h2 id="ab-initio-prediction-of-the-gene-structure">Ab initio prediction of the gene structure</h2>

<p>Codon usage, ORFs and splice sites can be used to infer a structure for the gene.
Use <a href="http://bioinf.uni-greifswald.de/augustus/submission.php">Augustus</a> for the practice and run it with the <a href="../static/annotation/genomic_frataxin.fasta">genomic</a>.</p>

<p>The Augustus predictions can be enhanced by providing Augustus ESTs and cDNAs sequences.</p>

<p>Analyze the Augustus results and compare them with our predictions.</p>

<p>Do a blastp with the protein predicted by Augustus and check if the prediction looks right.</p>

<h2 id="a-possible-sequencing-error">A possible sequencing error</h2>

<p>Redo the practice with a new sequence for the <a href="../static/annotation/genomic_frataxin2.fasta">genomic</a> region.</p>

<p>Which are the differences between this sequence and the previous one?</p>

<p>Which looks better according to the CDS prediction?</p>

<h1 id="gene-structure-predition">Gene Structure predition</h1>

<h2 id="1">1</h2>

<p>Propose a structure for the gene or genes and the mRNA or mRNAs, including CDSs. The cDNA library was not directional, the sequenced strand might be the sense or the antisense one. A blastx was carried out against the swissprot database with the genomic sequence.</p>

<div class="highlighter-rouge"><pre class="highlight"><code>Genomic xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx

EST1    TT--------------              -----&gt;
EST2                                  &lt;---------      ----------
EST3          &lt;---------                              -----
EST4                 ---                              --------&gt;

blastx hit 1         &lt;--              -----           ----
blastx hit 2     &lt;------                              ----
</code></pre>
</div>

<h2 id="2">2</h2>

<p>Propose a structure for structure of the gene or genes and the mRNA or mRNAs. The cDNA library was not directional, the sequenced strand might be the sense or the antisense one.</p>

<div class="highlighter-rouge"><pre class="highlight"><code>Genomic xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx

EST1    ----------   ---&gt;
EST2                                    ---------    ----------&gt;
EST3                &lt;----
EST4                                         ----    ----&gt;
EST5         -----  -----    -&gt;
EST6                                       &lt;-----         -----
</code></pre>
</div>

<h2 id="3">3</h2>

<p>Propose a structure for structure of the gene or genes and the mRNA or mRNAs, including CDSs. The cDNA library was directional, the sequenced strand is the sense one. A blastx was carried out against the swissprot database with the genomic sequence.</p>

<div class="highlighter-rouge"><pre class="highlight"><code>Genomic xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx

EST1    -----------------           ---------------&gt;
EST2                                    &lt;----------------------- 
EST3                -----           ---------&gt;

blastx     --------------           ---------&gt;
</code></pre>
</div>

<h2 id="4">4</h2>

<p>Propose a structure for structure of the gene or genes and the mRNA or mRNAs, including CDSs. The cDNA library was not directional, the sequenced strand might be the sense or the antisense one. A blastx was carried out against the swissprot database with the genomic sequence.</p>

<div class="highlighter-rouge"><pre class="highlight"><code>Genomic xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx

EST1    -----------------&gt;
EST2                -----    ----&gt;

blast hit1 -------------&gt;
blast hit2                                         &lt;------------- 

stop f1         *          *         *               *     *
stop f2   *              *  *          *         *      *     *
stop f3        *              *  *          *         *      *
stop r1    *              *  *          *         *      *     *
stop r2             *         *      *     * *              *    * 
stop r3    *              *  *                         *     *  * 
</code></pre>
</div>

<h2 id="5">5</h2>

<p>Propose a structure for structure of the gene or genes and the mRNA or mRNAs, including CDSs. The cDNA library was not directional, the sequenced strand might be the sense or the antisense one.  Probabilities, based on the codon usage were determined for the forward and reverse strands. The reversed ones were low and are not represented.</p>

<div class="highlighter-rouge"><pre class="highlight"><code>Genomic xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx

EST1         -----------------&gt;
EST3                     -----    ------------&gt;
EST3                                     ----------------&gt;

stop f1         *               *         *         *      *     *
stop f2     *              *  *          *     *       *
stop f3           *            *                       *      *

Prob codon f1   ---------------         --                 --
Prob codon f2             ---             ---
Prob codon f3       --            --------------------- 
</code></pre>
</div>

<h2 id="6">6</h2>

<p>Propose a structure for structure of the gene or genes and the mRNA or mRNAs, including CDSs. The cDNA library was directional, the sequenced strand is the sense one. A blastx was carried out against the swissprot database with the genomic sequence.</p>

<div class="highlighter-rouge"><pre class="highlight"><code>Genomic    xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx

EST1       &lt;----
EST2       &lt;------
EST3                                   ---------&gt;
EST4       &lt;---
EST4       &lt;----

blastx                              --------------------&gt;

Prob codon f1   -- --   --            --                 --
Prob codon f2             ---             ---
Prob codon f3       --            --------------------- 
</code></pre>
</div>


  </div>

</article>

      </div>
    </main>

    

  </body>

</html>
