
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    <meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

    <title>Maize SNP calling walkthrough &mdash; Bioinformatics at COMAV 0.1 documentation</title>
    <link rel="stylesheet" href="_static/style.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <link rel="stylesheet" href="_static/style.css" type="text/css" />
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    './',
        VERSION:     '0.1',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="_static/documentation_options.js"></script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <link rel="shortcut icon" href="_static/favicon.ico"/>
    <link rel="search" title="Search" href="search.html" />
    <link rel="top" title="Bioinformatics at COMAV 0.1 documentation" href="index.html" />
    <link rel="next" title="Practical tasks" href="final_practice.html" />
    <link rel="prev" title="BEDTools" href="bedtools.html" /> 
  </head>
  <body>

<div class="body">
    <header>
        <h1><a href="index.html">Bioinformatics at COMAV</a></h1>
    </header>


    <nav class="horizontal_nav page_nav">
        <ul>
            <li><a href="/index.html">Bioinformatics & genomics</a>
                <ul>
                    <li><a href="/services.html">Services</a></li>
                    <li><a href="/people.html">People</a></li>
                    <li><a href="/publications.html">Publications</a></li>
                    <li><a href="/getting_here.html">Getting here</a></li>
<!--                    <li><a href="http://bioinf.comav.upv.es/notes/">Wiki</a></li>-->
               </ul>
            </li>
            <li><a href="/software.html">Software</a>
                <ul>

                    <li><a href="https://github.com/JoseBlanca/">Git</a></li><!--
                    <li><a href="/seq_crumbs/index.html">seq_crumbs</a></li>
                    <li><a href="/ngs_backbone/index.html">ngs_backbone</a></li>
                    <li><a href="/clean_reads/index.html">clean_reads</a></li>
                    <li><a href="/sff_extract/index.html">sff_extract</a></li>
                    <li><a href="/psubprocess/index.html">psubprocess</a></li>-->
               </ul>
            </li>
            <li><a href="/databases.html">Databases</a>
                <ul>
                    <li><a href="https://melogene.upv.es">Melogene</a></li>
                    <li><a href="https://cucurbigene.upv.es">Cucurbigene</a></li>
               </ul>
            </li>
            <li><a ref=''>Downloads</a>
                <ul>
                    <li><a href="/downloads/zucchini">Zucchini</a></li>
               </ul>
            </li>
            <li><a href="/courses.html">Courses</a>
                <ul>
                    <li><a href="/courses/intro_bioinf/index.html">Bionformática básica</a></li>
                    <li><a href="/courses/sequence_analysis/index.html">NGS sequence analysis</a></li>
                    <li><a href="/courses/linux/index.html">Linux y Python</a></li>
                    <li><a href="/courses/unix/index.html">Linux para Biologos</a></li>
               </ul>
        </ul>
    </nav>

    <main>
                <nav class="vertical_nav page_nav">
                    <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="sequencing_technologies.html">Sequencing technologies</a></li>
<li class="toctree-l1"><a class="reference internal" href="ngs_applications_examples.html">NGS application examples</a></li>
<li class="toctree-l1"><a class="reference internal" href="sequence_analysis_software.html">NGS analysis software</a></li>
<li class="toctree-l1"><a class="reference internal" href="cmd_primer.html">Command line primer</a></li>
<li class="toctree-l1"><a class="reference internal" href="sequence_file_formats.html">NGS sequence file formats</a></li>
<li class="toctree-l1"><a class="reference internal" href="read_cleaning.html">NGS read cleaning</a></li>
<li class="toctree-l1"><a class="reference internal" href="assembly.html">Sequence assembly</a></li>
<li class="toctree-l1"><a class="reference internal" href="mapping.html">Sequence mapping</a></li>
<li class="toctree-l1"><a class="reference internal" href="snp_calling.html">SNP calling and filtering</a></li>
<li class="toctree-l1"><a class="reference internal" href="rnaseq.html">NGS expression analysis</a></li>
<li class="toctree-l1"><a class="reference internal" href="rnaseq.html#practical-tasks">Practical tasks</a></li>
<li class="toctree-l1"><a class="reference internal" href="text_files.html">Dealing with big result files</a></li>
<li class="toctree-l1"><a class="reference internal" href="bedtools.html">BEDTools</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Maize SNP calling practice</a></li>
<li class="toctree-l1"><a class="reference internal" href="final_practice.html">Final practice</a></li>
<li class="toctree-l1"><a class="reference internal" href="final_practice_walkthrough.html">Final practice walkthrough</a></li>
</ul>

                </nav>
                
    
  <section id="maize-snp-calling-walkthrough">
<h1>Maize SNP calling walkthrough<a class="headerlink" href="#maize-snp-calling-walkthrough" title="Permalink to this headline">¶</a></h1>
<p>We have some 454 and Illumina maize reads and we want to look for SNPs in the the maize transcriptome.
We are going to write down the commands to do the complete practice to document one possible route to obtain the SNPs, but other approaches could be used.</p>
<p>Download the sequence files <a class="reference download internal" download="" href="_downloads/947c0cfdf67128981e9c4081f5b0f059/snp_calling_practice.tar.gz"><code class="xref download docutils literal notranslate"><span class="pre">sequence</span> <span class="pre">files</span></code></a> into a directory named maize.</p>
<section id="read-quality-evaluation">
<h2>Read quality evaluation<a class="headerlink" href="#read-quality-evaluation" title="Permalink to this headline">¶</a></h2>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>ngs_user@machine:~$ cd
ngs_user@machine:~$ mkdir maize
ngs_user@machine:~$ cd maize/
ngs_user@machine:~/maize$ ls
snp_calling_practice.tar.gz
ngs_user@machine:~/maize$ tar -xvzf snp_calling_practice.tar.gz
snp_calling/
snp_calling/maize_unigenes.fasta
snp_calling/sm_B73XMo17.lb_MB_14day.pl_illumina.sfastq
snp_calling/sm_Mo17.lb_Mo17_root.pl_454.sfastq
ngs_user@machine:~/maize$ mkdir raw_reads
ngs_user@machine:~/maize$ mv snp_calling/sm_Mo17.lb_Mo17_root.pl_454.sfastq raw_reads/mo17.454.fastq
ngs_user@machine:~/maize$ mv snp_calling/sm_B73XMo17.lb_MB_14day.pl_illumina.sfastq raw_reads/b73xmo17.illumina.fastq
</pre></div>
</div>
<p>Evaluate the quality with using fastqc or calculate_stats from seq_crumbs.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>ngs_user@machine:~/maize$ calculate_stats raw_reads/mo17.454.fastq | head -n 40
Length stats and distribution.
------------------------------
minimum: 45
maximum: 310
average: 211.73
num. seqs.: 4888
Quality stats and distribution.
-------------------------------
Q20: 93.35
Q30: 79.86
average: 32.53
ngs_user@machine:~/maize$ calculate_stats raw_reads/b73xmo17.illumina.fastq | head -n 40
Length stats and distribution.
------------------------------
minimum: 35
maximum: 35
average: 35.00
num. seqs.: 16084
Quality stats and distribution.
-------------------------------
Q20: 93.38
Q30: 68.61
average: 29.47
</pre></div>
</div>
</section>
<section id="read-cleaning">
<h2>Read cleaning<a class="headerlink" href="#read-cleaning" title="Permalink to this headline">¶</a></h2>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>ngs_user@machine:~/maize$ mkdir clean_reads
ngs_user@machine:~/maize$ trim_quality raw_reads/mo17.454.fastq | filter_by_length -n 150 &gt; clean_reads/mo17.454.fastq
ngs_user@machine:~/maize$ trim_quality -q 20 raw_reads/b73xmo17.illumina.fastq | filter_by_length -n 30 &gt; clean_reads/b73xmo17.illumina.fastq
ngs_user@machine:~/maize$ ls clean_reads
b73xmo17.illumina.fastq  mo17.454.fastq
</pre></div>
</div>
<p>Evaluate the result.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>ngs_user@machine:~/maize$ calculate_stats clean_reads/mo17.454.fastq | head -n 20
Length stats and distribution.
------------------------------
minimum: 150
maximum: 303
average: 226.81
num. seqs.: 2004
Quality stats and distribution.
-------------------------------
Q20: 99.81
Q30: 96.59
average: 35.95
ngs_user@machine:~/maize$ calculate_stats clean_reads/b73xmo17.illumina.fastq | head -n 40
Length stats and distribution.
------------------------------
minimum: 30
maximum: 35
average: 34.70
num. seqs.: 14415
Quality stats and distribution.
-------------------------------
Q20: 95.71
Q30: 72.56
average: 30.18
</pre></div>
</div>
</section>
<section id="mapping-to-the-reference">
<h2>Mapping to the reference<a class="headerlink" href="#mapping-to-the-reference" title="Permalink to this headline">¶</a></h2>
<p>We are going to map the 454 reads with bwasw and the Illumina reads with Illumina.</p>
<section id="bwasw-mapping">
<h3>bwasw mapping<a class="headerlink" href="#bwasw-mapping" title="Permalink to this headline">¶</a></h3>
<p>We create the reference:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>ngs_user@machine:~/maize$ mkdir mapping
ngs_user@machine:~/maize$ cp snp_calling/maize_unigenes.fasta mapping/reference.fasta
ngs_user@machine:~/maize$ cd mapping
ngs_user@machine:~/maize/mapping$ bwa index -p reference -a bwtsw reference.fasta
[bwa_index] Pack FASTA... 0.02 sec
[bwa_index] Construct BWT for the packed sequence...
[main] Real time: 0.892 sec; CPU: 0.548 sec
</pre></div>
</div>
<p>Now we can map:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>ngs_user@machine:~/maize/mapping$ bwa bwasw reference ../clean_reads/mo17.454.fastq &gt; mapped_454.sam
[bsw2_aln] read 2004 sequences/pairs (454536 bp) ...
[main] Version: 0.7.2-r351
[main] CMD: bwa bwasw reference ../clean_reads/mo17.454.fastq
[main] Real time: 3.163 sec; CPU: 2.672 sec
ngs_user@machine:~/maize/mapping$ samtools view -b -S -o mapped_454.bam mapped_454.sam
[samopen] SAM header is present: 100 sequences.
ngs_user@machine:~/maize/mapping$ rm mapped_454.sam
ngs_user@machine:~/maize/mapping$ ls mapped_454.bam
mapped_454.bam
</pre></div>
</div>
</section>
<section id="bowtie2-mapping">
<h3>bowtie2 mapping<a class="headerlink" href="#bowtie2-mapping" title="Permalink to this headline">¶</a></h3>
<p>We create a bowtie2 index:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>ngs_user@machine:~/maize/mapping$ bowtie2-build reference.fasta reference
Wrote 4312022 bytes to primary EBWT file: reference.rev.1.bt2
Wrote 76768 bytes to secondary EBWT file: reference.rev.2.bt2
Re-opening _in1 and _in2 as input streams
Returning from Ebwt constructor
</pre></div>
</div>
<p>We map the Illumina reads:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>ngs_user@machine:~/maize/mapping$ bowtie2 --very-sensitive-local -x reference -U ../clean_reads/b73xmo17.illumina.fastq -S illumina.sam

14415 reads; of these:
14415 (100.00%) were unpaired; of these:
  210 (1.46%) aligned 0 times
  13192 (91.52%) aligned exactly 1 time
  1013 (7.03%) aligned &gt;1 times
98.54% overall alignment rate
ngs_user@machine:~/maize/mapping$ samtools view -b -S -o mapped_illumina.bam illumina.sam
[samopen] SAM header is present: 100 sequences.
ngs_user@machine:~/maize/mapping$ rm illumina.sam
</pre></div>
</div>
</section>
<section id="merge-sort-and-index">
<h3>Merge, sort and index<a class="headerlink" href="#merge-sort-and-index" title="Permalink to this headline">¶</a></h3>
<p>We add the read groups to the 454 and Illumina BAMs to be able to differentiate the reads once we merge the BAM files</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>ngs_user@machine:~/maize/mapping$ picard-tools AddOrReplaceReadGroups ID=illumina LB=illumina PL=illumina PU=none SM=b73xmo17 SORT_ORDER=coordinate INPUT=mapped_illumina.bam OUTPUT=illumina.bam
ngs_user@machine:~/maize/mapping$ picard-tools AddOrReplaceReadGroups ID=454 LB=454 PL=454 PU=none SM=mo17 SORT_ORDER=coordinate INPUT=mapped_454.bam OUTPUT=454.bam
ngs_user@machine:~/maize/mapping$ rm  mapped_*
</pre></div>
</div>
<p>Now that we have the read groups set we can merge both BAM files:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>ngs_user@machine:~/maize/mapping$ picard-tools MergeSamFiles SORT_ORDER=coordinate ASSUME_SORTED=true INPUT=454.bam INPUT=illumina.bam OUTPUT=maize.bam
</pre></div>
</div>
<p>Finally we can index the BAM file:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>ngs_user@machine:~/maize/mapping$ samtools index maize.bam
</pre></div>
</div>
</section>
<section id="bam-statistics">
<h3>BAM statistics<a class="headerlink" href="#bam-statistics" title="Permalink to this headline">¶</a></h3>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>ngs_user@machine:~/maize/mapping$ samtools flagstat maize.bam
16526 + 0 in total (QC-passed reads + QC-failed reads)
0 + 0 duplicates
16211 + 0 mapped (98.09%:-nan%)
0 + 0 paired in sequencing
0 + 0 read1
0 + 0 read2
0 + 0 properly paired (-nan%:-nan%)
0 + 0 with itself and mate mapped
0 + 0 singletons (-nan%:-nan%)
0 + 0 with mate mapped to a different chr
0 + 0 with mate mapped to a different chr (mapQ&gt;=5)
ngs_user@machine:~/maize/mapping$ samtools idxstats maize.bam
TC458157      8763    34      0
TC458158      7672    321     0
TC458159      6012    404     0
TC458160      4906    364     0
TC458161      5087    476     0
TC458162      4895    274     0
TC458163      3960    19      0
(...)
</pre></div>
</div>
<p>It would be also advisable to check the coverage and MAPQ distributions obtained.</p>
<p>Take a look at the BAM files obtained with IGV.</p>
</section>
</section>
<section id="snps-calling">
<h2>SNPs calling<a class="headerlink" href="#snps-calling" title="Permalink to this headline">¶</a></h2>
<p>Use Varscan or other software to identify the SNPs.</p>
<p>First change BAM format to mpileup format.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>ngs_user@machine:~/maize/mapping$ cd ..
ngs_user@machine:~/maize$ cd snp_calling
ngs_user@machine:~/maize/snp_calling$ samtools mpileup -f maize_unigenes.fasta ../mapping/maize.bam &gt; maize.mpileup
</pre></div>
</div>
<p>Now we can look for SNPs with Varscan:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>ngs_user@machine:~/maize/snp_calling$ java -jar /home/ngs_user/software/VarScan.v2.3.7.jar mpileup2snp maize.mpileup --strand-filter 0  -output-vcf 1 &gt;maize_snp.vcf

Only SNPs will be reported
Warning: No p-value threshold provided, so p-values will not be calculated
Min coverage: 8
Min reads2:   2
Min var freq: 0.2
Min avg qual: 15
P-value thresh:       0.01
Reading input from maize.mpileup
161738 bases in pileup file
120 variant positions (120 SNP, 0 indel)
0 were failed by the strand-filter
120 variant positions reported (120 SNP, 0 indel)
</pre></div>
</div>
</section>
</section>



        

    </main>
            <div class="prevnext">
                    <!--<a class="prev" href="bedtools.html">BEDTools</a>-->
                    <a class="prev" href="bedtools.html" title="BEDTools">previous</a>
                    |
                    <!--<a class="next" href="final_practice.html">Practical tasks</a>-->
                    <a class="next" href="final_practice.html" title="Practical tasks">next</a>
                    |

                <a class="index" href="/courses/sequence_analysis/index.html" title="Index">index</a>

            </div>

</div>
    <footer>
            <a rel="license" href="https://creativecommons.org/licenses/by/4.0/"><img alt="Licencia de Creative Commons" style="border-width:0" src="https://i.creativecommons.org/l/by/4.0/88x31.png" /></a><br /><span xmlns:dct="https://purl.org/dc/terms/" property="dct:title">Bioinformatics at COMAV</span> is licensed under a <a rel="license" href="https://creativecommons.org/licenses/by/4.0/">Creative Commons Reconocimiento 4.0 Internacional License</a>.
    </footer>

<script type="text/javascript">
var gaJsHost = (("https:" == document.location.protocol) ? "https://ssl." : "http://www.");
document.write(unescape("%3Cscript src='" + gaJsHost + "google-analytics.com/ga.js' type='text/javascript'%3E%3C/script%3E"));
</script>
<script type="text/javascript">
try {
var pageTracker = _gat._getTracker("UA-16108274-1");
pageTracker._trackPageview();
} catch(err) {}</script>

  </body>
</html>