<!DOCTYPE html>

<html lang="en" data-content_root="./">
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="viewport" content="width=device-width, initial-scale=1" />

    <title>NGS expression analysis &#8212; Bioinformatics at COMAV 0.1 documentation</title>
    <link rel="stylesheet" type="text/css" href="_static/pygments.css?v=fa44fd50" />
    <link rel="stylesheet" type="text/css" href="_static/basic.css?v=c058f7c8" />
    <link rel="stylesheet" type="text/css" href="_static/alabaster.css?v=27fed22d" />
    <script src="_static/documentation_options.js?v=2709fde1"></script>
    <script src="_static/doctools.js?v=9a2dae69"></script>
    <script src="_static/sphinx_highlight.js?v=dc90522c"></script>
    <link rel="icon" href="_static/favicon.ico"/>
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Dealing with big result files" href="text_files.html" />
    <link rel="prev" title="SNP calling" href="snp_calling.html" />
   
  <link rel="stylesheet" href="_static/custom.css" type="text/css" />
  

  
  

  </head><body>
  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          

          <div class="body" role="main">
            
  <section id="ngs-expression-analysis">
<h1>NGS expression analysis<a class="headerlink" href="#ngs-expression-analysis" title="Link to this heading">¶</a></h1>
<iframe src="https://docs.google.com/presentation/d/e/2PACX-1vR3x87hIUakq-ObmWls9FaxCoWwCvK8EFc9cTk5b2g9TV0y94Fuwbg3WoAyzTtj7He_WtX_bM12qjTh/embed?start=false&loop=false&delayms=60000" frameborder="0" width="480" height="389" allowfullscreen="true" mozallowfullscreen="true" webkitallowfullscreen="true"></iframe><p>There are several interesting RNA-Seq reviews: <a class="reference external" href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-016-0881-8">A survey of best practices for RNA-seq data analysis</a>, <a class="reference external" href="http://www.bioconductor.org/help/workflows/rnaseqGene/">RNA-seq workflow: gene-level exploratory analysis and differential expression</a>, <a class="reference external" href="http://genomebiology.com/2010/11/12/220">From RNA-seq reads to differential expression results</a>, <a class="reference external" href="http://www.biomedcentral.com/1741-7007/9/34">Microarrays, deep sequencing and the true measure of the transcriptome</a>, <a class="reference external" href="http://www.nature.com/nmeth/journal/v8/n6/full/nmeth.1613.html">Computational methods for transcriptome annotation and quantification using RNA-seq</a>, <a class="reference external" href="http://www.nature.com/nrg/journal/v12/n10/full/nrg3068.html">Next-generation transcriptome assembly</a>, <a class="reference external" href="http://biorxiv.org/content/early/2016/02/18/035642">An opinionated guide to the proper care and feeding of your transcriptome</a> and <a class="reference external" href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-014-0553-5">Evaluation of de novo transcriptome assemblies from RNA-Seq data</a>.</p>
<p>There are also some interesting studies regarding the <a class="reference external" href="http://www.biomedcentral.com/1471-2164/14/328">post-processing</a>, <a class="reference external" href="http://www.biomedcentral.com/content/pdf/1471-2164-14-465.pdf">quality</a> of the assembled transcriptomes, the <a class="reference external" href="http://biorxiv.org/content/biorxiv/early/2014/07/14/007088.full.pdf">alternative transcript</a> reconstruction success of different algorithms, the <a class="reference external" href="http://genomebiology.com/2014/15/12/553/abstract">evaluation</a> of different assembly strategies.</p>
<p><a class="reference external" href="http://www.nature.com/nbt/journal/v33/n3/full/nbt.3122.html">StringTie</a> reconstruction algorithm.</p>
<p><a class="reference external" href="http://hibberdlab.com/transrate/">transrate</a>, a software to evaluate the transcriptome assembly quality.</p>
</section>
<section id="practical-tasks">
<h1>Practical tasks<a class="headerlink" href="#practical-tasks" title="Link to this heading">¶</a></h1>
<section id="mapping-and-count-sequences">
<h2>Mapping and count sequences<a class="headerlink" href="#mapping-and-count-sequences" title="Link to this heading">¶</a></h2>
<p>The objective is to map and count redas using an <a class="reference download internal" download="" href="_downloads/8b0bd8b64c5c16bef1f43a4e9488ffb6/SRR45_region.fastq"><code class="xref download docutils literal notranslate"><span class="pre">example</span> <span class="pre">file</span></code></a>  of a RNAseq experience  with only a sample. We use <a class="reference external" href="http://daehwankimlab.github.io/hisat2/">Hisat2</a>  to map read againt the <a class="reference download internal" download="" href="_downloads/970c0364472e11c04558b39a7495b09a/sl2.fasta"><code class="xref download docutils literal notranslate"><span class="pre">genome</span></code></a> and <a class="reference external" href="https://ccb.jhu.edu/software/stringtie/">StringTie</a> to count the reads.</p>
<p>First we download the <a class="reference download internal" download="" href="_downloads/2917d130a06ff95710fced6d4c9a6d11/sl2_index.zip"><code class="xref download docutils literal notranslate"><span class="pre">index</span> <span class="pre">files</span> <span class="pre">of</span> <span class="pre">genome</span></code></a>.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>ngs_user@machine: mkdir rnaseq
ngs_user@machine:~$ cd rnaseq
ngs_user@machine:~/rnaseq$ unzip sl2.index.zip
</pre></div>
</div>
<p>Now, we map the reads.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>ngs_user@machine:~/rnaseq$ hisat2 -x sl2_index -U SRR45_region.fastq -S SRR45.sam
</pre></div>
</div>
<p>We sort and index the bam file to open with IGV.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>ngs_user@machine:~/rnaseq/tophat_out$ samtools view -hb -o SRR45.bam -S SRR45.sam

ngs_user@machine:~/rnaseq/tophat_out$ samtools sort SRR45.bam -o SRR45_sort.bam
</pre></div>
</div>
<p>Now, we can construct the transcripts and the number of reads in each transcript.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>ngs_user@machine:~/rnaseq/tophat_out$ stringtie  SRR45_sort.bam  -o SRR45.gtf -A SRR45_exp.txt
</pre></div>
</div>
<p>To use the gene models as reference, we need also the <a class="reference download internal" download="" href="_downloads/c6390e112c9e9176d201208d13aca3a2/sl2.gff3"><code class="xref download docutils literal notranslate"><span class="pre">genome</span> <span class="pre">annotation</span> <span class="pre">file</span></code></a></p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>ngs_user@machine:~/rnaseq/tophat_out$ stringtie  SRR45_sort.bam -G sl2.gff3 -o SRR45_ref.gtf -A SRR45_exp_ref.txt
</pre></div>
</div>
<p>Load the bam sorted file, the gtf files and the annotation file sl2.gff3 on the IGV to test the mapping and transcript assembly done with StrinTie. The next steps will be integrate all transcript assemblies of all samples to create a new annotation file and use it to quantify again the reads of each sample. After that we will start the normalization, differential analysis, etc.</p>
</section>
</section>


          </div>
          
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="Main">
        <div class="sphinxsidebarwrapper">
            <p class="logo"><a href="index.html">
              <img class="logo" src="_static/comav.png" alt="Logo of Bioinformatics at COMAV"/>
            </a></p>
<h1 class="logo"><a href="index.html">Bioinformatics at COMAV</a></h1>









<search id="searchbox" style="display: none" role="search">
    <div class="searchformwrapper">
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" aria-labelledby="searchlabel" autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" placeholder="Search"/>
      <input type="submit" value="Go" />
    </form>
    </div>
</search>
<script>document.getElementById('searchbox').style.display = "block"</script><h3>Navigation</h3>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="sequencing_technologies.html">Sequencing technologies</a></li>
<li class="toctree-l1"><a class="reference internal" href="ngs_applications_examples.html">NGS application examples</a></li>
<li class="toctree-l1"><a class="reference internal" href="sequence_analysis_software.html">NGS analysis software</a></li>
<li class="toctree-l1"><a class="reference internal" href="cmd_primer.html">Command line primer</a></li>
<li class="toctree-l1"><a class="reference internal" href="sequence_file_formats.html">NGS sequence file formats</a></li>
<li class="toctree-l1"><a class="reference internal" href="read_cleaning.html">NGS read cleaning</a></li>
<li class="toctree-l1"><a class="reference internal" href="assembly.html">Sequence assembly</a></li>
<li class="toctree-l1"><a class="reference internal" href="mapping.html">Sequence mapping</a></li>
<li class="toctree-l1"><a class="reference internal" href="snp_calling.html">SNP calling and filtering</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">NGS expression analysis</a></li>
<li class="toctree-l1"><a class="reference internal" href="#practical-tasks">Practical tasks</a></li>
<li class="toctree-l1"><a class="reference internal" href="text_files.html">Dealing with big result files</a></li>
<li class="toctree-l1"><a class="reference internal" href="bedtools.html">BEDTools</a></li>
<li class="toctree-l1"><a class="reference internal" href="final_practice.html">Final practice</a></li>
</ul>

<div class="relations">
<h3>Related Topics</h3>
<ul>
  <li><a href="index.html">Documentation overview</a><ul>
      <li>Previous: <a href="snp_calling.html" title="previous chapter">SNP calling</a></li>
      <li>Next: <a href="text_files.html" title="next chapter">Dealing with big result files</a></li>
  </ul></li>
</ul>
</div>








        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="footer">
      &#169;2010, Jose Blanca & Joaquin Cañizares.
      
      |
      Powered by <a href="https://www.sphinx-doc.org/">Sphinx 8.0.2</a>
      &amp; <a href="https://alabaster.readthedocs.io">Alabaster 1.0.0</a>
      
    </div>

    

    
  </body>
</html>