
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    <meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

    <title>NGS file formats &mdash; Bioinformatics at COMAV 0.1 documentation</title>
    <link rel="stylesheet" href="_static/style.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <link rel="stylesheet" href="_static/style.css" type="text/css" />
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    './',
        VERSION:     '0.1',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="_static/documentation_options.js"></script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <link rel="shortcut icon" href="_static/favicon.ico"/>
    <link rel="search" title="Search" href="search.html" />
    <link rel="top" title="Bioinformatics at COMAV 0.1 documentation" href="index.html" />
    <link rel="next" title="Read QA and Cleaning" href="read_cleaning.html" />
    <link rel="prev" title="Command line primer" href="cmd_primer.html" /> 
  </head>
  <body>

<div class="body">
    <header>
        <h1><a href="index.html">Bioinformatics at COMAV</a></h1>
    </header>


    <nav class="horizontal_nav page_nav">
        <ul>
            <li><a href="index.html">Bioinformatics & genomics</a>
                <ul>
                    <li><a href="services.html">Services</a></li>
                    <li><a href="people.html">People</a></li>
                    <li><a href="publications.html">Publications</a></li>
                    <li><a href="getting_here.html">Getting here</a></li>
<!--                    <li><a href="http://bioinf.comav.upv.es/notes/">Wiki</a></li>-->
               </ul>
            </li>
            <li><a>Software</a>
                <ul>
                    <li><a href="https://github.com/bioinfcomav">Git</a></li>
                    <li><a href="https://github.com/JoseBlanca">Git Jose Blanca</a></li>
               </ul>
            </li>
            <li><a href="courses.html">Courses</a>
                <ul>
                    <li><a href="courses/intro_bioinf/index.html">Bionformática básica</a></li>
                    <li><a href="courses/sequence_analysis/index.html">NGS sequence analysis</a></li>
                    <li><a href="courses/linux/index.html">Linux y Python</a></li>
                    <li><a href="courses/unix/index.html">Linux</a></li>
                    <li><a href="courses/biotech3/index.html">Bioinf biotecnología</a></li>
               </ul>
        </ul>
    </nav>

    <main>
                
    
  <section id="ngs-file-formats">
<h1>NGS file formats<a class="headerlink" href="#ngs-file-formats" title="Permalink to this headline">¶</a></h1>
<p>There are lots of file formats related to NGS analyses.
The most common ones are:</p>
<a class="reference internal image-reference" href="_images/ngs_map_read_file_formats.png"><img alt="_images/ngs_map_read_file_formats.png" src="_images/ngs_map_read_file_formats.png" style="width: 650px;" /></a>
<a class="reference internal image-reference" href="_images/ngs_file_formats.png"><img alt="_images/ngs_file_formats.png" src="_images/ngs_file_formats.png" style="width: 650px;" /></a>
<section id="sequence-file-formats">
<h2>Sequence file formats<a class="headerlink" href="#sequence-file-formats" title="Permalink to this headline">¶</a></h2>
<p>The different sequence related formats include different information about the sequence.
The most common file formats in the NGS world are: fastq and sff.</p>
</section>
<section id="sff">
<h2>SFF<a class="headerlink" href="#sff" title="Permalink to this headline">¶</a></h2>
<p>The <a class="reference external" href="http://www.ncbi.nlm.nih.gov/Traces/trace.cgi?cmd=show&amp;f=formats&amp;m=doc&amp;s=format#sff">SFF</a> (Standard Flowgram Format) files are the 454 equivalent to the ABI chromatogram files.
They hold information about:</p>
<blockquote>
<div><ul class="simple">
<li><p>the flowgram,</p></li>
<li><p>the called sequence,</p></li>
<li><p>the qualities,</p></li>
<li><p>and the recommended quality and adaptor clipping.</p></li>
</ul>
</div></blockquote>
<p>These recommended clippings are given by the 454 sequencer.
The Roche software takes into account the quality and the adaptor sequence to recommend a clipping for each sequence.
Like the ABI files, these are binary files that should be opened with specialized programs.
There are several tools to extract the sequences and to convert them to a more usable format.
Roche provides one executable able to do it with the 454 machine.
Alternatively we can use the <a class="reference external" href="http://bioinf.comav.upv.es/sff_extract/index.html">sff_extract</a> tool to obtain a fasta file.</p>
</section>
<section id="fasta">
<h2>Fasta<a class="headerlink" href="#fasta" title="Permalink to this headline">¶</a></h2>
<p>The fasta format is based on a simple text.
Each sequence starts with a “&gt;” followed by the sequence name, an space and, optionally, the description</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="o">&gt;</span><span class="n">seq_1</span> <span class="n">description</span>
<span class="n">GATTTGGGGTTCAAAGCAGTATCGATCAAATAGTAAATCCATTTGTTCAACTCACAGTTT</span>
<span class="o">&gt;</span><span class="n">seq_2</span>
<span class="n">ATCGTAGTCTAGTCTATGCTAGTGCGATGCTAGTGCTAGTCGTATGCATGGCTATGTGTG</span>
</pre></div>
</div>
<p>Usually, if we have quality information, another fasta file with the quality information could be provided.
In this cases both the sequence and the quality file should have the sequences in the same order.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="o">&gt;</span><span class="n">seq_1</span> <span class="n">description</span>
<span class="mi">54</span> <span class="mi">57</span> <span class="mi">54</span> <span class="mi">57</span> <span class="mi">48</span> <span class="mi">48</span> <span class="mi">48</span> <span class="mi">48</span> <span class="mi">57</span> <span class="mi">57</span> <span class="mi">57</span> <span class="mi">47</span> <span class="mi">47</span> <span class="mi">41</span> <span class="mi">42</span> <span class="mi">41</span> <span class="mi">47</span> <span class="mi">57</span> <span class="mi">57</span> <span class="mi">57</span> <span class="mi">57</span> <span class="mi">47</span> <span class="mi">44</span> <span class="mi">44</span> <span class="mi">44</span> <span class="mi">44</span> <span class="mi">50</span> <span class="mi">50</span>
<span class="mi">54</span> <span class="mi">57</span> <span class="mi">57</span> <span class="mi">46</span> <span class="mi">43</span> <span class="mi">37</span> <span class="mi">44</span> <span class="mi">43</span> <span class="mi">57</span> <span class="mi">37</span> <span class="mi">37</span> <span class="mi">37</span> <span class="mi">57</span> <span class="mi">57</span> <span class="mi">57</span> <span class="mi">57</span> <span class="mi">52</span> <span class="mi">52</span> <span class="mi">52</span> <span class="mi">52</span> <span class="mi">57</span> <span class="mi">50</span> <span class="mi">47</span> <span class="mi">47</span> <span class="mi">52</span>
<span class="o">&gt;</span><span class="n">seq_2</span>
<span class="mi">52</span> <span class="mi">47</span> <span class="mi">52</span> <span class="mi">52</span> <span class="mi">50</span> <span class="mi">50</span> <span class="mi">50</span> <span class="mi">50</span> <span class="mi">50</span> <span class="mi">57</span> <span class="mi">57</span> <span class="mi">54</span> <span class="mi">57</span> <span class="mi">57</span> <span class="mi">57</span> <span class="mi">57</span> <span class="mi">57</span> <span class="mi">57</span> <span class="mi">57</span> <span class="mi">46</span> <span class="mi">46</span> <span class="mi">57</span> <span class="mi">57</span> <span class="mi">57</span> <span class="mi">57</span> <span class="mi">57</span> <span class="mi">57</span> <span class="mi">57</span>
<span class="mi">57</span> <span class="mi">57</span> <span class="mi">57</span> <span class="mi">57</span> <span class="mi">57</span> <span class="mi">57</span> <span class="mi">57</span> <span class="mi">57</span> <span class="mi">57</span> <span class="mi">57</span> <span class="mi">57</span> <span class="mi">57</span> <span class="mi">29</span> <span class="mi">29</span>
</pre></div>
</div>
</section>
<section id="sanger-fastq">
<h2>sanger fastq<a class="headerlink" href="#sanger-fastq" title="Permalink to this headline">¶</a></h2>
<p>The <a class="reference external" href="http://en.wikipedia.org/wiki/FASTQ_format">fastq</a> format was developed to provide a convenient way of storing the sequence and the quality scores in the same file.
These are text files and they look like:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>@seq_1
GATTTGGGGTTCAAAGCAGTATCGATCAAATAGTAAATCCATTTGTTCAACTCACAGTTT
+
!&#39;&#39;*((((***+))%%%++)(%%%%).1***-+*&#39;&#39;))**55CCF&gt;&gt;&gt;&gt;&gt;&gt;CCCCCCC65
@seq_2
ATCGTAGTCTAGTCTATGCTAGTGCGATGCTAGTGCTAGTCGTATGCATGGCTATGTGTG
+
208DA8308AD8SF83FH0SD8F08APFIDJFN34JW830UDS8UFDSADPFIJ3N8DAA
</pre></div>
</div>
<p>In this file every sequence has 4 lines.
In the first line we get the sequence name after the symbol “&#64;” and, optionally, the description.
The second line has the sequence and the fourth line has the quality scores encoded as letters.</p>
<p>Quality coding (modified from wikipedia).</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS......................
..........................XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX......................
...............................IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII......................
!&quot;#$%&amp;&#39;()*+,-./0123456789:;&lt;=&gt;?@ABCDEFGHIJKLMNOPQRSTUVWXYZ[\]^_`abcdefghijklmnopqrstuvwxyz{|}~
|                         |    |        |                              |                     |
33                        59   64       73                            104                   126

S - Sanger        Phred+33,  raw reads typically (0, 40)
X - Solexa        Solexa+64, raw reads typically (-5, 40)
I - Illumina      Phred+64,  raw reads typically (0, 40)
</pre></div>
</div>
</section>
<section id="illummina-fastq">
<h2>Illummina fastq<a class="headerlink" href="#illummina-fastq" title="Permalink to this headline">¶</a></h2>
<p>This file is almost identical to a sanger fastq file, but the encoding for the quality scores is different.
When we deal with a fastq file we have to be sure about which kind of file we are dealing with, an illumina fastq or a sanger fastq. Unfortunately they are not easy to differentiate.
Also you have to take into account that solexa used to had a third fastq format, the solexa fastq, although this one is mostly obsoleted.
Recently Illumina has also decided to distribute its files as Sanger fastq, so the Illumina fastq will be not used any more.</p>
<p>One of the seq_crumbs utilities, guess_seq_format, is able to differentiate the Sanger from the Illumina version by looking for quality characters exclusive of the Sanger version.</p>
</section>
<section id="sra">
<h2>SRA<a class="headerlink" href="#sra" title="Permalink to this headline">¶</a></h2>
<p>SRA is the file format in which all <a class="reference external" href="http://www.ncbi.nlm.nih.gov/sra">NCBI SRA</a> content is provided.
SRA files are binary files and we need specific tools to extract the information.
There is a toolkit (<a class="reference external" href="https://github.com/ncbi/sra-tools">SRA Toolkit</a>)developed by NCBI to deal with these binary files.</p>
</section>
<section id="compressed-files">
<h2>Compressed files<a class="headerlink" href="#compressed-files" title="Permalink to this headline">¶</a></h2>
<p>Sometime these sequence text file can be found compressed to save up hard drive space.
The most common compression formats are <a class="reference external" href="http://en.wikipedia.org/wiki/Gzip">gzip</a> and <a class="reference external" href="http://samtools.sourceforge.net/tabix.shtml">bgzip</a>.
bgzip is a gzip variant commonly used in genomics because, although it is a little less efficient in the compression ratio, it allows random access.
Most software is becoming compatible with these formats.</p>
</section>
<section id="paired-files">
<h2>Paired files<a class="headerlink" href="#paired-files" title="Permalink to this headline">¶</a></h2>
<p>It is common to obtain two reads from a single molecule.
Examples of these techniques are the Illumina pair-ends and mate-pairs.
In this cases for each read there is another paired read.
One common way to store those paired reads is to create to fastq files, one for the first read of the pairs and another one for the second.
In this case the files should hold the reads exactly in the same order.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">Fastq</span> <span class="n">file</span> <span class="mi">1</span>
<span class="nd">@molecule_1</span> <span class="mi">1</span><span class="n">st_read_from_pair</span>
<span class="nd">@molecule_2</span> <span class="mi">1</span><span class="n">st_read_from_pair</span>
<span class="nd">@molecule_3</span> <span class="mi">1</span><span class="n">st_read_from_pair</span>

<span class="n">Fastq</span> <span class="n">file</span> <span class="mi">2</span>
<span class="nd">@molecule_1</span> <span class="mi">2</span><span class="n">nd_read_from_pair</span>
<span class="nd">@molecule_2</span> <span class="mi">2</span><span class="n">nd_read_from_pair</span>
<span class="nd">@molecule_3</span> <span class="mi">2</span><span class="n">nd_read_from_pair</span>
</pre></div>
</div>
<p>Another option is to interleave the reads in a single file alternating the first and second read for each pair.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">Interleaved</span> <span class="n">Fastq</span> <span class="n">file</span>
<span class="nd">@molecule_1</span> <span class="mi">1</span><span class="n">st_read_from_pair</span>
<span class="nd">@molecule_1</span> <span class="mi">2</span><span class="n">nd_read_from_pair</span>
<span class="nd">@molecule_2</span> <span class="mi">1</span><span class="n">st_read_from_pair</span>
<span class="nd">@molecule_2</span> <span class="mi">2</span><span class="n">nd_read_from_pair</span>
<span class="nd">@molecule_3</span> <span class="mi">1</span><span class="n">st_read_from_pair</span>
<span class="nd">@molecule_3</span> <span class="mi">2</span><span class="n">nd_read_from_pair</span>
</pre></div>
</div>
<p>Depending on the software that we want to use we should the interleaved or the two file version; but most of actual ones use the one file version.</p>
<section id="id2">
<h3>SRA toolkit<a class="headerlink" href="#id2" title="Permalink to this headline">¶</a></h3>
<p>Download the <a class="reference external" href="https://sra-pub-run-odp.s3.amazonaws.com/sra/SRR2970642/SRR2970642">sra file</a> <a class="reference external" href="https://www.ncbi.nlm.nih.gov/sra/SRX1460040">(file info)</a> and use fastq-dump to convert to fastq format.</p>
<p>First, you can run the program with the default options and look at the output file:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>ngs_user@ngsmachine:~$ fastq-dump SRR2970642
</pre></div>
</div>
<p>Depending on the software that we want to use we should the interleaved or the two file version; but most of actual ones use the one file version.</p>
<p>Is this useful? Taking into account that the reads are paired reads, can we improve the convertion?:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>ngs_user@ngsmachine:~$ fastq-dump --split-files --defline-qual &quot;+&quot; --defline-seq &#39;@$sn/$ri&#39; --gzip SRR2970642
</pre></div>
</div>
</section>
</section>
</section>



        

    </main>

</div>
    <footer>
            <a rel="license" href="https://creativecommons.org/licenses/by/4.0/"><img alt="Licencia de Creative Commons" style="border-width:0" src="https://i.creativecommons.org/l/by/4.0/88x31.png" /></a><br /><span xmlns:dct="https://purl.org/dc/terms/" property="dct:title">Bioinformatics at COMAV</span> is licensed under a <a rel="license" href="https://creativecommons.org/licenses/by/4.0/">Creative Commons Reconocimiento 4.0 Internacional License</a>.
    </footer>

<script type="text/javascript">
var gaJsHost = (("https:" == document.location.protocol) ? "https://ssl." : "http://www.");
document.write(unescape("%3Cscript src='" + gaJsHost + "google-analytics.com/ga.js' type='text/javascript'%3E%3C/script%3E"));
</script>
<script type="text/javascript">
try {
var pageTracker = _gat._getTracker("UA-16108274-1");
pageTracker._trackPageview();
} catch(err) {}</script>

  </body>
</html>