
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    <meta name="generator" content="Docutils 0.18.1: http://docutils.sourceforge.net/" />

    <title>SNP calling &mdash; Bioinformatics at COMAV 0.1 documentation</title>
    <link rel="stylesheet" href="_static/style.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <link rel="stylesheet" href="_static/style.css" type="text/css" />
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    './',
        VERSION:     '0.1',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="_static/documentation_options.js"></script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/_sphinx_javascript_frameworks_compat.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <link rel="shortcut icon" href="_static/favicon.ico"/>
    <link rel="search" title="Search" href="search.html" />
    <link rel="top" title="Bioinformatics at COMAV 0.1 documentation" href="index.html" />
    <link rel="next" title="NGS expression analysis" href="rnaseq.html" />
    <link rel="prev" title="Mapping" href="mapping.html" /> 
  </head>
  <body>

<div class="body">
    <header>
        <h1><a href="index.html">Bioinformatics at COMAV</a></h1>
    </header>


    <nav class="horizontal_nav page_nav">
        <ul>
            <li><a href="/index.html">Bioinformatics & genomics</a>
                <ul>
                    <li><a href="/services.html">Services</a></li>
                    <li><a href="/people.html">People</a></li>
                    <li><a href="/publications.html">Publications</a></li>
                    <li><a href="/getting_here.html">Getting here</a></li>
<!--                    <li><a href="http://bioinf.comav.upv.es/notes/">Wiki</a></li>-->
               </ul>
            </li>
            <li><a href="/software.html">Software</a>
                <ul>
                    <li><a href="https://github.com/bioinfcomav/">Bioinf COMAV</a></li>
                    <li><a href="https://github.com/JoseBlanca/">Jose Blanca</a></li>
                    <!--
                    <li><a href="/seq_crumbs/index.html">seq_crumbs</a></li>
                    <li><a href="/ngs_backbone/index.html">ngs_backbone</a></li>
                    <li><a href="/clean_reads/index.html">clean_reads</a></li>
                    <li><a href="/sff_extract/index.html">sff_extract</a></li>
                    <li><a href="/psubprocess/index.html">psubprocess</a></li>-->
               </ul>
            </li>
            <li><a href="/courses.html">Courses</a>
                <ul>
                    <li><a href="/courses/intro_bioinf/index.html">Bionformática básica</a></li>
                    <li><a href="/courses/sequence_analysis/index.html">NGS sequence analysis</a></li>
                    <li><a href="/courses/linux/index.html">Linux y Python</a></li>
                    <li><a href="/courses/unix/index.html">Linux para Biologos</a></li>
               </ul>
        </ul>
    </nav>

    <main>
                
    
  <section id="snp-calling">
<h1>SNP calling<a class="headerlink" href="#snp-calling" title="Permalink to this heading">¶</a></h1>
<iframe src="https://docs.google.com/presentation/d/e/2PACX-1vR2qiKIr-tDVtajc3qiE_4lIacAK2AEnSANfKCiNbViv3MJxRTrj466hEqyjUM1ZLDwBd1eYt8rD4Iw/embed?start=false&loop=false&delayms=60000" frameborder="0" width="480" height="389" allowfullscreen="true" mozallowfullscreen="true" webkitallowfullscreen="true"></iframe><section id="practical-tasks">
<h2>Practical tasks<a class="headerlink" href="#practical-tasks" title="Permalink to this heading">¶</a></h2>
<section id="snp-calling-with-mpileup">
<h3>SNP calling with mpileup<a class="headerlink" href="#snp-calling-with-mpileup" title="Permalink to this heading">¶</a></h3>
<ol class="arabic simple">
<li><p>SNP calling with samtools’ mpileup.</p></li>
</ol>
<p>Create a directory named snp_call and download  the files.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>ngs_user@ngsmachine:~$ cd
ngs_user@ngsmachine:~$ pwd
/home/ngs_user
ngs_user@ngsmachine:~$ mkdir snp_call
ngs_user@ngsmachine:~$ cd snp_call/
</pre></div>
</div>
<p>Download the <a class="reference download internal" download="" href="_downloads/eaef58983741eb3091f95f9b176023a0/human_cambridge_reference_mito.fasta"><code class="xref download docutils literal notranslate"><span class="pre">reference</span></code></a> and the  <a class="reference download internal" download="" href="_downloads/d101fa348c0ea727f218ab3d17e7c6c2/mito_yoruba_reads_pe1.sorted.bam"><code class="xref download docutils literal notranslate"><span class="pre">BAM</span> <span class="pre">file</span></code></a> in to this directory.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>ngs_user@ngsmachine:~/snp_call$ ls

human_cambridge_reference_mito.fasta  mito_yoruba_reads_pe1.sorted.bam
</pre></div>
</div>
<p>Use Samtools to index the bam file.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>ngs_user@ngsmachine:~/snp_call$samtools index mito_yoruba_reads_pe1.sorted.bam
</pre></div>
</div>
<p>Use <a class="reference external" href="http://samtools.sourceforge.net/mpileup.shtml">mpileup</a> and bcftools to call the SNPs</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>ngs_user@ngsmachine:~/snp_call$ bcftools view
Usage:   bcftools view [options] &lt;in.bcf&gt; [reg]
Options: -c        SNP calling
         -v        output potential variant sites only (force -c)
         -g        call genotypes at variant sites (force -c)
         -b        output BCF instead of VCF
         (...)
ngs_user@machine:~/snp_call$ samtools mpileup -ugf human_cambridge_reference_mito.fasta mito_yoruba_reads_pe1.sorted.bam | bcftools call -cv - &gt; var.raw.vcf

[mpileup] 1 samples in 1 input files
&lt;mpileup&gt; Set max per-file depth to 8000
[afs] 0:15459.489 1:10.430 2:54.081

ngs_user@machine:~/snp_call$ bcftools view var.raw.vcf | vcfutils.pl varFilter -D100 &gt; var_mito_yoruba_mpileup.flt.vcf
</pre></div>
</div>
<p>The resulting vcf file contains our SNPs.
There are plenty of options to tweak the SNP calling process.
To get an idea of the possibilities go to the <a class="reference external" href="http://samtools.sourceforge.net/mpileup.shtml">mpileup</a> page.</p>
</section>
<section id="looking-at-the-snps-using-igv">
<h3>Looking at the SNPs using IGV<a class="headerlink" href="#looking-at-the-snps-using-igv" title="Permalink to this heading">¶</a></h3>
<p>In the IGV we can load the BAM, GFF and the VCF files.
In that way we can compare the mapping with the annotation.
To do it, open IGV and load the BAM file from the mapping subdirectory.
This time you won’t need to import the reference genome, it will be automatically selected, because it was the last reference used.
Load the BAM file as you did the last time.
Also load the compressed and indexed VCF file in annotation/features/ (you could sort and index the vcf file with an IGV tool).
Now you should have two tracks in IGV, one with the mapping and another one with the SNP annotation.
We could also load the GFF file with all the annotations, such us SSRs, ORFs, etc.
In this case the GFF and VCF tracks would show similar annotation.</p>
<p>Sometimes IGV shows the GFF track collapsed by default, you can expand it by clicking on the right mouse button above the track and selecting expand.</p>
<a class="reference internal image-reference" href="_images/igv_bam_and_vcf.png"><img alt="_images/igv_bam_and_vcf.png" src="_images/igv_bam_and_vcf.png" style="width: 650px;" /></a>
</section>
<section id="snp-calling-with-freebayes">
<h3>SNP calling with FreeBayes<a class="headerlink" href="#snp-calling-with-freebayes" title="Permalink to this heading">¶</a></h3>
<p><a class="reference external" href="http://bioinformatics.bc.edu/marthlab/FreeBayes">Freebayes</a> is a SNP calling program based on bayesian statistics. It is able to deal with individual and populations  or pooled and polyploid samples. FreeBayes is versatil and ajustable, then is necesary to deal with their parameters and options.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>ngs_user@ngsmachine:~/snp_call$ freebayes -h
</pre></div>
</div>
<p>Use FreeBayes to identify SNPs in the previous files.:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>ngs_user@ngsmachine:~/snp_call$ freebayes -f human_cambridge_reference_mito.fasta -b mito_yoruba_reads_pe1.sorted.bam  -v ./var_mito_yoruba_freebayes.vcf
</pre></div>
</div>
<p>We have used the FreeBayes with the default configuration and our sample is haploid, run again with this parameter modified.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>ngs_user@ngsmachine:~/snp_call$ freebayes -f human_cambridge_reference_mito.fasta -b mito_yoruba_reads_pe1.sorted.bam -p 1   -v ./var_mito_yoruba_freebayes_haplo.vcf
</pre></div>
</div>
<p>It is posible improve the result with quality paramenters such as:</p>
<ul class="simple">
<li><dl class="simple">
<dt>-m –min-mapping-quality Q</dt><dd><p>Exclude alignments from analysis if they have a mapping
quality less than Q.  default: 30</p>
</dd>
</dl>
</li>
<li><dl class="simple">
<dt>-q –min-base-quality Q</dt><dd><p>Exclude alleles from analysis if their supporting base
quality is less than Q.  default: 20</p>
</dd>
</dl>
</li>
<li><dl class="simple">
<dt>-C –min-alternate-count N</dt><dd><p>Require at least this count of observations supporting
an alternate allele within a single individual in order
to evaluate the position.  default: 1</p>
</dd>
</dl>
</li>
</ul>
<p>Run now limiting to alternate alleles detected in three or more reads.:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>ngs_user@ngsmachine:~/snp_call$ freebayes -f human_cambridge_reference_mito.fasta -b mito_yoruba_reads_pe1.sorted.bam -p 1 -C 3  -v ./var_mito_yoruba_freebayes_haplo_reads3.vcf
</pre></div>
</div>
</section>
<section id="snp-calling-with-varscan">
<h3>SNP calling with VarScan<a class="headerlink" href="#snp-calling-with-varscan" title="Permalink to this heading">¶</a></h3>
<p>The <a class="reference external" href="http://varscan.sourceforge.net/">VarsScan</a> is a SNP calling than works with more simple statistics that may be more robust in extreme read depth, pooled samples, and contaminated or impure samples. VarScan employs statistics based on thresholds for read depth, base quality, variant allele frequency, etc.</p>
<p>First is necessary to change the alignement file from BAM to mpileuo format using samtools.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>ngs_user@ngsmachine:~/snp_call$ samtools mpileup -f human_cambridge_reference_mito.fasta mito_yoruba_reads_pe1.sorted.bam &gt;mito_yoruba_reads_pe1.mpileup
</pre></div>
</div>
<p>There are several programs in Varscan.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>ngs_user@ngsmachine:~/snp_call$ varscan
</pre></div>
</div>
<p>mpileup snp is the comand to identify SNPs, but it do not call indels.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>ngs_user@ngsmachine:~/snp_call$ varscan mpileup2snp -h

ngs_user@ngsmachine:~/snp_call$ varscan mpileup2snp mito_yoruba_reads_pe1.mpileup -output-vcf 1 &gt;var_mito_yoruba_varscan.vcf
</pre></div>
</div>
<p>Using IGV, compare the mito_yoruba_haplo_reads3.vcf and mito_yoruba_snps.vcf files, why have  not VarScan detected SNPs called by Freebayes?. We have used this option by default.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="o">--</span><span class="n">strand</span><span class="o">-</span><span class="nb">filter</span>       <span class="n">Ignore</span> <span class="n">variants</span> <span class="k">with</span> <span class="o">&gt;</span><span class="mi">90</span><span class="o">%</span> <span class="n">support</span> <span class="n">on</span> <span class="n">one</span> <span class="n">strand</span> <span class="p">[</span><span class="mi">1</span><span class="p">]</span>
</pre></div>
</div>
<p>Look for SNPs without this strand filter.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">varscan</span> <span class="n">mpileup2snp</span> <span class="n">mito_yoruba_reads_pe1</span><span class="o">.</span><span class="n">mpileup</span> <span class="o">--</span><span class="n">strand</span><span class="o">-</span><span class="nb">filter</span> <span class="mi">0</span>  <span class="o">-</span><span class="n">output</span><span class="o">-</span><span class="n">vcf</span> <span class="mi">1</span> <span class="o">&gt;</span><span class="n">var_mito_yoruba_snps_varscan_not_filter_strand</span><span class="o">.</span><span class="n">vcf</span>
</pre></div>
</div>
<p>What did happen with the mileup snps?. Detect the mistake on the command.</p>
</section>
<section id="snp-calling-using-calmd">
<h3>SNP calling using calmd<a class="headerlink" href="#snp-calling-using-calmd" title="Permalink to this heading">¶</a></h3>
<p>Use freebayes to identify snps in these <a class="reference download internal" download="" href="_downloads/e39b52a94ecf5ebc10db706fb17ad458/alignments.bam"><code class="xref download docutils literal notranslate"><span class="pre">Celegans</span> <span class="pre">alignments</span></code></a> against <a class="reference download internal" download="" href="_downloads/79185616fdfc37e971281a69d48adfc1/ref.fasta"><code class="xref download docutils literal notranslate"><span class="pre">this</span> <span class="pre">reference</span> <span class="pre">sequence</span></code></a>. Take into account that this bam file has not been processed by samtools calmd. Search for snps with pre calmd processed alignment file and with the post processed file.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>ngs_user@ngsmachine:~/snp_call$ samtools index alignments.bam

ngs_user@ngsmachine:~/snp_call$ freebayes -f ref.fasta --min-base-quality 20  alignments.bam |bgzip &gt; alignments.vcf.gz

ngs_user@ngsmachine:~/snp_call$ tabix -p vcf alignments.vcf.gz
</pre></div>
</div>
<p>Now look for the snvs after procesing the alignment file with calmd</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>ngs_user@ngsmachine:~/snp_call$ samtools calmd -Arb alignments.bam ref.fasta &gt; alignments.calmd.bam

ngs_user@ngsmachine:~/snp_call$ samtools index alignments.calmd.bam

ngs_user@ngsmachine:~/snp_call$ freebayes -f ref.fasta --min-base-quality 20  alignments.calmd.bam |bgzip &gt; alignments.calmd.vcf.gz

ngs_user@ngsmachine:~/snp_call$ tabix -p vcf alignments.calmd.vcf.gz
</pre></div>
</div>
<p>Look for the differences using IGV. Take a look at the position 7950020.</p>
<p>If you load the bam files into IGV, can you see differences between them?</p>
<hr class="docutils" />
<p>We have a <a class="reference download internal" download="" href="_downloads/ce3dfe1c084c24d74923fdf5ddb12f51/ril.vcf.gz"><code class="xref download docutils literal notranslate"><span class="pre">VCF</span> <span class="pre">file</span></code></a> from a RIL segregant population.</p>
<p>We can use vcftools to filter and to do some statistics.</p>
<p>Filter SNPs with quality minor than 10:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ vcftools --gzvcf ril.vcf.gz --recode --recode-INFO-all --minQ 10 --stdout | gzip -c &gt; ril_vcf_Q10only.vcf.gz
</pre></div>
</div>
<p>Filter SNPs with more than 20% of missing data:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ vcftools --gzvcf ril.vcf.gz --recode --recode-INFO-all --max-missing 0.8 --stdout |  gzip -c &gt; ril_vcf_0.8_missing.vcf.gz
</pre></div>
</div>
<p>Select all SNPs in a determiante region:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ vcftools --gzvcf ril.vcf.gz --recode --recode-INFO-all --chr CP4_pseudomolecule00 --from-bp 161000 --to-bp 245000 --stdout |  gzip -c &gt; ril_vcf_candidate-region.vcf.gz
</pre></div>
</div>
<p>Calculate the observed heterozygosity of all SNPs:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ vcftools --gzvcf ril.vcf.gz   --het --out ril
</pre></div>
</div>
<p>Calculate the SNP density:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ vcftools --gzvcf ril.vcf.gz   --SNPdensity 10000 --out ril
</pre></div>
</div>
</section>
</section>
</section>



        

    </main>

</div>
    <footer>
            <a rel="license" href="https://creativecommons.org/licenses/by/4.0/"><img alt="Licencia de Creative Commons" style="border-width:0" src="https://i.creativecommons.org/l/by/4.0/88x31.png" /></a><br /><span xmlns:dct="https://purl.org/dc/terms/" property="dct:title">Bioinformatics at COMAV</span> is licensed under a <a rel="license" href="https://creativecommons.org/licenses/by/4.0/">Creative Commons Reconocimiento 4.0 Internacional License</a>.
    </footer>

<script type="text/javascript">
var gaJsHost = (("https:" == document.location.protocol) ? "https://ssl." : "http://www.");
document.write(unescape("%3Cscript src='" + gaJsHost + "google-analytics.com/ga.js' type='text/javascript'%3E%3C/script%3E"));
</script>
<script type="text/javascript">
try {
var pageTracker = _gat._getTracker("UA-16108274-1");
pageTracker._trackPageview();
} catch(err) {}</script>

  </body>
</html>